<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">

<head>
  <meta charset="utf-8"/>
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1"/>
  <meta name="generator" content="distill" />

  <style type="text/css">
  /* Hide doc at startup (prevent jankiness while JS renders/transforms) */
  body {
    visibility: hidden;
  }
  </style>

 <!--radix_placeholder_import_source-->
 <!--/radix_placeholder_import_source-->

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  { color: #00769e; background-color: #f1f3f5; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span { color: #00769e; } /* Normal */
code span.al { color: #ad0000; } /* Alert */
code span.an { color: #5e5e5e; } /* Annotation */
code span.at { color: #657422; } /* Attribute */
code span.bn { color: #ad0000; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #00769e; } /* ControlFlow */
code span.ch { color: #20794d; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #5e5e5e; } /* Comment */
code span.cv { color: #5e5e5e; font-style: italic; } /* CommentVar */
code span.do { color: #5e5e5e; font-style: italic; } /* Documentation */
code span.dt { color: #ad0000; } /* DataType */
code span.dv { color: #ad0000; } /* DecVal */
code span.er { color: #ad0000; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #ad0000; } /* Float */
code span.fu { color: #4758ab; } /* Function */
code span.im { } /* Import */
code span.in { color: #5e5e5e; } /* Information */
code span.kw { color: #00769e; } /* Keyword */
code span.op { color: #5e5e5e; } /* Operator */
code span.ot { color: #00769e; } /* Other */
code span.pp { color: #ad0000; } /* Preprocessor */
code span.sc { color: #5e5e5e; } /* SpecialChar */
code span.ss { color: #20794d; } /* SpecialString */
code span.st { color: #20794d; } /* String */
code span.va { color: #111111; } /* Variable */
code span.vs { color: #20794d; } /* VerbatimString */
code span.wa { color: #5e5e5e; font-style: italic; } /* Warning */
</style>


  <!--radix_placeholder_meta_tags-->
  <title>Socs3 Knockout Bulk RNA-seq: Analysis of Filtered output 3</title>



  <!--  https://schema.org/Article -->
  <meta property="article:published" itemprop="datePublished" content="2025-06-26"/>
  <meta property="article:created" itemprop="dateCreated" content="2025-06-26"/>
  <meta name="article:author" content="Millie Dunstone and Natalie Koh"/>

  <!--  https://developers.facebook.com/docs/sharing/webmasters#markup -->
  <meta property="og:title" content="Socs3 Knockout Bulk RNA-seq: Analysis of Filtered output 3"/>
  <meta property="og:type" content="article"/>
  <meta property="og:locale" content="en_US"/>
  <meta property="og:site_name" content="Socs3 Knockout Bulk RNA-seq"/>

  <!--  https://dev.twitter.com/cards/types/summary -->
  <meta property="twitter:card" content="summary"/>
  <meta property="twitter:title" content="Socs3 Knockout Bulk RNA-seq: Analysis of Filtered output 3"/>

  <!--/radix_placeholder_meta_tags-->
  <!--radix_placeholder_rmarkdown_metadata-->

  <script type="text/json" id="radix-rmarkdown-metadata">
  {"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["title","output","date","author","editor_options"]}},"value":[{"type":"character","attributes":{},"value":["Analysis of Filtered output 3"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["html_document","pdf_document","word_document"]}},"value":[{"type":"character","attributes":{},"value":["default"]},{"type":"character","attributes":{},"value":["default"]},{"type":"character","attributes":{},"value":["default"]}]},{"type":"character","attributes":{},"value":["2025-06-26"]},{"type":"list","attributes":{},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name"]}},"value":[{"type":"character","attributes":{},"value":["Millie Dunstone and Natalie Koh"]}]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["chunk_output_type"]}},"value":[{"type":"character","attributes":{},"value":["console"]}]}]}
  </script>
  <!--/radix_placeholder_rmarkdown_metadata-->
  <!--radix_placeholder_navigation_in_header-->
  <meta name="distill:offset" content=""/>

  <script type="application/javascript">

    window.headroom_prevent_pin = false;

    window.document.addEventListener("DOMContentLoaded", function (event) {

      // initialize headroom for banner
      var header = $('header').get(0);
      var headerHeight = header.offsetHeight;
      var headroom = new Headroom(header, {
        tolerance: 5,
        onPin : function() {
          if (window.headroom_prevent_pin) {
            window.headroom_prevent_pin = false;
            headroom.unpin();
          }
        }
      });
      headroom.init();
      if(window.location.hash)
        headroom.unpin();
      $(header).addClass('headroom--transition');

      // offset scroll location for banner on hash change
      // (see: https://github.com/WickyNilliams/headroom.js/issues/38)
      window.addEventListener("hashchange", function(event) {
        window.scrollTo(0, window.pageYOffset - (headerHeight + 25));
      });

      // responsive menu
      $('.distill-site-header').each(function(i, val) {
        var topnav = $(this);
        var toggle = topnav.find('.nav-toggle');
        toggle.on('click', function() {
          topnav.toggleClass('responsive');
        });
      });

      // nav dropdowns
      $('.nav-dropbtn').click(function(e) {
        $(this).next('.nav-dropdown-content').toggleClass('nav-dropdown-active');
        $(this).parent().siblings('.nav-dropdown')
           .children('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $("body").click(function(e){
        $('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $(".nav-dropdown").click(function(e){
        e.stopPropagation();
      });
    });
  </script>

  <style type="text/css">

  /* Theme (user-documented overrideables for nav appearance) */

  .distill-site-nav {
    color: rgba(255, 255, 255, 0.8);
    background-color: #0F2E3D;
    font-size: 15px;
    font-weight: 300;
  }

  .distill-site-nav a {
    color: inherit;
    text-decoration: none;
  }

  .distill-site-nav a:hover {
    color: white;
  }

  @media print {
    .distill-site-nav {
      display: none;
    }
  }

  .distill-site-header {

  }

  .distill-site-footer {

  }


  /* Site Header */

  .distill-site-header {
    width: 100%;
    box-sizing: border-box;
    z-index: 3;
  }

  .distill-site-header .nav-left {
    display: inline-block;
    margin-left: 8px;
  }

  @media screen and (max-width: 768px) {
    .distill-site-header .nav-left {
      margin-left: 0;
    }
  }


  .distill-site-header .nav-right {
    float: right;
    margin-right: 8px;
  }

  .distill-site-header a,
  .distill-site-header .title {
    display: inline-block;
    text-align: center;
    padding: 14px 10px 14px 10px;
  }

  .distill-site-header .title {
    font-size: 18px;
    min-width: 150px;
  }

  .distill-site-header .logo {
    padding: 0;
  }

  .distill-site-header .logo img {
    display: none;
    max-height: 20px;
    width: auto;
    margin-bottom: -4px;
  }

  .distill-site-header .nav-image img {
    max-height: 18px;
    width: auto;
    display: inline-block;
    margin-bottom: -3px;
  }



  @media screen and (min-width: 1000px) {
    .distill-site-header .logo img {
      display: inline-block;
    }
    .distill-site-header .nav-left {
      margin-left: 20px;
    }
    .distill-site-header .nav-right {
      margin-right: 20px;
    }
    .distill-site-header .title {
      padding-left: 12px;
    }
  }


  .distill-site-header .nav-toggle {
    display: none;
  }

  .nav-dropdown {
    display: inline-block;
    position: relative;
  }

  .nav-dropdown .nav-dropbtn {
    border: none;
    outline: none;
    color: rgba(255, 255, 255, 0.8);
    padding: 16px 10px;
    background-color: transparent;
    font-family: inherit;
    font-size: inherit;
    font-weight: inherit;
    margin: 0;
    margin-top: 1px;
    z-index: 2;
  }

  .nav-dropdown-content {
    display: none;
    position: absolute;
    background-color: white;
    min-width: 200px;
    border: 1px solid rgba(0,0,0,0.15);
    border-radius: 4px;
    box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.1);
    z-index: 1;
    margin-top: 2px;
    white-space: nowrap;
    padding-top: 4px;
    padding-bottom: 4px;
  }

  .nav-dropdown-content hr {
    margin-top: 4px;
    margin-bottom: 4px;
    border: none;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }

  .nav-dropdown-active {
    display: block;
  }

  .nav-dropdown-content a, .nav-dropdown-content .nav-dropdown-header {
    color: black;
    padding: 6px 24px;
    text-decoration: none;
    display: block;
    text-align: left;
  }

  .nav-dropdown-content .nav-dropdown-header {
    display: block;
    padding: 5px 24px;
    padding-bottom: 0;
    text-transform: uppercase;
    font-size: 14px;
    color: #999999;
    white-space: nowrap;
  }

  .nav-dropdown:hover .nav-dropbtn {
    color: white;
  }

  .nav-dropdown-content a:hover {
    background-color: #ddd;
    color: black;
  }

  .nav-right .nav-dropdown-content {
    margin-left: -45%;
    right: 0;
  }

  @media screen and (max-width: 768px) {
    .distill-site-header a, .distill-site-header .nav-dropdown  {display: none;}
    .distill-site-header a.nav-toggle {
      float: right;
      display: block;
    }
    .distill-site-header .title {
      margin-left: 0;
    }
    .distill-site-header .nav-right {
      margin-right: 0;
    }
    .distill-site-header {
      overflow: hidden;
    }
    .nav-right .nav-dropdown-content {
      margin-left: 0;
    }
  }


  @media screen and (max-width: 768px) {
    .distill-site-header.responsive {position: relative; min-height: 500px; }
    .distill-site-header.responsive a.nav-toggle {
      position: absolute;
      right: 0;
      top: 0;
    }
    .distill-site-header.responsive a,
    .distill-site-header.responsive .nav-dropdown {
      display: block;
      text-align: left;
    }
    .distill-site-header.responsive .nav-left,
    .distill-site-header.responsive .nav-right {
      width: 100%;
    }
    .distill-site-header.responsive .nav-dropdown {float: none;}
    .distill-site-header.responsive .nav-dropdown-content {position: relative;}
    .distill-site-header.responsive .nav-dropdown .nav-dropbtn {
      display: block;
      width: 100%;
      text-align: left;
    }
  }

  /* Site Footer */

  .distill-site-footer {
    width: 100%;
    overflow: hidden;
    box-sizing: border-box;
    z-index: 3;
    margin-top: 30px;
    padding-top: 30px;
    padding-bottom: 30px;
    text-align: center;
  }

  /* Headroom */

  d-title {
    padding-top: 6rem;
  }

  @media print {
    d-title {
      padding-top: 4rem;
    }
  }

  .headroom {
    z-index: 1000;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
  }

  .headroom--transition {
    transition: all .4s ease-in-out;
  }

  .headroom--unpinned {
    top: -100px;
  }

  .headroom--pinned {
    top: 0;
  }

  /* adjust viewport for navbar height */
  /* helps vertically center bootstrap (non-distill) content */
  .min-vh-100 {
    min-height: calc(100vh - 100px) !important;
  }

  </style>

  <script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
  <link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"/>
  <link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"/>
  <script src="site_libs/headroom-0.9.4/headroom.min.js"></script>
  <script src="site_libs/autocomplete-0.37.1/autocomplete.min.js"></script>
  <script src="site_libs/fuse-6.4.1/fuse.min.js"></script>

  <script type="application/javascript">

  function getMeta(metaName) {
    var metas = document.getElementsByTagName('meta');
    for (let i = 0; i < metas.length; i++) {
      if (metas[i].getAttribute('name') === metaName) {
        return metas[i].getAttribute('content');
      }
    }
    return '';
  }

  function offsetURL(url) {
    var offset = getMeta('distill:offset');
    return offset ? offset + '/' + url : url;
  }

  function createFuseIndex() {

    // create fuse index
    var options = {
      keys: [
        { name: 'title', weight: 20 },
        { name: 'categories', weight: 15 },
        { name: 'description', weight: 10 },
        { name: 'contents', weight: 5 },
      ],
      ignoreLocation: true,
      threshold: 0
    };
    var fuse = new window.Fuse([], options);

    // fetch the main search.json
    return fetch(offsetURL('search.json'))
      .then(function(response) {
        if (response.status == 200) {
          return response.json().then(function(json) {
            // index main articles
            json.articles.forEach(function(article) {
              fuse.add(article);
            });
            // download collections and index their articles
            return Promise.all(json.collections.map(function(collection) {
              return fetch(offsetURL(collection)).then(function(response) {
                if (response.status === 200) {
                  return response.json().then(function(articles) {
                    articles.forEach(function(article) {
                      fuse.add(article);
                    });
                  })
                } else {
                  return Promise.reject(
                    new Error('Unexpected status from search index request: ' +
                              response.status)
                  );
                }
              });
            })).then(function() {
              return fuse;
            });
          });

        } else {
          return Promise.reject(
            new Error('Unexpected status from search index request: ' +
                        response.status)
          );
        }
      });
  }

  window.document.addEventListener("DOMContentLoaded", function (event) {

    // get search element (bail if we don't have one)
    var searchEl = window.document.getElementById('distill-search');
    if (!searchEl)
      return;

    createFuseIndex()
      .then(function(fuse) {

        // make search box visible
        searchEl.classList.remove('hidden');

        // initialize autocomplete
        var options = {
          autoselect: true,
          hint: false,
          minLength: 2,
        };
        window.autocomplete(searchEl, options, [{
          source: function(query, callback) {
            const searchOptions = {
              isCaseSensitive: false,
              shouldSort: true,
              minMatchCharLength: 2,
              limit: 10,
            };
            var results = fuse.search(query, searchOptions);
            callback(results
              .map(function(result) { return result.item; })
            );
          },
          templates: {
            suggestion: function(suggestion) {
              var img = suggestion.preview && Object.keys(suggestion.preview).length > 0
                ? `<img src="${offsetURL(suggestion.preview)}"</img>`
                : '';
              var html = `
                <div class="search-item">
                  <h3>${suggestion.title}</h3>
                  <div class="search-item-description">
                    ${suggestion.description || ''}
                  </div>
                  <div class="search-item-preview">
                    ${img}
                  </div>
                </div>
              `;
              return html;
            }
          }
        }]).on('autocomplete:selected', function(event, suggestion) {
          window.location.href = offsetURL(suggestion.path);
        });
        // remove inline display style on autocompleter (we want to
        // manage responsive display via css)
        $('.algolia-autocomplete').css("display", "");
      })
      .catch(function(error) {
        console.log(error);
      });

  });

  </script>

  <style type="text/css">

  .nav-search {
    font-size: x-small;
  }

  /* Algolioa Autocomplete */

  .algolia-autocomplete {
    display: inline-block;
    margin-left: 10px;
    vertical-align: sub;
    background-color: white;
    color: black;
    padding: 6px;
    padding-top: 8px;
    padding-bottom: 0;
    border-radius: 6px;
    border: 1px #0F2E3D solid;
    width: 180px;
  }


  @media screen and (max-width: 768px) {
    .distill-site-nav .algolia-autocomplete {
      display: none;
      visibility: hidden;
    }
    .distill-site-nav.responsive .algolia-autocomplete {
      display: inline-block;
      visibility: visible;
    }
    .distill-site-nav.responsive .algolia-autocomplete .aa-dropdown-menu {
      margin-left: 0;
      width: 400px;
      max-height: 400px;
    }
  }

  .algolia-autocomplete .aa-input, .algolia-autocomplete .aa-hint {
    width: 90%;
    outline: none;
    border: none;
  }

  .algolia-autocomplete .aa-hint {
    color: #999;
  }
  .algolia-autocomplete .aa-dropdown-menu {
    width: 550px;
    max-height: 70vh;
    overflow-x: visible;
    overflow-y: scroll;
    padding: 5px;
    margin-top: 3px;
    margin-left: -150px;
    background-color: #fff;
    border-radius: 5px;
    border: 1px solid #999;
    border-top: none;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion {
    cursor: pointer;
    padding: 5px 4px;
    border-bottom: 1px solid #eee;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion:last-of-type {
    border-bottom: none;
    margin-bottom: 2px;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item {
    overflow: hidden;
    font-size: 0.8em;
    line-height: 1.4em;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item h3 {
    font-size: 1rem;
    margin-block-start: 0;
    margin-block-end: 5px;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-description {
    display: inline-block;
    overflow: hidden;
    height: 2.8em;
    width: 80%;
    margin-right: 4%;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview {
    display: inline-block;
    width: 15%;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview img {
    height: 3em;
    width: auto;
    display: none;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview img[src] {
    display: initial;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion.aa-cursor {
    background-color: #eee;
  }
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion em {
    font-weight: bold;
    font-style: normal;
  }

  </style>


  <!--/radix_placeholder_navigation_in_header-->
  <!--radix_placeholder_distill-->

  <style type="text/css">

  body {
    background-color: white;
  }

  .pandoc-table {
    width: 100%;
  }

  .pandoc-table>caption {
    margin-bottom: 10px;
  }

  .pandoc-table th:not([align]) {
    text-align: left;
  }

  .pagedtable-footer {
    font-size: 15px;
  }

  d-byline .byline {
    grid-template-columns: 2fr 2fr;
  }

  d-byline .byline h3 {
    margin-block-start: 1.5em;
  }

  d-byline .byline .authors-affiliations h3 {
    margin-block-start: 0.5em;
  }

  .authors-affiliations .orcid-id {
    width: 16px;
    height:16px;
    margin-left: 4px;
    margin-right: 4px;
    vertical-align: middle;
    padding-bottom: 2px;
  }

  d-title .dt-tags {
    margin-top: 1em;
    grid-column: text;
  }

  .dt-tags .dt-tag {
    text-decoration: none;
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0em 0.4em;
    margin-right: 0.5em;
    margin-bottom: 0.4em;
    font-size: 70%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }

  d-article table.gt_table td,
  d-article table.gt_table th {
    border-bottom: none;
    font-size: 100%;
  }

  .html-widget {
    margin-bottom: 2.0em;
  }

  .l-screen-inset {
    padding-right: 16px;
  }

  .l-screen .caption {
    margin-left: 10px;
  }

  .shaded {
    background: rgb(247, 247, 247);
    padding-top: 20px;
    padding-bottom: 20px;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }

  .shaded .html-widget {
    margin-bottom: 0;
    border: 1px solid rgba(0, 0, 0, 0.1);
  }

  .shaded .shaded-content {
    background: white;
  }

  .text-output {
    margin-top: 0;
    line-height: 1.5em;
  }

  .hidden {
    display: none !important;
  }

  hr.section-separator {
    border: none;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    margin: 0px;
  }


  d-byline {
    border-top: none;
  }

  d-article {
    padding-top: 2.5rem;
    padding-bottom: 30px;
    border-top: none;
  }

  d-appendix {
    padding-top: 30px;
  }

  d-article>p>img {
    width: 100%;
  }

  d-article h2 {
    margin: 1rem 0 1.5rem 0;
  }

  d-article h3 {
    margin-top: 1.5rem;
  }

  d-article iframe {
    border: 1px solid rgba(0, 0, 0, 0.1);
    margin-bottom: 2.0em;
    width: 100%;
  }

  /* Tweak code blocks */

  d-article div.sourceCode code,
  d-article pre code {
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
  }

  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: auto;
  }

  d-article div.sourceCode {
    background-color: white;
  }

  d-article div.sourceCode pre {
    padding-left: 10px;
    font-size: 12px;
    border-left: 2px solid rgba(0,0,0,0.1);
  }

  d-article pre {
    font-size: 12px;
    color: black;
    background: none;
    margin-top: 0;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;

    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;

    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }

  d-article pre a {
    border-bottom: none;
  }

  d-article pre a:hover {
    border-bottom: none;
    text-decoration: underline;
  }

  d-article details {
    grid-column: text;
    margin-bottom: 0.8em;
  }

  @media(min-width: 768px) {

  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: visible !important;
  }

  d-article div.sourceCode pre {
    padding-left: 18px;
    font-size: 14px;
  }

  /* tweak for Pandoc numbered line within distill */
  d-article pre.numberSource code > span {
      left: -2em;
  }

  d-article pre {
    font-size: 14px;
  }

  }

  figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }

  /* CSS for d-contents */

  .d-contents {
    grid-column: text;
    color: rgba(0,0,0,0.8);
    font-size: 0.9em;
    padding-bottom: 1em;
    margin-bottom: 1em;
    padding-bottom: 0.5em;
    margin-bottom: 1em;
    padding-left: 0.25em;
    justify-self: start;
  }

  @media(min-width: 1000px) {
    .d-contents.d-contents-float {
      height: 0;
      grid-column-start: 1;
      grid-column-end: 4;
      justify-self: center;
      padding-right: 3em;
      padding-left: 2em;
    }
  }

  .d-contents nav h3 {
    font-size: 18px;
    margin-top: 0;
    margin-bottom: 1em;
  }

  .d-contents li {
    list-style-type: none
  }

  .d-contents nav > ul {
    padding-left: 0;
  }

  .d-contents ul {
    padding-left: 1em
  }

  .d-contents nav ul li {
    margin-top: 0.6em;
    margin-bottom: 0.2em;
  }

  .d-contents nav a {
    font-size: 13px;
    border-bottom: none;
    text-decoration: none
    color: rgba(0, 0, 0, 0.8);
  }

  .d-contents nav a:hover {
    text-decoration: underline solid rgba(0, 0, 0, 0.6)
  }

  .d-contents nav > ul > li > a {
    font-weight: 600;
  }

  .d-contents nav > ul > li > ul {
    font-weight: inherit;
  }

  .d-contents nav > ul > li > ul > li {
    margin-top: 0.2em;
  }


  .d-contents nav ul {
    margin-top: 0;
    margin-bottom: 0.25em;
  }

  .d-article-with-toc h2:nth-child(2) {
    margin-top: 0;
  }


  /* Figure */

  .figure {
    position: relative;
    margin-bottom: 2.5em;
    margin-top: 1.5em;
  }

  .figure .caption {
    color: rgba(0, 0, 0, 0.6);
    font-size: 12px;
    line-height: 1.5em;
  }

  .figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }

  .figure .caption a {
    color: rgba(0, 0, 0, 0.6);
  }

  .figure .caption b,
  .figure .caption strong, {
    font-weight: 600;
    color: rgba(0, 0, 0, 1.0);
  }

  /* Citations */

  d-article .citation {
    color: inherit;
    cursor: inherit;
  }

  div.hanging-indent{
    margin-left: 1em; text-indent: -1em;
  }

  /* Citation hover box */

  .tippy-box[data-theme~=light-border] {
    background-color: rgba(250, 250, 250, 0.95);
  }

  .tippy-content > p {
    margin-bottom: 0;
    padding: 2px;
  }


  /* Tweak 1000px media break to show more text */

  @media(min-width: 1000px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 80px [middle-start] 50px [text-start kicker-end] 65px 65px 65px 65px 65px 65px 65px 65px [text-end gutter-start] 65px [middle-end] 65px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 16px;
    }

    .grid {
      grid-column-gap: 16px;
    }

    d-article {
      font-size: 1.06rem;
      line-height: 1.7em;
    }
    figure .caption, .figure .caption, figure figcaption {
      font-size: 13px;
    }
  }

  @media(min-width: 1180px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 60px [middle-start] 60px [text-start kicker-end] 60px 60px 60px 60px 60px 60px 60px 60px [text-end gutter-start] 60px [middle-end] 60px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 32px;
    }

    .grid {
      grid-column-gap: 32px;
    }
  }


  /* Get the citation styles for the appendix (not auto-injected on render since
     we do our own rendering of the citation appendix) */

  d-appendix .citation-appendix,
  .d-appendix .citation-appendix {
    font-size: 11px;
    line-height: 15px;
    border-left: 1px solid rgba(0, 0, 0, 0.1);
    padding-left: 18px;
    border: 1px solid rgba(0,0,0,0.1);
    background: rgba(0, 0, 0, 0.02);
    padding: 10px 18px;
    border-radius: 3px;
    color: rgba(150, 150, 150, 1);
    overflow: hidden;
    margin-top: -12px;
    white-space: pre-wrap;
    word-wrap: break-word;
  }

  /* Include appendix styles here so they can be overridden */

  d-appendix {
    contain: layout style;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-top: 60px;
    margin-bottom: 0;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    color: rgba(0,0,0,0.5);
    padding-top: 60px;
    padding-bottom: 48px;
  }

  d-appendix h3 {
    grid-column: page-start / text-start;
    font-size: 15px;
    font-weight: 500;
    margin-top: 1em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.65);
  }

  d-appendix h3 + * {
    margin-top: 1em;
  }

  d-appendix ol {
    padding: 0 0 0 15px;
  }

  @media (min-width: 768px) {
    d-appendix ol {
      padding: 0 0 0 30px;
      margin-left: -30px;
    }
  }

  d-appendix li {
    margin-bottom: 1em;
  }

  d-appendix a {
    color: rgba(0, 0, 0, 0.6);
  }

  d-appendix > * {
    grid-column: text;
  }

  d-appendix > d-footnote-list,
  d-appendix > d-citation-list,
  d-appendix > distill-appendix {
    grid-column: screen;
  }

  /* Include footnote styles here so they can be overridden */

  d-footnote-list {
    contain: layout style;
  }

  d-footnote-list > * {
    grid-column: text;
  }

  d-footnote-list a.footnote-backlink {
    color: rgba(0,0,0,0.3);
    padding-left: 0.5em;
  }



  /* Anchor.js */

  .anchorjs-link {
    /*transition: all .25s linear; */
    text-decoration: none;
    border-bottom: none;
  }
  *:hover > .anchorjs-link {
    margin-left: -1.125em !important;
    text-decoration: none;
    border-bottom: none;
  }

  /* Social footer */

  .social_footer {
    margin-top: 30px;
    margin-bottom: 0;
    color: rgba(0,0,0,0.67);
  }

  .disqus-comments {
    margin-right: 30px;
  }

  .disqus-comment-count {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
    cursor: pointer;
  }

  #disqus_thread {
    margin-top: 30px;
  }

  .article-sharing a {
    border-bottom: none;
    margin-right: 8px;
  }

  .article-sharing a:hover {
    border-bottom: none;
  }

  .sidebar-section.subscribe {
    font-size: 12px;
    line-height: 1.6em;
  }

  .subscribe p {
    margin-bottom: 0.5em;
  }


  .article-footer .subscribe {
    font-size: 15px;
    margin-top: 45px;
  }


  .sidebar-section.custom {
    font-size: 12px;
    line-height: 1.6em;
  }

  .custom p {
    margin-bottom: 0.5em;
  }

  /* Styles for listing layout (hide title) */
  .layout-listing d-title, .layout-listing .d-title {
    display: none;
  }

  /* Styles for posts lists (not auto-injected) */


  .posts-with-sidebar {
    padding-left: 45px;
    padding-right: 45px;
  }

  .posts-list .description h2,
  .posts-list .description p {
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
  }

  .posts-list .description h2 {
    font-weight: 700;
    border-bottom: none;
    padding-bottom: 0;
  }

  .posts-list h2.post-tag {
    border-bottom: 1px solid rgba(0, 0, 0, 0.2);
    padding-bottom: 12px;
  }
  .posts-list {
    margin-top: 60px;
    margin-bottom: 24px;
  }

  .posts-list .post-preview {
    text-decoration: none;
    overflow: hidden;
    display: block;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
    padding: 24px 0;
  }

  .post-preview-last {
    border-bottom: none !important;
  }

  .posts-list .posts-list-caption {
    grid-column: screen;
    font-weight: 400;
  }

  .posts-list .post-preview h2 {
    margin: 0 0 6px 0;
    line-height: 1.2em;
    font-style: normal;
    font-size: 24px;
  }

  .posts-list .post-preview p {
    margin: 0 0 12px 0;
    line-height: 1.4em;
    font-size: 16px;
  }

  .posts-list .post-preview .thumbnail {
    box-sizing: border-box;
    margin-bottom: 24px;
    position: relative;
    max-width: 500px;
  }
  .posts-list .post-preview img {
    width: 100%;
    display: block;
  }

  .posts-list .metadata {
    font-size: 12px;
    line-height: 1.4em;
    margin-bottom: 18px;
  }

  .posts-list .metadata > * {
    display: inline-block;
  }

  .posts-list .metadata .publishedDate {
    margin-right: 2em;
  }

  .posts-list .metadata .dt-authors {
    display: block;
    margin-top: 0.3em;
    margin-right: 2em;
  }

  .posts-list .dt-tags {
    display: block;
    line-height: 1em;
  }

  .posts-list .dt-tags .dt-tag {
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0.3em 0.4em;
    margin-right: 0.2em;
    margin-bottom: 0.4em;
    font-size: 60%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }

  .posts-list img {
    opacity: 1;
  }

  .posts-list img[data-src] {
    opacity: 0;
  }

  .posts-more {
    clear: both;
  }


  .posts-sidebar {
    font-size: 16px;
  }

  .posts-sidebar h3 {
    font-size: 16px;
    margin-top: 0;
    margin-bottom: 0.5em;
    font-weight: 400;
    text-transform: uppercase;
  }

  .sidebar-section {
    margin-bottom: 30px;
  }

  .categories ul {
    list-style-type: none;
    margin: 0;
    padding: 0;
  }

  .categories li {
    color: rgba(0, 0, 0, 0.8);
    margin-bottom: 0;
  }

  .categories li>a {
    border-bottom: none;
  }

  .categories li>a:hover {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
  }

  .categories .active {
    font-weight: 600;
  }

  .categories .category-count {
    color: rgba(0, 0, 0, 0.4);
  }


  @media(min-width: 768px) {
    .posts-list .post-preview h2 {
      font-size: 26px;
    }
    .posts-list .post-preview .thumbnail {
      float: right;
      width: 30%;
      margin-bottom: 0;
    }
    .posts-list .post-preview .description {
      float: left;
      width: 45%;
    }
    .posts-list .post-preview .metadata {
      float: left;
      width: 20%;
      margin-top: 8px;
    }
    .posts-list .post-preview p {
      margin: 0 0 12px 0;
      line-height: 1.5em;
      font-size: 16px;
    }
    .posts-with-sidebar .posts-list {
      float: left;
      width: 75%;
    }
    .posts-with-sidebar .posts-sidebar {
      float: right;
      width: 20%;
      margin-top: 60px;
      padding-top: 24px;
      padding-bottom: 24px;
    }
  }


  /* Improve display for browsers without grid (IE/Edge <= 15) */

  .downlevel {
    line-height: 1.6em;
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
    margin: 0;
  }

  .downlevel .d-title {
    padding-top: 6rem;
    padding-bottom: 1.5rem;
  }

  .downlevel .d-title h1 {
    font-size: 50px;
    font-weight: 700;
    line-height: 1.1em;
    margin: 0 0 0.5rem;
  }

  .downlevel .d-title p {
    font-weight: 300;
    font-size: 1.2rem;
    line-height: 1.55em;
    margin-top: 0;
  }

  .downlevel .d-byline {
    padding-top: 0.8em;
    padding-bottom: 0.8em;
    font-size: 0.8rem;
    line-height: 1.8em;
  }

  .downlevel .section-separator {
    border: none;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
  }

  .downlevel .d-article {
    font-size: 1.06rem;
    line-height: 1.7em;
    padding-top: 1rem;
    padding-bottom: 2rem;
  }


  .downlevel .d-appendix {
    padding-left: 0;
    padding-right: 0;
    max-width: none;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.5);
    padding-top: 40px;
    padding-bottom: 48px;
  }

  .downlevel .footnotes ol {
    padding-left: 13px;
  }

  .downlevel .base-grid,
  .downlevel .distill-header,
  .downlevel .d-title,
  .downlevel .d-abstract,
  .downlevel .d-article,
  .downlevel .d-appendix,
  .downlevel .distill-appendix,
  .downlevel .d-byline,
  .downlevel .d-footnote-list,
  .downlevel .d-citation-list,
  .downlevel .distill-footer,
  .downlevel .appendix-bottom,
  .downlevel .posts-container {
    padding-left: 40px;
    padding-right: 40px;
  }

  @media(min-width: 768px) {
    .downlevel .base-grid,
    .downlevel .distill-header,
    .downlevel .d-title,
    .downlevel .d-abstract,
    .downlevel .d-article,
    .downlevel .d-appendix,
    .downlevel .distill-appendix,
    .downlevel .d-byline,
    .downlevel .d-footnote-list,
    .downlevel .d-citation-list,
    .downlevel .distill-footer,
    .downlevel .appendix-bottom,
    .downlevel .posts-container {
    padding-left: 150px;
    padding-right: 150px;
    max-width: 900px;
  }
  }

  .downlevel pre code {
    display: block;
    border-left: 2px solid rgba(0, 0, 0, .1);
    padding: 0 0 0 20px;
    font-size: 14px;
  }

  .downlevel code, .downlevel pre {
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;

    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;

    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }

  .downlevel .posts-list .post-preview {
    color: inherit;
  }



  </style>

  <script type="application/javascript">

  function is_downlevel_browser() {
    if (bowser.isUnsupportedBrowser({ msie: "12", msedge: "16"},
                                   window.navigator.userAgent)) {
      return true;
    } else {
      return window.load_distill_framework === undefined;
    }
  }

  // show body when load is complete
  function on_load_complete() {

    // add anchors
    if (window.anchors) {
      window.anchors.options.placement = 'left';
      window.anchors.add('d-article > h2, d-article > h3, d-article > h4, d-article > h5');
    }


    // set body to visible
    document.body.style.visibility = 'visible';

    // force redraw for leaflet widgets
    if (window.HTMLWidgets) {
      var maps = window.HTMLWidgets.findAll(".leaflet");
      $.each(maps, function(i, el) {
        var map = this.getMap();
        map.invalidateSize();
        map.eachLayer(function(layer) {
          if (layer instanceof L.TileLayer)
            layer.redraw();
        });
      });
    }

    // trigger 'shown' so htmlwidgets resize
    $('d-article').trigger('shown');
  }

  function init_distill() {

    init_common();

    // create front matter
    var front_matter = $('<d-front-matter></d-front-matter>');
    $('#distill-front-matter').wrap(front_matter);

    // create d-title
    $('.d-title').changeElementType('d-title');

    // separator
    var separator = '<hr class="section-separator" style="clear: both"/>';
    // prepend separator above appendix
    $('.d-byline').before(separator);
    $('.d-article').before(separator);

    // create d-byline
    var byline = $('<d-byline></d-byline>');
    $('.d-byline').replaceWith(byline);

    // create d-article
    var article = $('<d-article></d-article>');
    $('.d-article').wrap(article).children().unwrap();

    // move posts container into article
    $('.posts-container').appendTo($('d-article'));

    // create d-appendix
    $('.d-appendix').changeElementType('d-appendix');

    // flag indicating that we have appendix items
    var appendix = $('.appendix-bottom').children('h3').length > 0;

    // replace footnotes with <d-footnote>
    $('.footnote-ref').each(function(i, val) {
      appendix = true;
      var href = $(this).attr('href');
      var id = href.replace('#', '');
      var fn = $('#' + id);
      var fn_p = $('#' + id + '>p');
      fn_p.find('.footnote-back').remove();
      var text = fn_p.html();
      var dtfn = $('<d-footnote></d-footnote>');
      dtfn.html(text);
      $(this).replaceWith(dtfn);
    });
    // remove footnotes
    $('.footnotes').remove();

    // move refs into #references-listing
    $('#references-listing').replaceWith($('#refs'));

    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      var id = $(this).attr('id');
      $('.d-contents a[href="#' + id + '"]').parent().remove();
      appendix = true;
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('d-appendix'));
    });

    // show d-appendix if we have appendix content
    $("d-appendix").css('display', appendix ? 'grid' : 'none');

    // localize layout chunks to just output
    $('.layout-chunk').each(function(i, val) {

      // capture layout
      var layout = $(this).attr('data-layout');

      // apply layout to markdown level block elements
      var elements = $(this).children().not('details, div.sourceCode, pre, script');
      elements.each(function(i, el) {
        var layout_div = $('<div class="' + layout + '"></div>');
        if (layout_div.hasClass('shaded')) {
          var shaded_content = $('<div class="shaded-content"></div>');
          $(this).wrap(shaded_content);
          $(this).parent().wrap(layout_div);
        } else {
          $(this).wrap(layout_div);
        }
      });


      // unwrap the layout-chunk div
      $(this).children().unwrap();
    });

    // remove code block used to force  highlighting css
    $('.distill-force-highlighting-css').parent().remove();

    // remove empty line numbers inserted by pandoc when using a
    // custom syntax highlighting theme, except when numbering line
    // in code chunk
    $('pre:not(.numberLines) code.sourceCode a:empty').remove();

    // load distill framework
    load_distill_framework();

    // wait for window.distillRunlevel == 4 to do post processing
    function distill_post_process() {

      if (!window.distillRunlevel || window.distillRunlevel < 4)
        return;

      // hide author/affiliations entirely if we have no authors
      var front_matter = JSON.parse($("#distill-front-matter").html());
      var have_authors = front_matter.authors && front_matter.authors.length > 0;
      if (!have_authors)
        $('d-byline').addClass('hidden');

      // article with toc class
      $('.d-contents').parent().addClass('d-article-with-toc');

      // strip links that point to #
      $('.authors-affiliations').find('a[href="#"]').removeAttr('href');

      // add orcid ids
      $('.authors-affiliations').find('.author').each(function(i, el) {
        var orcid_id = front_matter.authors[i].orcidID;
        var author_name = front_matter.authors[i].author
        if (orcid_id) {
          var a = $('<a></a>');
          a.attr('href', 'https://orcid.org/' + orcid_id);
          var img = $('<img></img>');
          img.addClass('orcid-id');
          img.attr('alt', author_name ? 'ORCID ID for ' + author_name : 'ORCID ID');
          img.attr('src','data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg==');
          a.append(img);
          $(this).append(a);
        }
      });

      // hide elements of author/affiliations grid that have no value
      function hide_byline_column(caption) {
        $('d-byline').find('h3:contains("' + caption + '")').parent().css('visibility', 'hidden');
      }

      // affiliations
      var have_affiliations = false;
      for (var i = 0; i<front_matter.authors.length; ++i) {
        var author = front_matter.authors[i];
        if (author.affiliation !== "&nbsp;") {
          have_affiliations = true;
          break;
        }
      }
      if (!have_affiliations)
        $('d-byline').find('h3:contains("Affiliations")').css('visibility', 'hidden');

      // published date
      if (!front_matter.publishedDate)
        hide_byline_column("Published");

      // document object identifier
      var doi = $('d-byline').find('h3:contains("DOI")');
      var doi_p = doi.next().empty();
      if (!front_matter.doi) {
        // if we have a citation and valid citationText then link to that
        if ($('#citation').length > 0 && front_matter.citationText) {
          doi.html('Citation');
          $('<a href="#citation"></a>')
            .text(front_matter.citationText)
            .appendTo(doi_p);
        } else {
          hide_byline_column("DOI");
        }
      } else {
        $('<a></a>')
           .attr('href', "https://doi.org/" + front_matter.doi)
           .html(front_matter.doi)
           .appendTo(doi_p);
      }

       // change plural form of authors/affiliations
      if (front_matter.authors.length === 1) {
        var grid = $('.authors-affiliations');
        grid.children('h3:contains("Authors")').text('Author');
        grid.children('h3:contains("Affiliations")').text('Affiliation');
      }

      // remove d-appendix and d-footnote-list local styles
      $('d-appendix > style:first-child').remove();
      $('d-footnote-list > style:first-child').remove();

      // move appendix-bottom entries to the bottom
      $('.appendix-bottom').appendTo('d-appendix').children().unwrap();
      $('.appendix-bottom').remove();

      // hoverable references
      $('span.citation[data-cites]').each(function() {
        const citeChild = $(this).children()[0]
        // Do not process if @xyz has been used without escaping and without bibliography activated
        // https://github.com/rstudio/distill/issues/466
        if (citeChild === undefined) return true

        if (citeChild.nodeName == "D-FOOTNOTE") {
          var fn = citeChild
          $(this).html(fn.shadowRoot.querySelector("sup"))
          $(this).id = fn.id
          fn.remove()
        }
        var refs = $(this).attr('data-cites').split(" ");
        var refHtml = refs.map(function(ref) {
          // Could use CSS.escape too here, we insure backward compatibility in navigator
          return "<p>" + $('div[id="ref-' + ref + '"]').html() + "</p>";
        }).join("\n");
        window.tippy(this, {
          allowHTML: true,
          content: refHtml,
          maxWidth: 500,
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start'
        });
      });

      // fix footnotes in tables (#411)
      // replacing broken distill.pub feature
      $('table d-footnote').each(function() {
        // we replace internal showAtNode methode which is triggered when hovering a footnote
        this.hoverBox.showAtNode = function(node) {
          // ported from https://github.com/distillpub/template/pull/105/files
          calcOffset = function(elem) {
              let x = elem.offsetLeft;
              let y = elem.offsetTop;
              // Traverse upwards until an `absolute` element is found or `elem`
              // becomes null.
              while (elem = elem.offsetParent && elem.style.position != 'absolute') {
                  x += elem.offsetLeft;
                  y += elem.offsetTop;
              }

              return { left: x, top: y };
          }
          // https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/offsetTop
          const bbox = node.getBoundingClientRect();
          const offset = calcOffset(node);
          this.show([offset.left + bbox.width, offset.top + bbox.height]);
        }
      })

      // clear polling timer
      clearInterval(tid);

      // show body now that everything is ready
      on_load_complete();
    }

    var tid = setInterval(distill_post_process, 50);
    distill_post_process();

  }

  function init_downlevel() {

    init_common();

     // insert hr after d-title
    $('.d-title').after($('<hr class="section-separator"/>'));

    // check if we have authors
    var front_matter = JSON.parse($("#distill-front-matter").html());
    var have_authors = front_matter.authors && front_matter.authors.length > 0;

    // manage byline/border
    if (!have_authors)
      $('.d-byline').remove();
    $('.d-byline').after($('<hr class="section-separator"/>'));
    $('.d-byline a').remove();

    // remove toc
    $('.d-contents').remove();

    // move appendix elements
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('.d-appendix'));
    });


    // inject headers into references and footnotes
    var refs_header = $('<h3></h3>');
    refs_header.text('References');
    $('#refs').prepend(refs_header);

    var footnotes_header = $('<h3></h3');
    footnotes_header.text('Footnotes');
    $('.footnotes').children('hr').first().replaceWith(footnotes_header);

    // move appendix-bottom entries to the bottom
    $('.appendix-bottom').appendTo('.d-appendix').children().unwrap();
    $('.appendix-bottom').remove();

    // remove appendix if it's empty
    if ($('.d-appendix').children().length === 0)
      $('.d-appendix').remove();

    // prepend separator above appendix
    $('.d-appendix').before($('<hr class="section-separator" style="clear: both"/>'));

    // trim code
    $('pre>code').each(function(i, val) {
      $(this).html($.trim($(this).html()));
    });

    // move posts-container right before article
    $('.posts-container').insertBefore($('.d-article'));

    $('body').addClass('downlevel');

    on_load_complete();
  }


  function init_common() {

    // jquery plugin to change element types
    (function($) {
      $.fn.changeElementType = function(newType) {
        var attrs = {};

        $.each(this[0].attributes, function(idx, attr) {
          attrs[attr.nodeName] = attr.nodeValue;
        });

        this.replaceWith(function() {
          return $("<" + newType + "/>", attrs).append($(this).contents());
        });
      };
    })(jQuery);

    // prevent underline for linked images
    $('a > img').parent().css({'border-bottom' : 'none'});

    // mark non-body figures created by knitr chunks as 100% width
    $('.layout-chunk').each(function(i, val) {
      var figures = $(this).find('img, .html-widget');
      // ignore leaflet img layers (#106)
      figures = figures.filter(':not(img[class*="leaflet"])')
      if ($(this).attr('data-layout') !== "l-body") {
        figures.css('width', '100%');
      } else {
        figures.css('max-width', '100%');
        figures.filter("[width]").each(function(i, val) {
          var fig = $(this);
          fig.css('width', fig.attr('width') + 'px');
        });

      }
    });

    // auto-append index.html to post-preview links in file: protocol
    // and in rstudio ide preview
    $('.post-preview').each(function(i, val) {
      if (window.location.protocol === "file:")
        $(this).attr('href', $(this).attr('href') + "index.html");
    });

    // get rid of index.html references in header
    if (window.location.protocol !== "file:") {
      $('.distill-site-header a[href]').each(function(i,val) {
        $(this).attr('href', $(this).attr('href').replace(/^index[.]html/, "./"));
      });
    }

    // add class to pandoc style tables
    $('tr.header').parent('thead').parent('table').addClass('pandoc-table');
    $('.kable-table').children('table').addClass('pandoc-table');

    // add figcaption style to table captions
    $('caption').parent('table').addClass("figcaption");

    // initialize posts list
    if (window.init_posts_list)
      window.init_posts_list();

    // implmement disqus comment link
    $('.disqus-comment-count').click(function() {
      window.headroom_prevent_pin = true;
      $('#disqus_thread').toggleClass('hidden');
      if (!$('#disqus_thread').hasClass('hidden')) {
        var offset = $(this).offset();
        $(window).resize();
        $('html, body').animate({
          scrollTop: offset.top - 35
        });
      }
    });
  }

  document.addEventListener('DOMContentLoaded', function() {
    if (is_downlevel_browser())
      init_downlevel();
    else
      window.addEventListener('WebComponentsReady', init_distill);
  });

  </script>

  <!--/radix_placeholder_distill-->
  <script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
  <script src="site_libs/popper-2.6.0/popper.min.js"></script>
  <link href="site_libs/tippy-6.2.7/tippy.css" rel="stylesheet" />
  <link href="site_libs/tippy-6.2.7/tippy-light-border.css" rel="stylesheet" />
  <script src="site_libs/tippy-6.2.7/tippy.umd.min.js"></script>
  <script src="site_libs/anchor-4.2.2/anchor.min.js"></script>
  <script src="site_libs/bowser-1.9.3/bowser.min.js"></script>
  <script src="site_libs/webcomponents-2.0.0/webcomponents.js"></script>
  <script src="site_libs/distill-2.2.21/template.v2.js"></script>
  <!--radix_placeholder_site_in_header-->
  <!--/radix_placeholder_site_in_header-->


</head>

<body>

<!--radix_placeholder_front_matter-->

<script id="distill-front-matter" type="text/json">
{"title":"Analysis of Filtered output 3","authors":[{"author":"Millie Dunstone and Natalie Koh","authorURL":"#","affiliation":"&nbsp;","affiliationURL":"#","orcidID":""}],"publishedDate":"2025-06-26T00:00:00.000+10:00","citationText":"Koh, 2025"}
</script>

<!--/radix_placeholder_front_matter-->
<!--radix_placeholder_navigation_before_body-->
<header class="header header--fixed" role="banner">
<nav class="distill-site-nav distill-site-header">
<div class="nav-left">
<a href="index.html" class="title">Socs3 Knockout Bulk RNA-seq</a>
<a href="index.html">Home</a>
<div class="nav-dropdown">
<button class="nav-dropbtn">
Analysis
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<a href="Kallisto_analysis.html">Kallisto analysis</a>
<a href="Filtered_output_3_analysis.html">Analysis of filtered output 3</a>
</div>
</div>
</div>
<div class="nav-right">
<a href="https://gitlab.svi.edu.au/igr-lab/Socs3_Knockout_RNA" aria-label="Source code">
<i class="fab fa-github" aria-hidden="true"></i>
</a>
<a href="javascript:void(0);" class="nav-toggle">&#9776;</a>
</div>
</nav>
</header>
<!--/radix_placeholder_navigation_before_body-->
<!--radix_placeholder_site_before_body-->
<!--/radix_placeholder_site_before_body-->

<div class="d-title">
<h1>Analysis of Filtered output 3</h1>

<!--radix_placeholder_categories-->
<!--/radix_placeholder_categories-->

</div>

<div class="d-byline">
  
  Millie Dunstone and Natalie Koh
  
<br/>2025-06-26
</div>

<div class="d-article">
<div class="d-contents d-contents-float">
<nav class="l-text toc figcaption" id="TOC">
<h3>Contents</h3>
<ul>
<li><a href="#introduction" id="toc-introduction">Introduction</a></li>
<li><a href="#setting-up-the-data" id="toc-setting-up-the-data">Setting up the data</a></li>
<li><a href="#pre-processing" id="toc-pre-processing">Pre-processing</a></li>
<li><a href="#visualising-the-data" id="toc-visualising-the-data">Visualising the data</a></li>
<li><a href="#differential-expression-analyses" id="toc-differential-expression-analyses">Differential Expression Analyses</a></li>
</ul>
</nav>
</div>
<h1 id="introduction">Introduction</h1>
<p>This analysis is based on the guide RNA-seq analysis is easy as 1-2-3 with limma, Glimma and edgeR (Law et al.2018), see <a href="https://pmc.ncbi.nlm.nih.gov/articles/PMC4937821/">https://pmc.ncbi.nlm.nih.gov/articles/PMC4937821/</a>.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<h1 id="setting-up-the-data">Setting up the data</h1>
<h2 id="reading-in-count-data">Reading in count data</h2>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># Read in data</span></span>
<span><span class='va'>counts</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/utils/read.table.html'>read.csv</a></span><span class='op'>(</span><span class='fu'>here</span><span class='op'>(</span><span class='st'>"data/old_counts/filtered_output_3_combined_counts.csv"</span><span class='op'>)</span>, header<span class='op'>=</span><span class='cn'>TRUE</span><span class='op'>)</span> </span>
<span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>counts</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='va'>counts</span><span class='op'>$</span><span class='va'>gene</span> </span>
<span><span class='va'>counts</span> <span class='op'>&lt;-</span> <span class='va'>counts</span><span class='op'>[</span>, <span class='op'>-</span><span class='fl'>1</span><span class='op'>]</span>   </span>
<span></span>
<span><span class='co'># Remove everything before the full stop</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>counts</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/grep.html'>gsub</a></span><span class='op'>(</span><span class='st'>"filtered_output2_|260_|985_|Tube._"</span>, <span class='st'>""</span>, <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>counts</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span></span>
<span><span class='va'>group</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>counts</span><span class='op'>)</span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>dge</span> <span class='op'>&lt;-</span> <span class='fu'>DGEList</span><span class='op'>(</span><span class='va'>counts</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/class.html'>class</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>)</span></span></code></pre>
</div>
<pre><code>[1] &quot;DGEList&quot;
attr(,&quot;package&quot;)
[1] &quot;edgeR&quot;</code></pre>
</div>
<h2 id="organising-sample-information">Organising sample information</h2>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># Better to get the batch from the colnames also</span></span>
<span><span class='va'>batch</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/grep.html'>gsub</a></span><span class='op'>(</span><span class='st'>".+_"</span>, <span class='st'>""</span>, <span class='va'>group</span><span class='op'>)</span></span>
<span><span class='va'>batch</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/grep.html'>gsub</a></span><span class='op'>(</span><span class='st'>"B"</span>, <span class='st'>"Batch"</span>, <span class='va'>batch</span><span class='op'>)</span><span class='op'>)</span></span>
<span></span>
<span><span class='co'># Replace uppercase with lowercase</span></span>
<span><span class='va'>group</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span><span class='va'>group</span><span class='op'>)</span></span>
<span><span class='co'># Treatment labels SOCS3 or WT</span></span>
<span><span class='va'>treatment</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/grep.html'>gsub</a></span><span class='op'>(</span><span class='st'>"_.+"</span>, <span class='st'>""</span>, <span class='va'>group</span><span class='op'>)</span><span class='op'>)</span></span>
<span></span>
<span><span class='co'># Put sample information info DGEList object</span></span>
<span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span> <span class='op'>&lt;-</span> <span class='va'>group</span></span>
<span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>batch</span> <span class='op'>&lt;-</span> <span class='va'>batch</span></span>
<span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>treatment</span> <span class='op'>&lt;-</span> <span class='va'>treatment</span></span>
<span></span>
<span><span class='co'># Check that everything is consistent here</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/table.html'>table</a></span><span class='op'>(</span><span class='va'>group</span>, <span class='va'>treatment</span><span class='op'>)</span></span></code></pre>
</div>
<pre><code>            treatment
group        SOCS3 WT
  SOCS3_1_B2     1  0
  SOCS3_2_B1     1  0
  SOCS3_3_B1     1  0
  WT_1_B2        0  1
  WT_2_B1        0  1
  WT_3_B1        0  1</code></pre>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='fu'><a href='https://rdrr.io/r/base/table.html'>table</a></span><span class='op'>(</span><span class='va'>group</span>, <span class='va'>batch</span><span class='op'>)</span></span></code></pre>
</div>
<pre><code>            batch
group        Batch1 Batch2
  SOCS3_1_B2      0      1
  SOCS3_2_B1      1      0
  SOCS3_3_B1      1      0
  WT_1_B2         0      1
  WT_2_B1         1      0
  WT_3_B1         1      0</code></pre>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span></span></code></pre>
</div>
<pre><code>                group lib.size norm.factors  batch treatment
WT_1_B2       WT_1_B2 10384129            1 Batch2        WT
SOCS3_1_B2 SOCS3_1_B2 11483449            1 Batch2     SOCS3
SOCS3_2_B1 SOCS3_2_B1 12461036            1 Batch1     SOCS3
SOCS3_3_B1 SOCS3_3_B1 10799372            1 Batch1     SOCS3
WT_2_B1       WT_2_B1  9901505            1 Batch1        WT
WT_3_B1       WT_3_B1 10645117            1 Batch1        WT</code></pre>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># Remove group, treatment and batch. These should only be accessed from dge!!!</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/rm.html'>rm</a></span><span class='op'>(</span><span class='va'>group</span>, <span class='va'>batch</span>, <span class='va'>treatment</span><span class='op'>)</span></span></code></pre>
</div>
</div>
<h2 id="organising-gene-annotations">Organising gene annotations</h2>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>geneid</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>)</span></span>
<span></span>
<span><span class='co'># We have Ensembl ID's</span></span>
<span><span class='va'>gene_map</span> <span class='op'>&lt;-</span> <span class='fu'>AnnotationDbi</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html'>select</a></span><span class='op'>(</span><span class='va'>Mus.musculus</span>, keys<span class='op'>=</span><span class='va'>geneid</span>, columns<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"SYMBOL"</span>, <span class='st'>"TXCHROM"</span><span class='op'>)</span>, </span>
<span>                keytype<span class='op'>=</span><span class='st'>"ENSEMBL"</span><span class='op'>)</span></span>
<span></span>
<span><span class='co'># Investigate duplicate gene IDs</span></span>
<span></span>
<span><span class='co'># There are 63 duplicate IDs, for these use their Ensembl ID ... or just remove them?</span></span>
<span><span class='co'># Code to remove them:</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/table.html'>table</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/duplicated.html'>duplicated</a></span><span class='op'>(</span><span class='va'>gene_map</span><span class='op'>$</span><span class='va'>ENSEMBL</span><span class='op'>)</span><span class='op'>)</span></span></code></pre>
</div>
<pre><code>
FALSE  TRUE 
21700   299 </code></pre>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>gene_map</span><span class='op'>$</span><span class='va'>SYMBOL</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/duplicated.html'>duplicated</a></span><span class='op'>(</span><span class='va'>gene_map</span><span class='op'>$</span><span class='va'>ENSEMBL</span><span class='op'>)</span><span class='op'>]</span> <span class='op'>&lt;-</span> <span class='va'>gene_map</span><span class='op'>$</span><span class='va'>ENSEMBL</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/duplicated.html'>duplicated</a></span><span class='op'>(</span><span class='va'>gene_map</span><span class='op'>$</span><span class='va'>ENSEMBL</span><span class='op'>)</span><span class='op'>]</span></span>
<span></span>
<span><span class='co'># For NA symbols, use the ensembl gene id</span></span>
<span><span class='va'>gene_map</span><span class='op'>$</span><span class='va'>SYMBOL</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/ifelse.html'>ifelse</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/NA.html'>is.na</a></span><span class='op'>(</span><span class='va'>gene_map</span><span class='op'>$</span><span class='va'>SYMBOL</span><span class='op'>)</span>, <span class='va'>gene_map</span><span class='op'>$</span><span class='va'>ENSEMBL</span>, <span class='va'>gene_map</span><span class='op'>$</span><span class='va'>SYMBOL</span><span class='op'>)</span> </span>
<span><span class='va'>dge</span><span class='op'>$</span><span class='va'>genes</span> <span class='op'>&lt;-</span> <span class='va'>gene_map</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/match.html'>match</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>)</span>, <span class='va'>gene_map</span><span class='op'>$</span><span class='va'>ENSEMBL</span><span class='op'>)</span>, <span class='op'>]</span></span>
<span></span>
<span><span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>genes</span><span class='op'>)</span></span></code></pre>
</div>
<pre><code>             ENSEMBL SYMBOL TXCHROM
1 ENSMUSG00000000001  Gnai3    chr3
2 ENSMUSG00000000003   Pbsn    chrX
3 ENSMUSG00000000028  Cdc45   chr16
4 ENSMUSG00000000037  Scml2    chrX
5 ENSMUSG00000000049   Apoh   chr11
6 ENSMUSG00000000056   Narf   chr11</code></pre>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># Check that the gene information matches the ordering of the count matrix</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/stopifnot.html'>stopifnot</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/identical.html'>identical</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>)</span>, <span class='va'>dge</span><span class='op'>$</span><span class='va'>genes</span><span class='op'>$</span><span class='va'>ENSEMBL</span><span class='op'>)</span><span class='op'>)</span></span></code></pre>
</div>
</div>
<h1 id="pre-processing">Pre-processing</h1>
<h2 id="filter-low-quality-genes">Filter low quality genes</h2>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># Transform from raw scale</span></span>
<span><span class='va'>cpm</span> <span class='op'>&lt;-</span> <span class='fu'>edgeR</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/cpm.html'>cpm</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>)</span></span>
<span><span class='va'>lcpm</span> <span class='op'>&lt;-</span> <span class='fu'>edgeR</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/cpm.html'>cpm</a></span><span class='op'>(</span><span class='va'>dge</span>, log <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span></span></code></pre>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># There are no genes that are expressed in no samples</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/table.html'>table</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colSums.html'>rowSums</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>counts</span> <span class='op'>==</span> <span class='fl'>0</span><span class='op'>)</span><span class='op'>)</span></span></code></pre>
</div>
<pre><code>
    0     1     2     3     4     5     6 
12129  1119   896   868  1035  1452  4201 </code></pre>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># The smallest amount of counts for a gene is 1 across the 6 samples</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/Extremes.html'>min</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colSums.html'>rowSums</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>counts</span><span class='op'>)</span><span class='op'>)</span></span></code></pre>
</div>
<pre><code>[1] 0</code></pre>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>keep.exprs</span> <span class='op'>&lt;-</span> <span class='fu'>filterByExpr</span><span class='op'>(</span><span class='va'>dge</span>, group<span class='op'>=</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span><span class='op'>)</span></span>
<span><span class='va'>dge</span> <span class='op'>&lt;-</span> <span class='va'>dge</span><span class='op'>[</span><span class='va'>keep.exprs</span>,, keep.lib.sizes<span class='op'>=</span><span class='cn'>FALSE</span><span class='op'>]</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/dim.html'>dim</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>)</span></span></code></pre>
</div>
<pre><code>[1] 13579     6</code></pre>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># We have removed 140 genes using filterByExpr.</span></span>
<span><span class='co'># This function keeps genes with about 10 read counts or more in a minimum</span></span>
<span><span class='co'># number of samples.</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/sum.html'>sum</a></span><span class='op'>(</span><span class='va'>keep.exprs</span> <span class='op'>==</span> <span class='cn'>FALSE</span><span class='op'>)</span></span></code></pre>
</div>
<pre><code>[1] 8121</code></pre>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># rowSums(dge$counts == 0)</span></span></code></pre>
</div>
</div>
<p>In the raw data, there is a spike of lowly expressed genes.
These are removed using the <code>filterByExpr</code> function.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># Figure 1: Density of the raw post filtered data</span></span>
<span></span>
<span><span class='co'># M is the mean, L is the median</span></span>
<span><span class='va'>L</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/mean.html'>mean</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>lib.size</span><span class='op'>)</span> <span class='op'>*</span> <span class='fl'>1e-6</span></span>
<span><span class='va'>M</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/median.html'>median</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>lib.size</span><span class='op'>)</span> <span class='op'>*</span> <span class='fl'>1e-6</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='va'>L</span>, <span class='va'>M</span><span class='op'>)</span></span></code></pre>
</div>
<pre><code>[1] 10.94030 10.71752</code></pre>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>lcpm.cutoff</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/Log.html'>log2</a></span><span class='op'>(</span><span class='fl'>10</span><span class='op'>/</span><span class='va'>M</span> <span class='op'>+</span> <span class='fl'>2</span><span class='op'>/</span><span class='va'>L</span><span class='op'>)</span></span>
<span><span class='va'>nsamples</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/nrow.html'>ncol</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>)</span></span>
<span><span class='va'>col</span> <span class='op'>&lt;-</span> <span class='fu'>brewer.pal</span><span class='op'>(</span><span class='va'>nsamples</span>, <span class='st'>"Paired"</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/par.html'>par</a></span><span class='op'>(</span>mfrow<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>,<span class='fl'>2</span><span class='op'>)</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/plot.default.html'>plot</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/stats/density.html'>density</a></span><span class='op'>(</span><span class='va'>lcpm</span><span class='op'>[</span>,<span class='fl'>1</span><span class='op'>]</span><span class='op'>)</span>, col<span class='op'>=</span><span class='va'>col</span><span class='op'>[</span><span class='fl'>1</span><span class='op'>]</span>, lwd<span class='op'>=</span><span class='fl'>2</span>, ylim<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>0</span>,<span class='fl'>0.26</span><span class='op'>)</span>, las<span class='op'>=</span><span class='fl'>2</span>, main<span class='op'>=</span><span class='st'>""</span>, xlab<span class='op'>=</span><span class='st'>""</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/title.html'>title</a></span><span class='op'>(</span>main<span class='op'>=</span><span class='st'>"A. Raw data"</span>, xlab<span class='op'>=</span><span class='st'>"Log-cpm"</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/abline.html'>abline</a></span><span class='op'>(</span>v<span class='op'>=</span><span class='va'>lcpm.cutoff</span>, lty<span class='op'>=</span><span class='fl'>3</span><span class='op'>)</span></span>
<span><span class='kw'>for</span> <span class='op'>(</span><span class='va'>i</span> <span class='kw'>in</span> <span class='fl'>2</span><span class='op'>:</span><span class='va'>nsamples</span><span class='op'>)</span><span class='op'>{</span></span>
<span>  <span class='va'>den</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/density.html'>density</a></span><span class='op'>(</span><span class='va'>lcpm</span><span class='op'>[</span>,<span class='va'>i</span><span class='op'>]</span><span class='op'>)</span></span>
<span>  <span class='fu'><a href='https://rdrr.io/r/graphics/lines.html'>lines</a></span><span class='op'>(</span><span class='va'>den</span><span class='op'>$</span><span class='va'>x</span>, <span class='va'>den</span><span class='op'>$</span><span class='va'>y</span>, col<span class='op'>=</span><span class='va'>col</span><span class='op'>[</span><span class='va'>i</span><span class='op'>]</span>, lwd<span class='op'>=</span><span class='fl'>2</span><span class='op'>)</span></span>
<span><span class='op'>}</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/legend.html'>legend</a></span><span class='op'>(</span><span class='st'>"topright"</span>, <span class='fu'><a href='https://rdrr.io/r/base/character.html'>as.character</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span><span class='op'>)</span>, text.col<span class='op'>=</span><span class='va'>col</span>, bty<span class='op'>=</span><span class='st'>"n"</span><span class='op'>)</span></span>
<span><span class='va'>lcpm</span> <span class='op'>&lt;-</span> <span class='fu'>edgeR</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/cpm.html'>cpm</a></span><span class='op'>(</span><span class='va'>dge</span>, log<span class='op'>=</span><span class='cn'>TRUE</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/plot.default.html'>plot</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/stats/density.html'>density</a></span><span class='op'>(</span><span class='va'>lcpm</span><span class='op'>[</span>,<span class='fl'>1</span><span class='op'>]</span><span class='op'>)</span>, col<span class='op'>=</span><span class='va'>col</span><span class='op'>[</span><span class='fl'>1</span><span class='op'>]</span>, lwd<span class='op'>=</span><span class='fl'>2</span>, ylim<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>0</span>,<span class='fl'>0.26</span><span class='op'>)</span>, las<span class='op'>=</span><span class='fl'>2</span>, main<span class='op'>=</span><span class='st'>""</span>, xlab<span class='op'>=</span><span class='st'>""</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/title.html'>title</a></span><span class='op'>(</span>main<span class='op'>=</span><span class='st'>"B. Filtered data"</span>, xlab<span class='op'>=</span><span class='st'>"Log-cpm"</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/abline.html'>abline</a></span><span class='op'>(</span>v<span class='op'>=</span><span class='va'>lcpm.cutoff</span>, lty<span class='op'>=</span><span class='fl'>3</span><span class='op'>)</span></span>
<span><span class='kw'>for</span> <span class='op'>(</span><span class='va'>i</span> <span class='kw'>in</span> <span class='fl'>2</span><span class='op'>:</span><span class='va'>nsamples</span><span class='op'>)</span><span class='op'>{</span></span>
<span>  <span class='va'>den</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/density.html'>density</a></span><span class='op'>(</span><span class='va'>lcpm</span><span class='op'>[</span>,<span class='va'>i</span><span class='op'>]</span><span class='op'>)</span></span>
<span>  <span class='fu'><a href='https://rdrr.io/r/graphics/lines.html'>lines</a></span><span class='op'>(</span><span class='va'>den</span><span class='op'>$</span><span class='va'>x</span>, <span class='va'>den</span><span class='op'>$</span><span class='va'>y</span>, col<span class='op'>=</span><span class='va'>col</span><span class='op'>[</span><span class='va'>i</span><span class='op'>]</span>, lwd<span class='op'>=</span><span class='fl'>2</span><span class='op'>)</span></span>
<span><span class='op'>}</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/legend.html'>legend</a></span><span class='op'>(</span><span class='st'>"topright"</span>, <span class='fu'><a href='https://rdrr.io/r/base/character.html'>as.character</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span><span class='op'>)</span>, text.col<span class='op'>=</span><span class='va'>col</span>, bty<span class='op'>=</span><span class='st'>"n"</span><span class='op'>)</span></span></code></pre>
</div>
<p><img src="Filtered_output_3_analysis_files/figure-html5/unnamed-chunk-7-1.png" width="624" /></p>
</div>
<h2 id="quality-control">Quality control</h2>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># Set colours for this data set</span></span>
<span><span class='va'>group_colours</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/setNames.html'>setNames</a></span><span class='op'>(</span></span>
<span>    <span class='fu'>ggthemes</span><span class='fu'>::</span><span class='fu'><a href='http://jrnold.github.io/ggthemes/reference/tableau_color_pal.html'>tableau_color_pal</a></span><span class='op'>(</span><span class='st'>"Tableau 10"</span><span class='op'>)</span><span class='op'>(</span><span class='fl'>6</span><span class='op'>)</span>, </span>
<span>    <span class='fu'><a href='https://rdrr.io/r/base/levels.html'>levels</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span><span class='op'>)</span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>batch_colours</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/setNames.html'>setNames</a></span><span class='op'>(</span></span>
<span>    <span class='fu'>brewer.pal</span><span class='op'>(</span><span class='fl'>8</span>, <span class='st'>"Dark2"</span><span class='op'>)</span><span class='op'>[</span><span class='fl'>3</span><span class='op'>:</span><span class='fl'>4</span><span class='op'>]</span>, </span>
<span>    <span class='fu'><a href='https://rdrr.io/r/base/levels.html'>levels</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>batch</span><span class='op'>)</span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>treatment_colours</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/setNames.html'>setNames</a></span><span class='op'>(</span></span>
<span>    <span class='fu'>brewer.pal</span><span class='op'>(</span><span class='fl'>8</span>, <span class='st'>"Dark2"</span><span class='op'>)</span><span class='op'>[</span><span class='fl'>1</span><span class='op'>:</span><span class='fl'>2</span><span class='op'>]</span>, </span>
<span>    <span class='fu'><a href='https://rdrr.io/r/base/levels.html'>levels</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>treatment</span><span class='op'>)</span><span class='op'>)</span></span></code></pre>
</div>
</div>
<p>The barcharts below show the library sizes for each sample, coloured by sample, batch and treatment.</p>
<h3 id="barchart-of-library-size">Barchart of library size</h3>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># this factor ordering for group is helpful for QC, as it puts everything from </span></span>
<span><span class='co'># the same batch next to each other. </span></span>
<span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span>,</span>
<span>    levels <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"WT_2_B1"</span>, <span class='st'>"WT_3_B1"</span>, <span class='st'>"SOCS3_2_B1"</span>, <span class='st'>"SOCS3_3_B1"</span>, <span class='st'>"WT_1_B2"</span>, <span class='st'>"SOCS3_1_B2"</span><span class='op'>)</span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>p1</span> <span class='op'>&lt;-</span> <span class='fu'>ggplot</span><span class='op'>(</span>data <span class='op'>=</span> <span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span>, mapping <span class='op'>=</span> <span class='fu'>aes</span><span class='op'>(</span>y <span class='op'>=</span> <span class='va'>group</span>, x <span class='op'>=</span> <span class='va'>lib.size</span>, fill <span class='op'>=</span> <span class='va'>group</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> </span>
<span>    <span class='fu'>geom_col</span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>    <span class='fu'>scale_fill_manual</span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>group_colours</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>    <span class='fu'>scale_y_discrete</span><span class='op'>(</span>limits <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/rev.html'>rev</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/levels.html'>levels</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>    <span class='fu'>theme_minimal</span><span class='op'>(</span><span class='op'>)</span></span>
<span><span class='va'>p2</span> <span class='op'>&lt;-</span> <span class='fu'>ggplot</span><span class='op'>(</span>data <span class='op'>=</span> <span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span>, mapping <span class='op'>=</span> <span class='fu'>aes</span><span class='op'>(</span>y <span class='op'>=</span> <span class='va'>group</span>, x <span class='op'>=</span> <span class='va'>lib.size</span>, fill <span class='op'>=</span> <span class='va'>batch</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> </span>
<span>    <span class='fu'>geom_col</span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>    <span class='fu'>scale_fill_manual</span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>batch_colours</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>    <span class='fu'>scale_y_discrete</span><span class='op'>(</span>limits <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/rev.html'>rev</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/levels.html'>levels</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>    <span class='fu'>theme_minimal</span><span class='op'>(</span><span class='op'>)</span></span>
<span><span class='va'>p3</span> <span class='op'>&lt;-</span> <span class='fu'>ggplot</span><span class='op'>(</span>data <span class='op'>=</span> <span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span>, mapping <span class='op'>=</span> <span class='fu'>aes</span><span class='op'>(</span>y <span class='op'>=</span> <span class='va'>group</span>, x <span class='op'>=</span> <span class='va'>lib.size</span>, fill <span class='op'>=</span> <span class='va'>treatment</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> </span>
<span>    <span class='fu'>geom_col</span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>    <span class='fu'>scale_fill_manual</span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>treatment_colours</span><span class='op'>)</span> <span class='op'>+</span> </span>
<span>    <span class='fu'>scale_y_discrete</span><span class='op'>(</span>limits <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/rev.html'>rev</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/levels.html'>levels</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>    <span class='fu'>theme_minimal</span><span class='op'>(</span><span class='op'>)</span></span>
<span><span class='op'>(</span><span class='va'>p1</span> <span class='op'>/</span> <span class='va'>p2</span> <span class='op'>/</span> <span class='va'>p3</span><span class='op'>)</span> <span class='op'>+</span> <span class='fu'>plot_layout</span><span class='op'>(</span>guides <span class='op'>=</span> <span class='st'>"collect"</span><span class='op'>)</span></span></code></pre>
</div>
<p><img src="Filtered_output_3_analysis_files/figure-html5/unnamed-chunk-9-1.png" width="624" /></p>
</div>
<h3 id="number-of-expressed-features">Number of expressed features</h3>
<p>Expressed features are defined as those with at least one count in a sample.
There is a similar number of expressed features in each sample.
Note however that 13,000 is quite low and the usual amount would be closer to 20,000 genes.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>nFeatures</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/apply.html'>apply</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>counts</span>, MARGIN <span class='op'>=</span> <span class='fl'>2</span>, FUN <span class='op'>=</span> <span class='kw'>function</span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span> <span class='op'>{</span><span class='fu'><a href='https://rdrr.io/r/base/sum.html'>sum</a></span><span class='op'>(</span><span class='va'>x</span> <span class='op'>&gt;</span> <span class='fl'>0</span><span class='op'>)</span><span class='op'>}</span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span> <span class='op'>|&gt;</span></span>
<span>    <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/select.html'>select</a></span><span class='op'>(</span><span class='va'>group</span>, <span class='va'>nFeatures</span><span class='op'>)</span> <span class='op'>|&gt;</span></span>
<span>    <span class='fu'>knitr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/knitr/man/kable.html'>kable</a></span><span class='op'>(</span><span class='op'>)</span></span></code></pre>
</div>
<table>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: left;">group</th>
<th style="text-align: right;">nFeatures</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">WT_1_B2</td>
<td style="text-align: left;">WT_1_B2</td>
<td style="text-align: right;">13358</td>
</tr>
<tr class="even">
<td style="text-align: left;">SOCS3_1_B2</td>
<td style="text-align: left;">SOCS3_1_B2</td>
<td style="text-align: right;">13368</td>
</tr>
<tr class="odd">
<td style="text-align: left;">SOCS3_2_B1</td>
<td style="text-align: left;">SOCS3_2_B1</td>
<td style="text-align: right;">12831</td>
</tr>
<tr class="even">
<td style="text-align: left;">SOCS3_3_B1</td>
<td style="text-align: left;">SOCS3_3_B1</td>
<td style="text-align: right;">13213</td>
</tr>
<tr class="odd">
<td style="text-align: left;">WT_2_B1</td>
<td style="text-align: left;">WT_2_B1</td>
<td style="text-align: right;">13193</td>
</tr>
<tr class="even">
<td style="text-align: left;">WT_3_B1</td>
<td style="text-align: left;">WT_3_B1</td>
<td style="text-align: right;">12746</td>
</tr>
</tbody>
</table>
</div>
<h3 id="proportion-of-mitochondrial-genes">Proportion of mitochondrial genes</h3>
<p>There are no mitochondrial genes in this count matrix.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='fu'><a href='https://rdrr.io/r/base/table.html'>table</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>genes</span><span class='op'>$</span><span class='va'>TXCHROM</span><span class='op'>)</span></span></code></pre>
</div>
<pre><code>
                chr1                chr10                chr11 
                 836                  644                 1145 
               chr12                chr13                chr14 
                 462                  436                  420 
               chr15                chr16                chr17 
                 559                  399                  663 
               chr18                chr19                 chr2 
                 343                  512                 1054 
                chr3                 chr4 chr4_JH584294_random 
                 578                  902                    1 
                chr5                 chr6                 chr7 
                 815                  755                 1043 
                chr8                 chr9                 chrX 
                 632                  758                  333 </code></pre>
</div>
<h2 id="normalisation">Normalisation</h2>
<p>The impact of normalisation is subtle.
Most notably, the third quantiles have been brought into alignment.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>dge_raw</span> <span class='op'>&lt;-</span> <span class='va'>dge</span></span>
<span><span class='va'>dge</span> <span class='op'>&lt;-</span> <span class='fu'>calcNormFactors</span><span class='op'>(</span><span class='va'>dge</span>, method <span class='op'>=</span> <span class='st'>"TMM"</span><span class='op'>)</span></span>
<span></span>
<span><span class='co'># Library sizes</span></span>
<span><span class='va'>dge_raw</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>lib.size</span></span></code></pre>
</div>
<pre><code>[1] 10377317 11475960 12456805 10793544  9896663 10641492</code></pre>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># Norm factors</span></span>
<span><span class='va'>dge_raw</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>norm.factors</span></span></code></pre>
</div>
<pre><code>[1] 1 1 1 1 1 1</code></pre>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>norm.factors</span></span></code></pre>
</div>
<pre><code>[1] 1.1514177 1.1882010 0.8713582 0.9910832 0.9664706 0.8757532</code></pre>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># log-cpm in appropriate order for plot</span></span>
<span><span class='co'># NOTE: NEVER USE WITH dge$samples!!! For plotting only!!</span></span>
<span><span class='va'>lcpm_raw_plot</span> <span class='op'>&lt;-</span> <span class='fu'>edgeR</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/cpm.html'>cpm</a></span><span class='op'>(</span><span class='va'>dge_raw</span>, log<span class='op'>=</span><span class='cn'>TRUE</span><span class='op'>)</span><span class='op'>[</span>, <span class='fu'><a href='https://rdrr.io/r/base/levels.html'>levels</a></span><span class='op'>(</span><span class='va'>dge_raw</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span><span class='op'>)</span><span class='op'>]</span></span>
<span><span class='va'>lcpm_plot</span> <span class='op'>&lt;-</span> <span class='fu'>edgeR</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/cpm.html'>cpm</a></span><span class='op'>(</span><span class='va'>dge</span>, log <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span><span class='op'>[</span>, <span class='fu'><a href='https://rdrr.io/r/base/levels.html'>levels</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span><span class='op'>)</span><span class='op'>]</span></span>
<span></span>
<span><span class='co'># y-limit for plot </span></span>
<span><span class='va'>ylims</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/range.html'>range</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='va'>lcpm_raw_plot</span>, <span class='va'>lcpm_plot</span><span class='op'>)</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/par.html'>par</a></span><span class='op'>(</span>mfrow<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>,<span class='fl'>2</span><span class='op'>)</span><span class='op'>)</span></span>
<span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/boxplot.html'>boxplot</a></span><span class='op'>(</span><span class='va'>lcpm_raw_plot</span>, las<span class='op'>=</span><span class='fl'>2</span>, col<span class='op'>=</span><span class='va'>col</span>, main<span class='op'>=</span><span class='st'>""</span>, <span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/title.html'>title</a></span><span class='op'>(</span>main<span class='op'>=</span><span class='st'>"A. Example: Unnormalised data"</span>, ylab<span class='op'>=</span><span class='st'>"Log-cpm"</span><span class='op'>)</span></span>
<span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/boxplot.html'>boxplot</a></span><span class='op'>(</span><span class='va'>lcpm_plot</span>, las<span class='op'>=</span><span class='fl'>2</span>, col<span class='op'>=</span><span class='va'>col</span>, main<span class='op'>=</span><span class='st'>""</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/title.html'>title</a></span><span class='op'>(</span>main<span class='op'>=</span><span class='st'>"B. Example: Normalised data"</span>, ylab<span class='op'>=</span><span class='st'>"Log-cpm"</span><span class='op'>)</span></span></code></pre>
</div>
<p><img src="Filtered_output_3_analysis_files/figure-html5/unnamed-chunk-13-1.png" width="624" /></p>
</div>
<p>The plots below show the MDS plot before and after normalisation.
This has a more obvious difference than the boxplot above.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='fu'><a href='https://rdrr.io/r/graphics/par.html'>par</a></span><span class='op'>(</span>mfrow <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>,<span class='fl'>2</span><span class='op'>)</span><span class='op'>)</span></span>
<span><span class='fu'>plotMDS</span><span class='op'>(</span><span class='va'>dge_raw</span>, main<span class='op'>=</span><span class='st'>"Before normalization"</span>, col<span class='op'>=</span><span class='va'>batch_colours</span><span class='op'>[</span><span class='va'>dge_raw</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>batch</span><span class='op'>]</span><span class='op'>)</span></span>
<span><span class='fu'>plotMDS</span><span class='op'>(</span><span class='va'>dge</span>, main<span class='op'>=</span><span class='st'>"After normalization"</span>, col<span class='op'>=</span><span class='va'>batch_colours</span><span class='op'>[</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>batch</span><span class='op'>]</span><span class='op'>)</span></span></code></pre>
</div>
<p><img src="Filtered_output_3_analysis_files/figure-html5/unnamed-chunk-14-1.png" width="624" /></p>
</div>
<p>After normalising the data, we need to evaluate the log counts again.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>lcpm</span> <span class='op'>&lt;-</span> <span class='fu'>edgeR</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/cpm.html'>cpm</a></span><span class='op'>(</span><span class='va'>dge</span>, log<span class='op'>=</span><span class='cn'>TRUE</span><span class='op'>)</span></span></code></pre>
</div>
</div>
<h1 id="visualising-the-data">Visualising the data</h1>
<p>A useful tool to visualise RNA-seq data is the multi-dimensional scaling (MDS) plot.
This plot can show us which samples are more similar to each other, and can help to identify outlying or low quality samples.
The first dimension captures the most amount of information about the data, while the second dimension captures the next biggest proportion of variation.</p>
<p>Things to look out for in this plot are:
- Samples should cluster within the primary condition of interest, i.e., treatment.
- Technical replicates should lie very close to each other.
- The first dimension should explain treatment, and the second should explain batch.</p>
<p>The MDS plot below shows that the first dimension can be explained by batch (32% variation).
The second dimension shows some separation by treatment and library size.</p>
<p>As the technical replicates have not clustered together, we cannot get DE for this data.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># NOTE: Even though it might seem like more typing, it is much better to use</span></span>
<span><span class='co'>#   dge$samples$group than group. </span></span>
<span><span class='co'>#   It is very easy to have inconsistencies with labelling if not using the dge</span></span>
<span><span class='co'>#   object every time. </span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/par.html'>par</a></span><span class='op'>(</span>mfrow <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>2</span>,<span class='fl'>2</span><span class='op'>)</span><span class='op'>)</span></span>
<span><span class='fu'>plotMDS</span><span class='op'>(</span><span class='va'>lcpm</span>, labels<span class='op'>=</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span>, col<span class='op'>=</span><span class='va'>group_colours</span><span class='op'>[</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span><span class='op'>]</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/title.html'>title</a></span><span class='op'>(</span>main<span class='op'>=</span><span class='st'>"A. Sample groups"</span><span class='op'>)</span></span>
<span></span>
<span><span class='fu'>plotMDS</span><span class='op'>(</span><span class='va'>lcpm</span>, labels<span class='op'>=</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>treatment</span>, col<span class='op'>=</span><span class='va'>treatment_colours</span><span class='op'>[</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>treatment</span><span class='op'>]</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/title.html'>title</a></span><span class='op'>(</span>main<span class='op'>=</span><span class='st'>"B. Treatment"</span><span class='op'>)</span></span>
<span></span>
<span><span class='fu'>plotMDS</span><span class='op'>(</span><span class='va'>lcpm</span>, labels<span class='op'>=</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>batch</span>, col<span class='op'>=</span><span class='va'>batch_colours</span><span class='op'>[</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>batch</span><span class='op'>]</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/title.html'>title</a></span><span class='op'>(</span>main<span class='op'>=</span><span class='st'>"C. Batch"</span><span class='op'>)</span></span>
<span></span>
<span><span class='co'># MDS plot with library size</span></span>
<span><span class='va'>libsizes</span> <span class='op'>&lt;-</span> <span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>lib.size</span></span>
<span></span>
<span><span class='co'># Create a viridis color function</span></span>
<span><span class='va'>color_func</span> <span class='op'>&lt;-</span> <span class='fu'>scales</span><span class='fu'>::</span><span class='fu'><a href='https://scales.r-lib.org/reference/col_numeric.html'>col_numeric</a></span><span class='op'>(</span><span class='fu'>viridis</span><span class='fu'>::</span><span class='fu'><a href='https://sjmgarnier.github.io/viridisLite/reference/viridis.html'>viridis</a></span><span class='op'>(</span><span class='fl'>100</span><span class='op'>)</span>, domain <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/range.html'>range</a></span><span class='op'>(</span><span class='va'>libsizes</span><span class='op'>)</span><span class='op'>)</span></span>
<span></span>
<span><span class='co'># Assign colors according to library sizes</span></span>
<span><span class='va'>point_colors</span> <span class='op'>&lt;-</span> <span class='fu'>color_func</span><span class='op'>(</span><span class='va'>libsizes</span><span class='op'>)</span></span>
<span></span>
<span><span class='fu'>plotMDS</span><span class='op'>(</span><span class='va'>lcpm</span>, labels<span class='op'>=</span><span class='cn'>NULL</span>, pch <span class='op'>=</span> <span class='fl'>16</span>, cex <span class='op'>=</span> <span class='fl'>4</span>, col<span class='op'>=</span><span class='va'>point_colors</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/title.html'>title</a></span><span class='op'>(</span>main<span class='op'>=</span><span class='st'>"D. Library Size"</span><span class='op'>)</span> </span></code></pre>
</div>
<p><img src="Filtered_output_3_analysis_files/figure-html5/unnamed-chunk-16-1.png" width="624" /></p>
</div>
<p>Plots showing genes of interest on MDS plot.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='fu'><a href='https://rdrr.io/r/graphics/par.html'>par</a></span><span class='op'>(</span>mfrow <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>2</span>,<span class='fl'>2</span><span class='op'>)</span><span class='op'>)</span></span>
<span></span>
<span><span class='co'># Plot 3: Treatments </span></span>
<span><span class='fu'>plotMDS</span><span class='op'>(</span><span class='va'>lcpm</span>, labels<span class='op'>=</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>treatment</span>, col<span class='op'>=</span><span class='va'>treatment_colours</span><span class='op'>[</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>treatment</span><span class='op'>]</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/title.html'>title</a></span><span class='op'>(</span>main<span class='op'>=</span><span class='st'>"C. Treatment"</span><span class='op'>)</span> </span>
<span></span>
<span></span>
<span><span class='va'>genes_of_interest</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Socs3"</span>, <span class='st'>"Socs1"</span>, <span class='st'>"Bcl3"</span><span class='op'>)</span></span>
<span></span>
<span><span class='co'># Plot 5: Socs gene. </span></span>
<span><span class='va'>gene_name</span> <span class='op'>&lt;-</span> <span class='va'>genes_of_interest</span><span class='op'>[</span><span class='fl'>1</span><span class='op'>]</span></span>
<span><span class='va'>gene_expression</span> <span class='op'>&lt;-</span> <span class='va'>lcpm</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/grep.html'>grep</a></span><span class='op'>(</span><span class='va'>gene_name</span>, <span class='va'>dge</span><span class='op'>$</span><span class='va'>genes</span><span class='op'>$</span><span class='va'>SYMBOL</span><span class='op'>)</span>, <span class='op'>]</span></span>
<span></span>
<span><span class='co'># Create a viridis color function</span></span>
<span><span class='va'>color_func</span> <span class='op'>&lt;-</span> <span class='fu'>scales</span><span class='fu'>::</span><span class='fu'><a href='https://scales.r-lib.org/reference/col_numeric.html'>col_numeric</a></span><span class='op'>(</span><span class='fu'>viridis</span><span class='fu'>::</span><span class='fu'><a href='https://sjmgarnier.github.io/viridisLite/reference/viridis.html'>viridis</a></span><span class='op'>(</span><span class='fl'>100</span><span class='op'>)</span>, domain <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/range.html'>range</a></span><span class='op'>(</span><span class='va'>gene_expression</span><span class='op'>)</span><span class='op'>)</span></span>
<span></span>
<span><span class='co'># Assign colors according to library sizes</span></span>
<span><span class='va'>point_colors</span> <span class='op'>&lt;-</span> <span class='fu'>color_func</span><span class='op'>(</span><span class='va'>gene_expression</span><span class='op'>)</span></span>
<span></span>
<span><span class='fu'>plotMDS</span><span class='op'>(</span><span class='va'>lcpm</span>, labels<span class='op'>=</span><span class='cn'>NULL</span>, pch <span class='op'>=</span> <span class='fl'>16</span>, cex <span class='op'>=</span> <span class='fl'>4</span>, col<span class='op'>=</span><span class='va'>point_colors</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/title.html'>title</a></span><span class='op'>(</span>main<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='va'>gene_name</span>, <span class='st'>" Expression (log CPM)"</span><span class='op'>)</span><span class='op'>)</span> </span>
<span></span>
<span></span>
<span><span class='va'>gene_name</span> <span class='op'>&lt;-</span> <span class='va'>genes_of_interest</span><span class='op'>[</span><span class='fl'>2</span><span class='op'>]</span></span>
<span><span class='va'>gene_expression</span> <span class='op'>&lt;-</span> <span class='va'>lcpm</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/grep.html'>grep</a></span><span class='op'>(</span><span class='va'>gene_name</span>, <span class='va'>dge</span><span class='op'>$</span><span class='va'>genes</span><span class='op'>$</span><span class='va'>SYMBOL</span><span class='op'>)</span>, <span class='op'>]</span></span>
<span></span>
<span><span class='co'># Create a viridis color function</span></span>
<span><span class='va'>color_func</span> <span class='op'>&lt;-</span> <span class='fu'>scales</span><span class='fu'>::</span><span class='fu'><a href='https://scales.r-lib.org/reference/col_numeric.html'>col_numeric</a></span><span class='op'>(</span><span class='fu'>viridis</span><span class='fu'>::</span><span class='fu'><a href='https://sjmgarnier.github.io/viridisLite/reference/viridis.html'>viridis</a></span><span class='op'>(</span><span class='fl'>100</span><span class='op'>)</span>, domain <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/range.html'>range</a></span><span class='op'>(</span><span class='va'>gene_expression</span><span class='op'>)</span><span class='op'>)</span></span>
<span></span>
<span><span class='co'># Assign colors according to library sizes</span></span>
<span><span class='va'>point_colors</span> <span class='op'>&lt;-</span> <span class='fu'>color_func</span><span class='op'>(</span><span class='va'>gene_expression</span><span class='op'>)</span></span>
<span></span>
<span><span class='fu'>plotMDS</span><span class='op'>(</span><span class='va'>lcpm</span>, labels<span class='op'>=</span><span class='cn'>NULL</span>, pch <span class='op'>=</span> <span class='fl'>16</span>, cex <span class='op'>=</span> <span class='fl'>4</span>, col<span class='op'>=</span><span class='va'>point_colors</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/title.html'>title</a></span><span class='op'>(</span>main<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='va'>gene_name</span>, <span class='st'>" Expression (log CPM)"</span><span class='op'>)</span><span class='op'>)</span> </span>
<span></span>
<span></span>
<span><span class='va'>gene_name</span> <span class='op'>&lt;-</span> <span class='va'>genes_of_interest</span><span class='op'>[</span><span class='fl'>3</span><span class='op'>]</span></span>
<span><span class='va'>gene_expression</span> <span class='op'>&lt;-</span> <span class='va'>lcpm</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/grep.html'>grep</a></span><span class='op'>(</span><span class='va'>gene_name</span>, <span class='va'>dge</span><span class='op'>$</span><span class='va'>genes</span><span class='op'>$</span><span class='va'>SYMBOL</span><span class='op'>)</span>, <span class='op'>]</span></span>
<span></span>
<span><span class='co'># Create a viridis color function</span></span>
<span><span class='va'>color_func</span> <span class='op'>&lt;-</span> <span class='fu'>scales</span><span class='fu'>::</span><span class='fu'><a href='https://scales.r-lib.org/reference/col_numeric.html'>col_numeric</a></span><span class='op'>(</span><span class='fu'>viridis</span><span class='fu'>::</span><span class='fu'><a href='https://sjmgarnier.github.io/viridisLite/reference/viridis.html'>viridis</a></span><span class='op'>(</span><span class='fl'>100</span><span class='op'>)</span>, domain <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/range.html'>range</a></span><span class='op'>(</span><span class='va'>gene_expression</span><span class='op'>)</span><span class='op'>)</span></span>
<span></span>
<span><span class='co'># Assign colors according to library sizes</span></span>
<span><span class='va'>point_colors</span> <span class='op'>&lt;-</span> <span class='fu'>color_func</span><span class='op'>(</span><span class='va'>gene_expression</span><span class='op'>)</span></span>
<span></span>
<span><span class='fu'>plotMDS</span><span class='op'>(</span><span class='va'>lcpm</span>, labels<span class='op'>=</span><span class='cn'>NULL</span>, pch <span class='op'>=</span> <span class='fl'>16</span>, cex <span class='op'>=</span> <span class='fl'>4</span>, col<span class='op'>=</span><span class='va'>point_colors</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/title.html'>title</a></span><span class='op'>(</span>main<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='va'>gene_name</span>, <span class='st'>" Expression (log CPM)"</span><span class='op'>)</span><span class='op'>)</span> </span></code></pre>
</div>
<p><img src="Filtered_output_3_analysis_files/figure-html5/unnamed-chunk-17-1.png" width="624" /></p>
</div>
<h2 id="comparing-batches-for-genes-of-interest">Comparing batches for genes of interest</h2>
<p>Based on a priori knowledge, the SOCS3 knockout samples should express Socs1 and Socs3 less than the WT.
Bcl3 should be higher in the SOCS3 group.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>genes_of_interest</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Socs3"</span>, <span class='st'>"Socs1"</span>, <span class='st'>"Bcl3"</span><span class='op'>)</span></span>
<span><span class='va'>selected_genes</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/match.html'>match</a></span><span class='op'>(</span><span class='va'>genes_of_interest</span>, <span class='va'>dge</span><span class='op'>$</span><span class='va'>genes</span><span class='op'>$</span><span class='va'>SYMBOL</span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>df</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span></span>
<span>    y <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/t.html'>t</a></span><span class='op'>(</span><span class='va'>lcpm</span><span class='op'>[</span><span class='va'>selected_genes</span>, <span class='op'>]</span><span class='op'>)</span>, </span>
<span>    treatment <span class='op'>=</span> <span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>treatment</span>, </span>
<span>    batch <span class='op'>=</span> <span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>batch</span><span class='op'>)</span></span>
<span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>df</span><span class='op'>)</span><span class='op'>[</span><span class='fl'>1</span><span class='op'>:</span><span class='fl'>3</span><span class='op'>]</span> <span class='op'>&lt;-</span> <span class='va'>genes_of_interest</span></span>
<span></span>
<span><span class='va'>df</span><span class='op'>$</span><span class='va'>labels</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='va'>df</span><span class='op'>$</span><span class='va'>batch</span>, <span class='st'>"."</span>, <span class='va'>df</span><span class='op'>$</span><span class='va'>treatment</span><span class='op'>)</span>, </span>
<span>    levels <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Batch1.WT"</span>, <span class='st'>"Batch1.SOCS3"</span>, <span class='st'>"Batch2.WT"</span>, <span class='st'>"Batch2.SOCS3"</span><span class='op'>)</span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>treatment_colours</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/setNames.html'>setNames</a></span><span class='op'>(</span></span>
<span>    <span class='fu'>RColorBrewer</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html'>brewer.pal</a></span><span class='op'>(</span><span class='fl'>3</span>, <span class='st'>"Dark2"</span><span class='op'>)</span><span class='op'>[</span><span class='fl'>1</span><span class='op'>:</span><span class='fl'>2</span><span class='op'>]</span>, </span>
<span>    <span class='fu'><a href='https://rdrr.io/r/base/levels.html'>levels</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>treatment</span><span class='op'>)</span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>plot_list</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span><span class='op'>)</span></span>
<span><span class='kw'>for</span> <span class='op'>(</span><span class='va'>i</span> <span class='kw'>in</span> <span class='fl'>1</span><span class='op'>:</span><span class='fl'>3</span><span class='op'>)</span> <span class='op'>{</span></span>
<span>    <span class='va'>plot_list</span><span class='op'>[[</span><span class='va'>i</span><span class='op'>]</span><span class='op'>]</span> <span class='op'>&lt;-</span></span>
<span>        <span class='fu'>ggplot</span><span class='op'>(</span>data <span class='op'>=</span> <span class='va'>df</span>, mapping <span class='op'>=</span> <span class='fu'>aes</span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>labels</span>, y <span class='op'>=</span> <span class='va'>.data</span><span class='op'>[[</span><span class='va'>genes_of_interest</span><span class='op'>[</span><span class='va'>i</span><span class='op'>]</span><span class='op'>]</span><span class='op'>]</span>, group <span class='op'>=</span> <span class='va'>labels</span>, colour <span class='op'>=</span> <span class='va'>treatment</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> </span>
<span>            <span class='fu'>geom_point</span><span class='op'>(</span>size <span class='op'>=</span> <span class='fl'>4</span><span class='op'>)</span> <span class='op'>+</span> </span>
<span>            <span class='fu'>theme_minimal</span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span> </span>
<span>            <span class='fu'>theme</span><span class='op'>(</span>axis.text.x <span class='op'>=</span> <span class='fu'>element_text</span><span class='op'>(</span>angle <span class='op'>=</span> <span class='fl'>45</span>, hjust <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> </span>
<span>            <span class='fu'>ylab</span><span class='op'>(</span><span class='st'>"Expression (log CPM)"</span><span class='op'>)</span> <span class='op'>+</span> </span>
<span>            <span class='fu'>scale_colour_manual</span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>treatment_colours</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>            <span class='fu'>ggtitle</span><span class='op'>(</span>label <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='va'>genes_of_interest</span><span class='op'>[</span><span class='va'>i</span><span class='op'>]</span><span class='op'>)</span></span>
<span>    <span class='op'>)</span></span>
<span><span class='op'>}</span></span>
<span><span class='fu'>patchwork</span><span class='fu'>::</span><span class='fu'><a href='https://patchwork.data-imaginist.com/reference/wrap_plots.html'>wrap_plots</a></span><span class='op'>(</span><span class='va'>plot_list</span>, ncol <span class='op'>=</span> <span class='fl'>3</span>, guides <span class='op'>=</span> <span class='st'>"collect"</span><span class='op'>)</span> </span></code></pre>
</div>
<p><img src="Filtered_output_3_analysis_files/figure-html5/unnamed-chunk-18-1.png" width="624" /></p>
</div>
<h1 id="differential-expression-analyses">Differential Expression Analyses</h1>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>design</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/model.matrix.html'>model.matrix</a></span><span class='op'>(</span><span class='op'>~</span><span class='fl'>0</span><span class='op'>+</span><span class='va'>treatment</span><span class='op'>+</span><span class='va'>batch</span>, data <span class='op'>=</span> <span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>)</span></span>
<span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>design</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/grep.html'>gsub</a></span><span class='op'>(</span><span class='st'>"treatment"</span>, <span class='st'>""</span>, <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>design</span><span class='op'>)</span><span class='op'>)</span></span>
<span><span class='va'>design</span></span></code></pre>
</div>
<pre><code>           SOCS3 WT batchBatch2
WT_1_B2        0  1           1
SOCS3_1_B2     1  0           1
SOCS3_2_B1     1  0           0
SOCS3_3_B1     1  0           0
WT_2_B1        0  1           0
WT_3_B1        0  1           0
attr(,&quot;assign&quot;)
[1] 1 1 2
attr(,&quot;contrasts&quot;)
attr(,&quot;contrasts&quot;)$treatment
[1] &quot;contr.treatment&quot;

attr(,&quot;contrasts&quot;)$batch
[1] &quot;contr.treatment&quot;</code></pre>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>contr.matrix</span> <span class='op'>&lt;-</span> <span class='fu'>makeContrasts</span><span class='op'>(</span></span>
<span>   SOCS3vsWT <span class='op'>=</span> <span class='va'>SOCS3</span> <span class='op'>-</span> <span class='va'>WT</span>,</span>
<span>   levels <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>design</span><span class='op'>)</span><span class='op'>)</span></span>
<span><span class='va'>contr.matrix</span></span></code></pre>
</div>
<pre><code>             Contrasts
Levels        SOCS3vsWT
  SOCS3               1
  WT                 -1
  batchBatch2         0</code></pre>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>v</span> <span class='op'>&lt;-</span> <span class='fu'>voom</span><span class='op'>(</span><span class='va'>dge</span>, <span class='va'>design</span>, plot<span class='op'>=</span><span class='cn'>TRUE</span><span class='op'>)</span></span></code></pre>
</div>
<img src="Filtered_output_3_analysis_files/figure-html5/unnamed-chunk-19-1.png" width="624" />
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>v</span></span></code></pre>
</div>
<pre><code>An object of class &quot;EList&quot;
$genes
             ENSEMBL SYMBOL TXCHROM
1 ENSMUSG00000000001  Gnai3    chr3
3 ENSMUSG00000000028  Cdc45   chr16
4 ENSMUSG00000000037  Scml2    chrX
5 ENSMUSG00000000049   Apoh   chr11
7 ENSMUSG00000000058   Cav2    chr6
13574 more rows ...

$targets
                group lib.size norm.factors  batch treatment
WT_1_B2       WT_1_B2 11948626    1.1514177 Batch2        WT
SOCS3_1_B2 SOCS3_1_B2 13635747    1.1882010 Batch2     SOCS3
SOCS3_2_B1 SOCS3_2_B1 10854339    0.8713582 Batch1     SOCS3
SOCS3_3_B1 SOCS3_3_B1 10697300    0.9910832 Batch1     SOCS3
WT_2_B1       WT_2_B1  9564834    0.9664706 Batch1        WT
WT_3_B1       WT_3_B1  9319320    0.8757532 Batch1        WT
           nFeatures
WT_1_B2        13358
SOCS3_1_B2     13368
SOCS3_2_B1     12831
SOCS3_3_B1     13213
WT_2_B1        13193
WT_3_B1        12746

$E
                    WT_1_B2 SOCS3_1_B2 SOCS3_2_B1 SOCS3_3_B1  WT_2_B1
ENSMUSG00000000001 7.821840   7.810701   8.991167   8.710913 8.808517
ENSMUSG00000000028 5.965225   5.555984   6.390315   6.099478 6.012555
ENSMUSG00000000037 7.383762   7.030364   7.652887   7.273876 7.316380
ENSMUSG00000000049 4.954557   4.646420   2.274045   3.597633 4.121638
ENSMUSG00000000058 5.682147   6.002993   7.270176   7.341961 7.017802
                    WT_3_B1
ENSMUSG00000000001 9.276754
ENSMUSG00000000028 6.503436
ENSMUSG00000000037 7.577031
ENSMUSG00000000049 3.606324
ENSMUSG00000000058 6.877149
13574 more rows ...

$weights
         [,1]     [,2]      [,3]      [,4]      [,5]      [,6]
[1,] 27.30716 27.39838 28.208208 28.194872 28.174692 28.149474
[2,] 19.62372 19.90988 20.906865 20.777044 20.674267 20.439958
[3,] 25.99418 26.20765 26.001644 25.944118 25.795056 25.693540
[4,] 15.05207 11.58310  4.560348  4.509299  6.094405  5.970889
[5,] 18.56868 21.85032 25.724248 25.667488 23.894672 23.743257
13574 more rows ...

$design
           SOCS3 WT batchBatch2
WT_1_B2        0  1           1
SOCS3_1_B2     1  0           1
SOCS3_2_B1     1  0           0
SOCS3_3_B1     1  0           0
WT_2_B1        0  1           0
WT_3_B1        0  1           0
attr(,&quot;assign&quot;)
[1] 1 1 2
attr(,&quot;contrasts&quot;)
attr(,&quot;contrasts&quot;)$treatment
[1] &quot;contr.treatment&quot;

attr(,&quot;contrasts&quot;)$batch
[1] &quot;contr.treatment&quot;</code></pre>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>vfit</span> <span class='op'>&lt;-</span> <span class='fu'>lmFit</span><span class='op'>(</span><span class='va'>v</span>, <span class='va'>design</span><span class='op'>)</span></span>
<span><span class='va'>vfit</span> <span class='op'>&lt;-</span> <span class='fu'>contrasts.fit</span><span class='op'>(</span><span class='va'>vfit</span>, contrasts<span class='op'>=</span><span class='va'>contr.matrix</span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>efit</span> <span class='op'>&lt;-</span> <span class='fu'>eBayes</span><span class='op'>(</span><span class='va'>vfit</span><span class='op'>)</span></span>
<span><span class='fu'>plotSA</span><span class='op'>(</span><span class='va'>efit</span><span class='op'>)</span></span></code></pre>
</div>
<img src="Filtered_output_3_analysis_files/figure-html5/unnamed-chunk-19-2.png" width="624" />
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>de_results</span> <span class='op'>&lt;-</span> <span class='fu'>decideTests</span><span class='op'>(</span><span class='va'>efit</span>, adjust.method <span class='op'>=</span> <span class='st'>"fdr"</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/summary.html'>summary</a></span><span class='op'>(</span><span class='va'>de_results</span><span class='op'>)</span></span></code></pre>
</div>
<pre><code>       SOCS3vsWT
Down           0
NotSig     13579
Up             0</code></pre>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='fu'>topTable</span><span class='op'>(</span><span class='va'>efit</span>, n <span class='op'>=</span> <span class='fl'>20</span><span class='op'>)</span></span></code></pre>
</div>
<pre><code>                              ENSEMBL        SYMBOL TXCHROM
ENSMUSG00000028238 ENSMUSG00000028238      Atp6v0d2    chr4
ENSMUSG00000019935 ENSMUSG00000019935       Slc17a8   chr10
ENSMUSG00000030433 ENSMUSG00000030433          Sbk2    chr7
ENSMUSG00000028211 ENSMUSG00000028211     Trp53inp1    chr4
ENSMUSG00000053508 ENSMUSG00000053508         Gtsf2   chr15
ENSMUSG00000025085 ENSMUSG00000025085        Ablim1   chr19
ENSMUSG00000045573 ENSMUSG00000045573          Penk    chr4
ENSMUSG00000072720 ENSMUSG00000072720        Myo18b    chr5
ENSMUSG00000032827 ENSMUSG00000032827       Ppp1r9a    chr6
ENSMUSG00000022150 ENSMUSG00000022150          Dab2   chr15
ENSMUSG00000055632 ENSMUSG00000055632         Hmcn2    chr2
ENSMUSG00000014030 ENSMUSG00000014030          Pax5    chr4
ENSMUSG00000028339 ENSMUSG00000028339       Col15a1    chr4
ENSMUSG00000041870 ENSMUSG00000041870      Ankrd13a    chr5
ENSMUSG00000028163 ENSMUSG00000028163         Nfkb1    chr3
ENSMUSG00000026553 ENSMUSG00000026553          Copa    chr1
ENSMUSG00000043727 ENSMUSG00000043727 F830045P16Rik    chr2
ENSMUSG00000024163 ENSMUSG00000024163      Mapk8ip3   chr17
ENSMUSG00000055360 ENSMUSG00000055360        Prl2c5   chr13
ENSMUSG00000076441 ENSMUSG00000076441          Ass1    chr2
                        logFC  AveExpr         t      P.Value
ENSMUSG00000028238  0.8598567 8.595271  5.399348 0.0002238033
ENSMUSG00000019935  0.5630412 7.988256  3.906128 0.0024923659
ENSMUSG00000030433  1.1506781 5.995617  5.108829 0.0003491585
ENSMUSG00000028211 -0.9642103 9.565562 -3.674647 0.0037144697
ENSMUSG00000053508  0.5985474 6.779710  3.902085 0.0025096690
ENSMUSG00000025085 -0.5157516 7.271214 -3.598688 0.0042390960
ENSMUSG00000045573 -0.5985459 6.322788 -3.875779 0.0026253569
ENSMUSG00000072720  0.5472046 7.110466  3.516553 0.0048929746
ENSMUSG00000032827  0.6508326 6.063368  3.960623 0.0022708415
ENSMUSG00000022150  0.5663092 7.377018  3.238351 0.0079855514
ENSMUSG00000055632  0.7459775 5.731070  4.189640 0.0015415931
ENSMUSG00000014030 -0.9334248 8.378684 -3.128003 0.0097110586
ENSMUSG00000028339  0.6778944 8.366808  3.104799 0.0101196086
ENSMUSG00000041870  0.7214781 6.489920  3.428460 0.0057104239
ENSMUSG00000028163  0.7222143 6.203134  3.659391 0.0038141789
ENSMUSG00000026553 -0.7175525 6.247766 -3.691373 0.0036082358
ENSMUSG00000043727  0.4992752 6.290711  3.301813 0.0071379495
ENSMUSG00000024163  0.7071335 7.192516  3.101426 0.0101804277
ENSMUSG00000055360  0.5931121 6.016363  3.517230 0.0048871826
ENSMUSG00000076441  0.8333460 5.824304  3.680259 0.0036784698
                   adj.P.Val         B
ENSMUSG00000028238 0.9996913 -3.338675
ENSMUSG00000019935 0.9996913 -3.667056
ENSMUSG00000030433 0.9996913 -3.672889
ENSMUSG00000028211 0.9996913 -3.706796
ENSMUSG00000053508 0.9996913 -3.751293
ENSMUSG00000025085 0.9996913 -3.796128
ENSMUSG00000045573 0.9996913 -3.804787
ENSMUSG00000072720 0.9996913 -3.838459
ENSMUSG00000032827 0.9996913 -3.866294
ENSMUSG00000022150 0.9996913 -3.881341
ENSMUSG00000055632 0.9996913 -3.882306
ENSMUSG00000014030 0.9996913 -3.887765
ENSMUSG00000028339 0.9996913 -3.889628
ENSMUSG00000041870 0.9996913 -3.891751
ENSMUSG00000028163 0.9996913 -3.893159
ENSMUSG00000026553 0.9996913 -3.927467
ENSMUSG00000043727 0.9996913 -3.931498
ENSMUSG00000024163 0.9996913 -3.932816
ENSMUSG00000055360 0.9996913 -3.950047
ENSMUSG00000076441 0.9996913 -3.956538</code></pre>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='fu'><a href='https://rdrr.io/r/base/files2.html'>dir.create</a></span><span class='op'>(</span><span class='fu'>here</span><span class='op'>(</span><span class='st'>"output/DEGs"</span><span class='op'>)</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/files2.html'>dir.create</a></span><span class='op'>(</span><span class='fu'>here</span><span class='op'>(</span><span class='st'>"output/DEGs/filtered_output_3"</span><span class='op'>)</span><span class='op'>)</span></span>
<span></span>
<span><span class='fu'>write.fit</span><span class='op'>(</span><span class='va'>efit</span>, <span class='va'>de_results</span>, file<span class='op'>=</span><span class='fu'>here</span><span class='op'>(</span><span class='st'>"output/DEGs/filtered_output_3/DE_results_voom.txt"</span><span class='op'>)</span><span class='op'>)</span></span></code></pre>
</div>
</div>
<div class="sourceCode" id="cb21"><pre class="sourceCode r distill-force-highlighting-css"><code class="sourceCode r"></code></pre></div>
<!--radix_placeholder_article_footer-->
<!--/radix_placeholder_article_footer-->
</div>

<div class="d-appendix">
</div>


<!--radix_placeholder_site_after_body-->
<!--/radix_placeholder_site_after_body-->
<!--radix_placeholder_appendices-->
<div class="appendix-bottom"></div>
<!--/radix_placeholder_appendices-->
<!--radix_placeholder_navigation_after_body-->
<!--/radix_placeholder_navigation_after_body-->

</body>

</html>
