<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">

<head>
  <meta charset="utf-8"/>
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1"/>
  <meta name="generator" content="distill" />

  <style type="text/css">
  /* Hide doc at startup (prevent jankiness while JS renders/transforms) */
  body {
    visibility: hidden;
  }
  </style>

 <!--radix_placeholder_import_source-->
 <!--/radix_placeholder_import_source-->

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  { color: #00769e; background-color: #f1f3f5; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span { color: #00769e; } /* Normal */
code span.al { color: #ad0000; } /* Alert */
code span.an { color: #5e5e5e; } /* Annotation */
code span.at { color: #657422; } /* Attribute */
code span.bn { color: #ad0000; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #00769e; } /* ControlFlow */
code span.ch { color: #20794d; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #5e5e5e; } /* Comment */
code span.cv { color: #5e5e5e; font-style: italic; } /* CommentVar */
code span.do { color: #5e5e5e; font-style: italic; } /* Documentation */
code span.dt { color: #ad0000; } /* DataType */
code span.dv { color: #ad0000; } /* DecVal */
code span.er { color: #ad0000; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #ad0000; } /* Float */
code span.fu { color: #4758ab; } /* Function */
code span.im { } /* Import */
code span.in { color: #5e5e5e; } /* Information */
code span.kw { color: #00769e; } /* Keyword */
code span.op { color: #5e5e5e; } /* Operator */
code span.ot { color: #00769e; } /* Other */
code span.pp { color: #ad0000; } /* Preprocessor */
code span.sc { color: #5e5e5e; } /* SpecialChar */
code span.ss { color: #20794d; } /* SpecialString */
code span.st { color: #20794d; } /* String */
code span.va { color: #111111; } /* Variable */
code span.vs { color: #20794d; } /* VerbatimString */
code span.wa { color: #5e5e5e; font-style: italic; } /* Warning */
</style>


  <!--radix_placeholder_meta_tags-->
  <title>Socs3 Knockout Bulk RNA-seq: Analysis of the Socs3 knockout data with kallisto alignment.</title>



  <!--  https://schema.org/Article -->
  <meta property="article:published" itemprop="datePublished" content="2025-07-01"/>
  <meta property="article:created" itemprop="dateCreated" content="2025-07-01"/>
  <meta name="article:author" content="Amelia Dunstone"/>

  <!--  https://developers.facebook.com/docs/sharing/webmasters#markup -->
  <meta property="og:title" content="Socs3 Knockout Bulk RNA-seq: Analysis of the Socs3 knockout data with kallisto alignment."/>
  <meta property="og:type" content="article"/>
  <meta property="og:locale" content="en_US"/>
  <meta property="og:site_name" content="Socs3 Knockout Bulk RNA-seq"/>

  <!--  https://dev.twitter.com/cards/types/summary -->
  <meta property="twitter:card" content="summary"/>
  <meta property="twitter:title" content="Socs3 Knockout Bulk RNA-seq: Analysis of the Socs3 knockout data with kallisto alignment."/>

  <!--/radix_placeholder_meta_tags-->
  <!--radix_placeholder_rmarkdown_metadata-->

  <script type="text/json" id="radix-rmarkdown-metadata">
  {"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["title","author","date","editor_options","output"]}},"value":[{"type":"character","attributes":{},"value":["Analysis of the Socs3 knockout data with kallisto alignment."]},{"type":"list","attributes":{},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","url","affiliation","affiliation_url"]}},"value":[{"type":"character","attributes":{},"value":["Amelia Dunstone"]},{"type":"character","attributes":{},"value":["https://github.com/dunstone-a"]},{"type":"character","attributes":{},"value":["St. Vincent's Institute of Medical Research"]},{"type":"character","attributes":{},"value":["https://gitlab.svi.edu.au/biocellgen-public"]}]}]},{"type":"character","attributes":{},"value":["2025-07-01"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["chunk_output_type"]}},"value":[{"type":"character","attributes":{},"value":["console"]}]},{"type":"character","attributes":{},"value":["distill::distill_article"]}]}
  </script>
  <!--/radix_placeholder_rmarkdown_metadata-->
  <!--radix_placeholder_navigation_in_header-->
  <meta name="distill:offset" content=""/>

  <script type="application/javascript">

    window.headroom_prevent_pin = false;

    window.document.addEventListener("DOMContentLoaded", function (event) {

      // initialize headroom for banner
      var header = $('header').get(0);
      var headerHeight = header.offsetHeight;
      var headroom = new Headroom(header, {
        tolerance: 5,
        onPin : function() {
          if (window.headroom_prevent_pin) {
            window.headroom_prevent_pin = false;
            headroom.unpin();
          }
        }
      });
      headroom.init();
      if(window.location.hash)
        headroom.unpin();
      $(header).addClass('headroom--transition');

      // offset scroll location for banner on hash change
      // (see: https://github.com/WickyNilliams/headroom.js/issues/38)
      window.addEventListener("hashchange", function(event) {
        window.scrollTo(0, window.pageYOffset - (headerHeight + 25));
      });

      // responsive menu
      $('.distill-site-header').each(function(i, val) {
        var topnav = $(this);
        var toggle = topnav.find('.nav-toggle');
        toggle.on('click', function() {
          topnav.toggleClass('responsive');
        });
      });

      // nav dropdowns
      $('.nav-dropbtn').click(function(e) {
        $(this).next('.nav-dropdown-content').toggleClass('nav-dropdown-active');
        $(this).parent().siblings('.nav-dropdown')
           .children('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $("body").click(function(e){
        $('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $(".nav-dropdown").click(function(e){
        e.stopPropagation();
      });
    });
  </script>

  <style type="text/css">

  /* Theme (user-documented overrideables for nav appearance) */

  .distill-site-nav {
    color: rgba(255, 255, 255, 0.8);
    background-color: #0F2E3D;
    font-size: 15px;
    font-weight: 300;
  }

  .distill-site-nav a {
    color: inherit;
    text-decoration: none;
  }

  .distill-site-nav a:hover {
    color: white;
  }

  @media print {
    .distill-site-nav {
      display: none;
    }
  }

  .distill-site-header {

  }

  .distill-site-footer {

  }


  /* Site Header */

  .distill-site-header {
    width: 100%;
    box-sizing: border-box;
    z-index: 3;
  }

  .distill-site-header .nav-left {
    display: inline-block;
    margin-left: 8px;
  }

  @media screen and (max-width: 768px) {
    .distill-site-header .nav-left {
      margin-left: 0;
    }
  }


  .distill-site-header .nav-right {
    float: right;
    margin-right: 8px;
  }

  .distill-site-header a,
  .distill-site-header .title {
    display: inline-block;
    text-align: center;
    padding: 14px 10px 14px 10px;
  }

  .distill-site-header .title {
    font-size: 18px;
    min-width: 150px;
  }

  .distill-site-header .logo {
    padding: 0;
  }

  .distill-site-header .logo img {
    display: none;
    max-height: 20px;
    width: auto;
    margin-bottom: -4px;
  }

  .distill-site-header .nav-image img {
    max-height: 18px;
    width: auto;
    display: inline-block;
    margin-bottom: -3px;
  }



  @media screen and (min-width: 1000px) {
    .distill-site-header .logo img {
      display: inline-block;
    }
    .distill-site-header .nav-left {
      margin-left: 20px;
    }
    .distill-site-header .nav-right {
      margin-right: 20px;
    }
    .distill-site-header .title {
      padding-left: 12px;
    }
  }


  .distill-site-header .nav-toggle {
    display: none;
  }

  .nav-dropdown {
    display: inline-block;
    position: relative;
  }

  .nav-dropdown .nav-dropbtn {
    border: none;
    outline: none;
    color: rgba(255, 255, 255, 0.8);
    padding: 16px 10px;
    background-color: transparent;
    font-family: inherit;
    font-size: inherit;
    font-weight: inherit;
    margin: 0;
    margin-top: 1px;
    z-index: 2;
  }

  .nav-dropdown-content {
    display: none;
    position: absolute;
    background-color: white;
    min-width: 200px;
    border: 1px solid rgba(0,0,0,0.15);
    border-radius: 4px;
    box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.1);
    z-index: 1;
    margin-top: 2px;
    white-space: nowrap;
    padding-top: 4px;
    padding-bottom: 4px;
  }

  .nav-dropdown-content hr {
    margin-top: 4px;
    margin-bottom: 4px;
    border: none;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }

  .nav-dropdown-active {
    display: block;
  }

  .nav-dropdown-content a, .nav-dropdown-content .nav-dropdown-header {
    color: black;
    padding: 6px 24px;
    text-decoration: none;
    display: block;
    text-align: left;
  }

  .nav-dropdown-content .nav-dropdown-header {
    display: block;
    padding: 5px 24px;
    padding-bottom: 0;
    text-transform: uppercase;
    font-size: 14px;
    color: #999999;
    white-space: nowrap;
  }

  .nav-dropdown:hover .nav-dropbtn {
    color: white;
  }

  .nav-dropdown-content a:hover {
    background-color: #ddd;
    color: black;
  }

  .nav-right .nav-dropdown-content {
    margin-left: -45%;
    right: 0;
  }

  @media screen and (max-width: 768px) {
    .distill-site-header a, .distill-site-header .nav-dropdown  {display: none;}
    .distill-site-header a.nav-toggle {
      float: right;
      display: block;
    }
    .distill-site-header .title {
      margin-left: 0;
    }
    .distill-site-header .nav-right {
      margin-right: 0;
    }
    .distill-site-header {
      overflow: hidden;
    }
    .nav-right .nav-dropdown-content {
      margin-left: 0;
    }
  }


  @media screen and (max-width: 768px) {
    .distill-site-header.responsive {position: relative; min-height: 500px; }
    .distill-site-header.responsive a.nav-toggle {
      position: absolute;
      right: 0;
      top: 0;
    }
    .distill-site-header.responsive a,
    .distill-site-header.responsive .nav-dropdown {
      display: block;
      text-align: left;
    }
    .distill-site-header.responsive .nav-left,
    .distill-site-header.responsive .nav-right {
      width: 100%;
    }
    .distill-site-header.responsive .nav-dropdown {float: none;}
    .distill-site-header.responsive .nav-dropdown-content {position: relative;}
    .distill-site-header.responsive .nav-dropdown .nav-dropbtn {
      display: block;
      width: 100%;
      text-align: left;
    }
  }

  /* Site Footer */

  .distill-site-footer {
    width: 100%;
    overflow: hidden;
    box-sizing: border-box;
    z-index: 3;
    margin-top: 30px;
    padding-top: 30px;
    padding-bottom: 30px;
    text-align: center;
  }

  /* Headroom */

  d-title {
    padding-top: 6rem;
  }

  @media print {
    d-title {
      padding-top: 4rem;
    }
  }

  .headroom {
    z-index: 1000;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
  }

  .headroom--transition {
    transition: all .4s ease-in-out;
  }

  .headroom--unpinned {
    top: -100px;
  }

  .headroom--pinned {
    top: 0;
  }

  /* adjust viewport for navbar height */
  /* helps vertically center bootstrap (non-distill) content */
  .min-vh-100 {
    min-height: calc(100vh - 100px) !important;
  }

  </style>

  <script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
  <link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"/>
  <link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"/>
  <script src="site_libs/headroom-0.9.4/headroom.min.js"></script>
  <script src="site_libs/autocomplete-0.37.1/autocomplete.min.js"></script>
  <script src="site_libs/fuse-6.4.1/fuse.min.js"></script>

  <script type="application/javascript">

  function getMeta(metaName) {
    var metas = document.getElementsByTagName('meta');
    for (let i = 0; i < metas.length; i++) {
      if (metas[i].getAttribute('name') === metaName) {
        return metas[i].getAttribute('content');
      }
    }
    return '';
  }

  function offsetURL(url) {
    var offset = getMeta('distill:offset');
    return offset ? offset + '/' + url : url;
  }

  function createFuseIndex() {

    // create fuse index
    var options = {
      keys: [
        { name: 'title', weight: 20 },
        { name: 'categories', weight: 15 },
        { name: 'description', weight: 10 },
        { name: 'contents', weight: 5 },
      ],
      ignoreLocation: true,
      threshold: 0
    };
    var fuse = new window.Fuse([], options);

    // fetch the main search.json
    return fetch(offsetURL('search.json'))
      .then(function(response) {
        if (response.status == 200) {
          return response.json().then(function(json) {
            // index main articles
            json.articles.forEach(function(article) {
              fuse.add(article);
            });
            // download collections and index their articles
            return Promise.all(json.collections.map(function(collection) {
              return fetch(offsetURL(collection)).then(function(response) {
                if (response.status === 200) {
                  return response.json().then(function(articles) {
                    articles.forEach(function(article) {
                      fuse.add(article);
                    });
                  })
                } else {
                  return Promise.reject(
                    new Error('Unexpected status from search index request: ' +
                              response.status)
                  );
                }
              });
            })).then(function() {
              return fuse;
            });
          });

        } else {
          return Promise.reject(
            new Error('Unexpected status from search index request: ' +
                        response.status)
          );
        }
      });
  }

  window.document.addEventListener("DOMContentLoaded", function (event) {

    // get search element (bail if we don't have one)
    var searchEl = window.document.getElementById('distill-search');
    if (!searchEl)
      return;

    createFuseIndex()
      .then(function(fuse) {

        // make search box visible
        searchEl.classList.remove('hidden');

        // initialize autocomplete
        var options = {
          autoselect: true,
          hint: false,
          minLength: 2,
        };
        window.autocomplete(searchEl, options, [{
          source: function(query, callback) {
            const searchOptions = {
              isCaseSensitive: false,
              shouldSort: true,
              minMatchCharLength: 2,
              limit: 10,
            };
            var results = fuse.search(query, searchOptions);
            callback(results
              .map(function(result) { return result.item; })
            );
          },
          templates: {
            suggestion: function(suggestion) {
              var img = suggestion.preview && Object.keys(suggestion.preview).length > 0
                ? `<img src="${offsetURL(suggestion.preview)}"</img>`
                : '';
              var html = `
                <div class="search-item">
                  <h3>${suggestion.title}</h3>
                  <div class="search-item-description">
                    ${suggestion.description || ''}
                  </div>
                  <div class="search-item-preview">
                    ${img}
                  </div>
                </div>
              `;
              return html;
            }
          }
        }]).on('autocomplete:selected', function(event, suggestion) {
          window.location.href = offsetURL(suggestion.path);
        });
        // remove inline display style on autocompleter (we want to
        // manage responsive display via css)
        $('.algolia-autocomplete').css("display", "");
      })
      .catch(function(error) {
        console.log(error);
      });

  });

  </script>

  <style type="text/css">

  .nav-search {
    font-size: x-small;
  }

  /* Algolioa Autocomplete */

  .algolia-autocomplete {
    display: inline-block;
    margin-left: 10px;
    vertical-align: sub;
    background-color: white;
    color: black;
    padding: 6px;
    padding-top: 8px;
    padding-bottom: 0;
    border-radius: 6px;
    border: 1px #0F2E3D solid;
    width: 180px;
  }


  @media screen and (max-width: 768px) {
    .distill-site-nav .algolia-autocomplete {
      display: none;
      visibility: hidden;
    }
    .distill-site-nav.responsive .algolia-autocomplete {
      display: inline-block;
      visibility: visible;
    }
    .distill-site-nav.responsive .algolia-autocomplete .aa-dropdown-menu {
      margin-left: 0;
      width: 400px;
      max-height: 400px;
    }
  }

  .algolia-autocomplete .aa-input, .algolia-autocomplete .aa-hint {
    width: 90%;
    outline: none;
    border: none;
  }

  .algolia-autocomplete .aa-hint {
    color: #999;
  }
  .algolia-autocomplete .aa-dropdown-menu {
    width: 550px;
    max-height: 70vh;
    overflow-x: visible;
    overflow-y: scroll;
    padding: 5px;
    margin-top: 3px;
    margin-left: -150px;
    background-color: #fff;
    border-radius: 5px;
    border: 1px solid #999;
    border-top: none;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion {
    cursor: pointer;
    padding: 5px 4px;
    border-bottom: 1px solid #eee;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion:last-of-type {
    border-bottom: none;
    margin-bottom: 2px;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item {
    overflow: hidden;
    font-size: 0.8em;
    line-height: 1.4em;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item h3 {
    font-size: 1rem;
    margin-block-start: 0;
    margin-block-end: 5px;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-description {
    display: inline-block;
    overflow: hidden;
    height: 2.8em;
    width: 80%;
    margin-right: 4%;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview {
    display: inline-block;
    width: 15%;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview img {
    height: 3em;
    width: auto;
    display: none;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview img[src] {
    display: initial;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion.aa-cursor {
    background-color: #eee;
  }
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion em {
    font-weight: bold;
    font-style: normal;
  }

  </style>


  <!--/radix_placeholder_navigation_in_header-->
  <!--radix_placeholder_distill-->

  <style type="text/css">

  body {
    background-color: white;
  }

  .pandoc-table {
    width: 100%;
  }

  .pandoc-table>caption {
    margin-bottom: 10px;
  }

  .pandoc-table th:not([align]) {
    text-align: left;
  }

  .pagedtable-footer {
    font-size: 15px;
  }

  d-byline .byline {
    grid-template-columns: 2fr 2fr;
  }

  d-byline .byline h3 {
    margin-block-start: 1.5em;
  }

  d-byline .byline .authors-affiliations h3 {
    margin-block-start: 0.5em;
  }

  .authors-affiliations .orcid-id {
    width: 16px;
    height:16px;
    margin-left: 4px;
    margin-right: 4px;
    vertical-align: middle;
    padding-bottom: 2px;
  }

  d-title .dt-tags {
    margin-top: 1em;
    grid-column: text;
  }

  .dt-tags .dt-tag {
    text-decoration: none;
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0em 0.4em;
    margin-right: 0.5em;
    margin-bottom: 0.4em;
    font-size: 70%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }

  d-article table.gt_table td,
  d-article table.gt_table th {
    border-bottom: none;
    font-size: 100%;
  }

  .html-widget {
    margin-bottom: 2.0em;
  }

  .l-screen-inset {
    padding-right: 16px;
  }

  .l-screen .caption {
    margin-left: 10px;
  }

  .shaded {
    background: rgb(247, 247, 247);
    padding-top: 20px;
    padding-bottom: 20px;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }

  .shaded .html-widget {
    margin-bottom: 0;
    border: 1px solid rgba(0, 0, 0, 0.1);
  }

  .shaded .shaded-content {
    background: white;
  }

  .text-output {
    margin-top: 0;
    line-height: 1.5em;
  }

  .hidden {
    display: none !important;
  }

  hr.section-separator {
    border: none;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    margin: 0px;
  }


  d-byline {
    border-top: none;
  }

  d-article {
    padding-top: 2.5rem;
    padding-bottom: 30px;
    border-top: none;
  }

  d-appendix {
    padding-top: 30px;
  }

  d-article>p>img {
    width: 100%;
  }

  d-article h2 {
    margin: 1rem 0 1.5rem 0;
  }

  d-article h3 {
    margin-top: 1.5rem;
  }

  d-article iframe {
    border: 1px solid rgba(0, 0, 0, 0.1);
    margin-bottom: 2.0em;
    width: 100%;
  }

  /* Tweak code blocks */

  d-article div.sourceCode code,
  d-article pre code {
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
  }

  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: auto;
  }

  d-article div.sourceCode {
    background-color: white;
  }

  d-article div.sourceCode pre {
    padding-left: 10px;
    font-size: 12px;
    border-left: 2px solid rgba(0,0,0,0.1);
  }

  d-article pre {
    font-size: 12px;
    color: black;
    background: none;
    margin-top: 0;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;

    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;

    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }

  d-article pre a {
    border-bottom: none;
  }

  d-article pre a:hover {
    border-bottom: none;
    text-decoration: underline;
  }

  d-article details {
    grid-column: text;
    margin-bottom: 0.8em;
  }

  @media(min-width: 768px) {

  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: visible !important;
  }

  d-article div.sourceCode pre {
    padding-left: 18px;
    font-size: 14px;
  }

  /* tweak for Pandoc numbered line within distill */
  d-article pre.numberSource code > span {
      left: -2em;
  }

  d-article pre {
    font-size: 14px;
  }

  }

  figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }

  /* CSS for d-contents */

  .d-contents {
    grid-column: text;
    color: rgba(0,0,0,0.8);
    font-size: 0.9em;
    padding-bottom: 1em;
    margin-bottom: 1em;
    padding-bottom: 0.5em;
    margin-bottom: 1em;
    padding-left: 0.25em;
    justify-self: start;
  }

  @media(min-width: 1000px) {
    .d-contents.d-contents-float {
      height: 0;
      grid-column-start: 1;
      grid-column-end: 4;
      justify-self: center;
      padding-right: 3em;
      padding-left: 2em;
    }
  }

  .d-contents nav h3 {
    font-size: 18px;
    margin-top: 0;
    margin-bottom: 1em;
  }

  .d-contents li {
    list-style-type: none
  }

  .d-contents nav > ul {
    padding-left: 0;
  }

  .d-contents ul {
    padding-left: 1em
  }

  .d-contents nav ul li {
    margin-top: 0.6em;
    margin-bottom: 0.2em;
  }

  .d-contents nav a {
    font-size: 13px;
    border-bottom: none;
    text-decoration: none
    color: rgba(0, 0, 0, 0.8);
  }

  .d-contents nav a:hover {
    text-decoration: underline solid rgba(0, 0, 0, 0.6)
  }

  .d-contents nav > ul > li > a {
    font-weight: 600;
  }

  .d-contents nav > ul > li > ul {
    font-weight: inherit;
  }

  .d-contents nav > ul > li > ul > li {
    margin-top: 0.2em;
  }


  .d-contents nav ul {
    margin-top: 0;
    margin-bottom: 0.25em;
  }

  .d-article-with-toc h2:nth-child(2) {
    margin-top: 0;
  }


  /* Figure */

  .figure {
    position: relative;
    margin-bottom: 2.5em;
    margin-top: 1.5em;
  }

  .figure .caption {
    color: rgba(0, 0, 0, 0.6);
    font-size: 12px;
    line-height: 1.5em;
  }

  .figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }

  .figure .caption a {
    color: rgba(0, 0, 0, 0.6);
  }

  .figure .caption b,
  .figure .caption strong, {
    font-weight: 600;
    color: rgba(0, 0, 0, 1.0);
  }

  /* Citations */

  d-article .citation {
    color: inherit;
    cursor: inherit;
  }

  div.hanging-indent{
    margin-left: 1em; text-indent: -1em;
  }

  /* Citation hover box */

  .tippy-box[data-theme~=light-border] {
    background-color: rgba(250, 250, 250, 0.95);
  }

  .tippy-content > p {
    margin-bottom: 0;
    padding: 2px;
  }


  /* Tweak 1000px media break to show more text */

  @media(min-width: 1000px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 80px [middle-start] 50px [text-start kicker-end] 65px 65px 65px 65px 65px 65px 65px 65px [text-end gutter-start] 65px [middle-end] 65px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 16px;
    }

    .grid {
      grid-column-gap: 16px;
    }

    d-article {
      font-size: 1.06rem;
      line-height: 1.7em;
    }
    figure .caption, .figure .caption, figure figcaption {
      font-size: 13px;
    }
  }

  @media(min-width: 1180px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 60px [middle-start] 60px [text-start kicker-end] 60px 60px 60px 60px 60px 60px 60px 60px [text-end gutter-start] 60px [middle-end] 60px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 32px;
    }

    .grid {
      grid-column-gap: 32px;
    }
  }


  /* Get the citation styles for the appendix (not auto-injected on render since
     we do our own rendering of the citation appendix) */

  d-appendix .citation-appendix,
  .d-appendix .citation-appendix {
    font-size: 11px;
    line-height: 15px;
    border-left: 1px solid rgba(0, 0, 0, 0.1);
    padding-left: 18px;
    border: 1px solid rgba(0,0,0,0.1);
    background: rgba(0, 0, 0, 0.02);
    padding: 10px 18px;
    border-radius: 3px;
    color: rgba(150, 150, 150, 1);
    overflow: hidden;
    margin-top: -12px;
    white-space: pre-wrap;
    word-wrap: break-word;
  }

  /* Include appendix styles here so they can be overridden */

  d-appendix {
    contain: layout style;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-top: 60px;
    margin-bottom: 0;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    color: rgba(0,0,0,0.5);
    padding-top: 60px;
    padding-bottom: 48px;
  }

  d-appendix h3 {
    grid-column: page-start / text-start;
    font-size: 15px;
    font-weight: 500;
    margin-top: 1em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.65);
  }

  d-appendix h3 + * {
    margin-top: 1em;
  }

  d-appendix ol {
    padding: 0 0 0 15px;
  }

  @media (min-width: 768px) {
    d-appendix ol {
      padding: 0 0 0 30px;
      margin-left: -30px;
    }
  }

  d-appendix li {
    margin-bottom: 1em;
  }

  d-appendix a {
    color: rgba(0, 0, 0, 0.6);
  }

  d-appendix > * {
    grid-column: text;
  }

  d-appendix > d-footnote-list,
  d-appendix > d-citation-list,
  d-appendix > distill-appendix {
    grid-column: screen;
  }

  /* Include footnote styles here so they can be overridden */

  d-footnote-list {
    contain: layout style;
  }

  d-footnote-list > * {
    grid-column: text;
  }

  d-footnote-list a.footnote-backlink {
    color: rgba(0,0,0,0.3);
    padding-left: 0.5em;
  }



  /* Anchor.js */

  .anchorjs-link {
    /*transition: all .25s linear; */
    text-decoration: none;
    border-bottom: none;
  }
  *:hover > .anchorjs-link {
    margin-left: -1.125em !important;
    text-decoration: none;
    border-bottom: none;
  }

  /* Social footer */

  .social_footer {
    margin-top: 30px;
    margin-bottom: 0;
    color: rgba(0,0,0,0.67);
  }

  .disqus-comments {
    margin-right: 30px;
  }

  .disqus-comment-count {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
    cursor: pointer;
  }

  #disqus_thread {
    margin-top: 30px;
  }

  .article-sharing a {
    border-bottom: none;
    margin-right: 8px;
  }

  .article-sharing a:hover {
    border-bottom: none;
  }

  .sidebar-section.subscribe {
    font-size: 12px;
    line-height: 1.6em;
  }

  .subscribe p {
    margin-bottom: 0.5em;
  }


  .article-footer .subscribe {
    font-size: 15px;
    margin-top: 45px;
  }


  .sidebar-section.custom {
    font-size: 12px;
    line-height: 1.6em;
  }

  .custom p {
    margin-bottom: 0.5em;
  }

  /* Styles for listing layout (hide title) */
  .layout-listing d-title, .layout-listing .d-title {
    display: none;
  }

  /* Styles for posts lists (not auto-injected) */


  .posts-with-sidebar {
    padding-left: 45px;
    padding-right: 45px;
  }

  .posts-list .description h2,
  .posts-list .description p {
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
  }

  .posts-list .description h2 {
    font-weight: 700;
    border-bottom: none;
    padding-bottom: 0;
  }

  .posts-list h2.post-tag {
    border-bottom: 1px solid rgba(0, 0, 0, 0.2);
    padding-bottom: 12px;
  }
  .posts-list {
    margin-top: 60px;
    margin-bottom: 24px;
  }

  .posts-list .post-preview {
    text-decoration: none;
    overflow: hidden;
    display: block;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
    padding: 24px 0;
  }

  .post-preview-last {
    border-bottom: none !important;
  }

  .posts-list .posts-list-caption {
    grid-column: screen;
    font-weight: 400;
  }

  .posts-list .post-preview h2 {
    margin: 0 0 6px 0;
    line-height: 1.2em;
    font-style: normal;
    font-size: 24px;
  }

  .posts-list .post-preview p {
    margin: 0 0 12px 0;
    line-height: 1.4em;
    font-size: 16px;
  }

  .posts-list .post-preview .thumbnail {
    box-sizing: border-box;
    margin-bottom: 24px;
    position: relative;
    max-width: 500px;
  }
  .posts-list .post-preview img {
    width: 100%;
    display: block;
  }

  .posts-list .metadata {
    font-size: 12px;
    line-height: 1.4em;
    margin-bottom: 18px;
  }

  .posts-list .metadata > * {
    display: inline-block;
  }

  .posts-list .metadata .publishedDate {
    margin-right: 2em;
  }

  .posts-list .metadata .dt-authors {
    display: block;
    margin-top: 0.3em;
    margin-right: 2em;
  }

  .posts-list .dt-tags {
    display: block;
    line-height: 1em;
  }

  .posts-list .dt-tags .dt-tag {
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0.3em 0.4em;
    margin-right: 0.2em;
    margin-bottom: 0.4em;
    font-size: 60%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }

  .posts-list img {
    opacity: 1;
  }

  .posts-list img[data-src] {
    opacity: 0;
  }

  .posts-more {
    clear: both;
  }


  .posts-sidebar {
    font-size: 16px;
  }

  .posts-sidebar h3 {
    font-size: 16px;
    margin-top: 0;
    margin-bottom: 0.5em;
    font-weight: 400;
    text-transform: uppercase;
  }

  .sidebar-section {
    margin-bottom: 30px;
  }

  .categories ul {
    list-style-type: none;
    margin: 0;
    padding: 0;
  }

  .categories li {
    color: rgba(0, 0, 0, 0.8);
    margin-bottom: 0;
  }

  .categories li>a {
    border-bottom: none;
  }

  .categories li>a:hover {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
  }

  .categories .active {
    font-weight: 600;
  }

  .categories .category-count {
    color: rgba(0, 0, 0, 0.4);
  }


  @media(min-width: 768px) {
    .posts-list .post-preview h2 {
      font-size: 26px;
    }
    .posts-list .post-preview .thumbnail {
      float: right;
      width: 30%;
      margin-bottom: 0;
    }
    .posts-list .post-preview .description {
      float: left;
      width: 45%;
    }
    .posts-list .post-preview .metadata {
      float: left;
      width: 20%;
      margin-top: 8px;
    }
    .posts-list .post-preview p {
      margin: 0 0 12px 0;
      line-height: 1.5em;
      font-size: 16px;
    }
    .posts-with-sidebar .posts-list {
      float: left;
      width: 75%;
    }
    .posts-with-sidebar .posts-sidebar {
      float: right;
      width: 20%;
      margin-top: 60px;
      padding-top: 24px;
      padding-bottom: 24px;
    }
  }


  /* Improve display for browsers without grid (IE/Edge <= 15) */

  .downlevel {
    line-height: 1.6em;
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
    margin: 0;
  }

  .downlevel .d-title {
    padding-top: 6rem;
    padding-bottom: 1.5rem;
  }

  .downlevel .d-title h1 {
    font-size: 50px;
    font-weight: 700;
    line-height: 1.1em;
    margin: 0 0 0.5rem;
  }

  .downlevel .d-title p {
    font-weight: 300;
    font-size: 1.2rem;
    line-height: 1.55em;
    margin-top: 0;
  }

  .downlevel .d-byline {
    padding-top: 0.8em;
    padding-bottom: 0.8em;
    font-size: 0.8rem;
    line-height: 1.8em;
  }

  .downlevel .section-separator {
    border: none;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
  }

  .downlevel .d-article {
    font-size: 1.06rem;
    line-height: 1.7em;
    padding-top: 1rem;
    padding-bottom: 2rem;
  }


  .downlevel .d-appendix {
    padding-left: 0;
    padding-right: 0;
    max-width: none;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.5);
    padding-top: 40px;
    padding-bottom: 48px;
  }

  .downlevel .footnotes ol {
    padding-left: 13px;
  }

  .downlevel .base-grid,
  .downlevel .distill-header,
  .downlevel .d-title,
  .downlevel .d-abstract,
  .downlevel .d-article,
  .downlevel .d-appendix,
  .downlevel .distill-appendix,
  .downlevel .d-byline,
  .downlevel .d-footnote-list,
  .downlevel .d-citation-list,
  .downlevel .distill-footer,
  .downlevel .appendix-bottom,
  .downlevel .posts-container {
    padding-left: 40px;
    padding-right: 40px;
  }

  @media(min-width: 768px) {
    .downlevel .base-grid,
    .downlevel .distill-header,
    .downlevel .d-title,
    .downlevel .d-abstract,
    .downlevel .d-article,
    .downlevel .d-appendix,
    .downlevel .distill-appendix,
    .downlevel .d-byline,
    .downlevel .d-footnote-list,
    .downlevel .d-citation-list,
    .downlevel .distill-footer,
    .downlevel .appendix-bottom,
    .downlevel .posts-container {
    padding-left: 150px;
    padding-right: 150px;
    max-width: 900px;
  }
  }

  .downlevel pre code {
    display: block;
    border-left: 2px solid rgba(0, 0, 0, .1);
    padding: 0 0 0 20px;
    font-size: 14px;
  }

  .downlevel code, .downlevel pre {
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;

    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;

    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }

  .downlevel .posts-list .post-preview {
    color: inherit;
  }



  </style>

  <script type="application/javascript">

  function is_downlevel_browser() {
    if (bowser.isUnsupportedBrowser({ msie: "12", msedge: "16"},
                                   window.navigator.userAgent)) {
      return true;
    } else {
      return window.load_distill_framework === undefined;
    }
  }

  // show body when load is complete
  function on_load_complete() {

    // add anchors
    if (window.anchors) {
      window.anchors.options.placement = 'left';
      window.anchors.add('d-article > h2, d-article > h3, d-article > h4, d-article > h5');
    }


    // set body to visible
    document.body.style.visibility = 'visible';

    // force redraw for leaflet widgets
    if (window.HTMLWidgets) {
      var maps = window.HTMLWidgets.findAll(".leaflet");
      $.each(maps, function(i, el) {
        var map = this.getMap();
        map.invalidateSize();
        map.eachLayer(function(layer) {
          if (layer instanceof L.TileLayer)
            layer.redraw();
        });
      });
    }

    // trigger 'shown' so htmlwidgets resize
    $('d-article').trigger('shown');
  }

  function init_distill() {

    init_common();

    // create front matter
    var front_matter = $('<d-front-matter></d-front-matter>');
    $('#distill-front-matter').wrap(front_matter);

    // create d-title
    $('.d-title').changeElementType('d-title');

    // separator
    var separator = '<hr class="section-separator" style="clear: both"/>';
    // prepend separator above appendix
    $('.d-byline').before(separator);
    $('.d-article').before(separator);

    // create d-byline
    var byline = $('<d-byline></d-byline>');
    $('.d-byline').replaceWith(byline);

    // create d-article
    var article = $('<d-article></d-article>');
    $('.d-article').wrap(article).children().unwrap();

    // move posts container into article
    $('.posts-container').appendTo($('d-article'));

    // create d-appendix
    $('.d-appendix').changeElementType('d-appendix');

    // flag indicating that we have appendix items
    var appendix = $('.appendix-bottom').children('h3').length > 0;

    // replace footnotes with <d-footnote>
    $('.footnote-ref').each(function(i, val) {
      appendix = true;
      var href = $(this).attr('href');
      var id = href.replace('#', '');
      var fn = $('#' + id);
      var fn_p = $('#' + id + '>p');
      fn_p.find('.footnote-back').remove();
      var text = fn_p.html();
      var dtfn = $('<d-footnote></d-footnote>');
      dtfn.html(text);
      $(this).replaceWith(dtfn);
    });
    // remove footnotes
    $('.footnotes').remove();

    // move refs into #references-listing
    $('#references-listing').replaceWith($('#refs'));

    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      var id = $(this).attr('id');
      $('.d-contents a[href="#' + id + '"]').parent().remove();
      appendix = true;
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('d-appendix'));
    });

    // show d-appendix if we have appendix content
    $("d-appendix").css('display', appendix ? 'grid' : 'none');

    // localize layout chunks to just output
    $('.layout-chunk').each(function(i, val) {

      // capture layout
      var layout = $(this).attr('data-layout');

      // apply layout to markdown level block elements
      var elements = $(this).children().not('details, div.sourceCode, pre, script');
      elements.each(function(i, el) {
        var layout_div = $('<div class="' + layout + '"></div>');
        if (layout_div.hasClass('shaded')) {
          var shaded_content = $('<div class="shaded-content"></div>');
          $(this).wrap(shaded_content);
          $(this).parent().wrap(layout_div);
        } else {
          $(this).wrap(layout_div);
        }
      });


      // unwrap the layout-chunk div
      $(this).children().unwrap();
    });

    // remove code block used to force  highlighting css
    $('.distill-force-highlighting-css').parent().remove();

    // remove empty line numbers inserted by pandoc when using a
    // custom syntax highlighting theme, except when numbering line
    // in code chunk
    $('pre:not(.numberLines) code.sourceCode a:empty').remove();

    // load distill framework
    load_distill_framework();

    // wait for window.distillRunlevel == 4 to do post processing
    function distill_post_process() {

      if (!window.distillRunlevel || window.distillRunlevel < 4)
        return;

      // hide author/affiliations entirely if we have no authors
      var front_matter = JSON.parse($("#distill-front-matter").html());
      var have_authors = front_matter.authors && front_matter.authors.length > 0;
      if (!have_authors)
        $('d-byline').addClass('hidden');

      // article with toc class
      $('.d-contents').parent().addClass('d-article-with-toc');

      // strip links that point to #
      $('.authors-affiliations').find('a[href="#"]').removeAttr('href');

      // add orcid ids
      $('.authors-affiliations').find('.author').each(function(i, el) {
        var orcid_id = front_matter.authors[i].orcidID;
        var author_name = front_matter.authors[i].author
        if (orcid_id) {
          var a = $('<a></a>');
          a.attr('href', 'https://orcid.org/' + orcid_id);
          var img = $('<img></img>');
          img.addClass('orcid-id');
          img.attr('alt', author_name ? 'ORCID ID for ' + author_name : 'ORCID ID');
          img.attr('src','data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg==');
          a.append(img);
          $(this).append(a);
        }
      });

      // hide elements of author/affiliations grid that have no value
      function hide_byline_column(caption) {
        $('d-byline').find('h3:contains("' + caption + '")').parent().css('visibility', 'hidden');
      }

      // affiliations
      var have_affiliations = false;
      for (var i = 0; i<front_matter.authors.length; ++i) {
        var author = front_matter.authors[i];
        if (author.affiliation !== "&nbsp;") {
          have_affiliations = true;
          break;
        }
      }
      if (!have_affiliations)
        $('d-byline').find('h3:contains("Affiliations")').css('visibility', 'hidden');

      // published date
      if (!front_matter.publishedDate)
        hide_byline_column("Published");

      // document object identifier
      var doi = $('d-byline').find('h3:contains("DOI")');
      var doi_p = doi.next().empty();
      if (!front_matter.doi) {
        // if we have a citation and valid citationText then link to that
        if ($('#citation').length > 0 && front_matter.citationText) {
          doi.html('Citation');
          $('<a href="#citation"></a>')
            .text(front_matter.citationText)
            .appendTo(doi_p);
        } else {
          hide_byline_column("DOI");
        }
      } else {
        $('<a></a>')
           .attr('href', "https://doi.org/" + front_matter.doi)
           .html(front_matter.doi)
           .appendTo(doi_p);
      }

       // change plural form of authors/affiliations
      if (front_matter.authors.length === 1) {
        var grid = $('.authors-affiliations');
        grid.children('h3:contains("Authors")').text('Author');
        grid.children('h3:contains("Affiliations")').text('Affiliation');
      }

      // remove d-appendix and d-footnote-list local styles
      $('d-appendix > style:first-child').remove();
      $('d-footnote-list > style:first-child').remove();

      // move appendix-bottom entries to the bottom
      $('.appendix-bottom').appendTo('d-appendix').children().unwrap();
      $('.appendix-bottom').remove();

      // hoverable references
      $('span.citation[data-cites]').each(function() {
        const citeChild = $(this).children()[0]
        // Do not process if @xyz has been used without escaping and without bibliography activated
        // https://github.com/rstudio/distill/issues/466
        if (citeChild === undefined) return true

        if (citeChild.nodeName == "D-FOOTNOTE") {
          var fn = citeChild
          $(this).html(fn.shadowRoot.querySelector("sup"))
          $(this).id = fn.id
          fn.remove()
        }
        var refs = $(this).attr('data-cites').split(" ");
        var refHtml = refs.map(function(ref) {
          // Could use CSS.escape too here, we insure backward compatibility in navigator
          return "<p>" + $('div[id="ref-' + ref + '"]').html() + "</p>";
        }).join("\n");
        window.tippy(this, {
          allowHTML: true,
          content: refHtml,
          maxWidth: 500,
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start'
        });
      });

      // fix footnotes in tables (#411)
      // replacing broken distill.pub feature
      $('table d-footnote').each(function() {
        // we replace internal showAtNode methode which is triggered when hovering a footnote
        this.hoverBox.showAtNode = function(node) {
          // ported from https://github.com/distillpub/template/pull/105/files
          calcOffset = function(elem) {
              let x = elem.offsetLeft;
              let y = elem.offsetTop;
              // Traverse upwards until an `absolute` element is found or `elem`
              // becomes null.
              while (elem = elem.offsetParent && elem.style.position != 'absolute') {
                  x += elem.offsetLeft;
                  y += elem.offsetTop;
              }

              return { left: x, top: y };
          }
          // https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/offsetTop
          const bbox = node.getBoundingClientRect();
          const offset = calcOffset(node);
          this.show([offset.left + bbox.width, offset.top + bbox.height]);
        }
      })

      // clear polling timer
      clearInterval(tid);

      // show body now that everything is ready
      on_load_complete();
    }

    var tid = setInterval(distill_post_process, 50);
    distill_post_process();

  }

  function init_downlevel() {

    init_common();

     // insert hr after d-title
    $('.d-title').after($('<hr class="section-separator"/>'));

    // check if we have authors
    var front_matter = JSON.parse($("#distill-front-matter").html());
    var have_authors = front_matter.authors && front_matter.authors.length > 0;

    // manage byline/border
    if (!have_authors)
      $('.d-byline').remove();
    $('.d-byline').after($('<hr class="section-separator"/>'));
    $('.d-byline a').remove();

    // remove toc
    $('.d-contents').remove();

    // move appendix elements
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('.d-appendix'));
    });


    // inject headers into references and footnotes
    var refs_header = $('<h3></h3>');
    refs_header.text('References');
    $('#refs').prepend(refs_header);

    var footnotes_header = $('<h3></h3');
    footnotes_header.text('Footnotes');
    $('.footnotes').children('hr').first().replaceWith(footnotes_header);

    // move appendix-bottom entries to the bottom
    $('.appendix-bottom').appendTo('.d-appendix').children().unwrap();
    $('.appendix-bottom').remove();

    // remove appendix if it's empty
    if ($('.d-appendix').children().length === 0)
      $('.d-appendix').remove();

    // prepend separator above appendix
    $('.d-appendix').before($('<hr class="section-separator" style="clear: both"/>'));

    // trim code
    $('pre>code').each(function(i, val) {
      $(this).html($.trim($(this).html()));
    });

    // move posts-container right before article
    $('.posts-container').insertBefore($('.d-article'));

    $('body').addClass('downlevel');

    on_load_complete();
  }


  function init_common() {

    // jquery plugin to change element types
    (function($) {
      $.fn.changeElementType = function(newType) {
        var attrs = {};

        $.each(this[0].attributes, function(idx, attr) {
          attrs[attr.nodeName] = attr.nodeValue;
        });

        this.replaceWith(function() {
          return $("<" + newType + "/>", attrs).append($(this).contents());
        });
      };
    })(jQuery);

    // prevent underline for linked images
    $('a > img').parent().css({'border-bottom' : 'none'});

    // mark non-body figures created by knitr chunks as 100% width
    $('.layout-chunk').each(function(i, val) {
      var figures = $(this).find('img, .html-widget');
      // ignore leaflet img layers (#106)
      figures = figures.filter(':not(img[class*="leaflet"])')
      if ($(this).attr('data-layout') !== "l-body") {
        figures.css('width', '100%');
      } else {
        figures.css('max-width', '100%');
        figures.filter("[width]").each(function(i, val) {
          var fig = $(this);
          fig.css('width', fig.attr('width') + 'px');
        });

      }
    });

    // auto-append index.html to post-preview links in file: protocol
    // and in rstudio ide preview
    $('.post-preview').each(function(i, val) {
      if (window.location.protocol === "file:")
        $(this).attr('href', $(this).attr('href') + "index.html");
    });

    // get rid of index.html references in header
    if (window.location.protocol !== "file:") {
      $('.distill-site-header a[href]').each(function(i,val) {
        $(this).attr('href', $(this).attr('href').replace(/^index[.]html/, "./"));
      });
    }

    // add class to pandoc style tables
    $('tr.header').parent('thead').parent('table').addClass('pandoc-table');
    $('.kable-table').children('table').addClass('pandoc-table');

    // add figcaption style to table captions
    $('caption').parent('table').addClass("figcaption");

    // initialize posts list
    if (window.init_posts_list)
      window.init_posts_list();

    // implmement disqus comment link
    $('.disqus-comment-count').click(function() {
      window.headroom_prevent_pin = true;
      $('#disqus_thread').toggleClass('hidden');
      if (!$('#disqus_thread').hasClass('hidden')) {
        var offset = $(this).offset();
        $(window).resize();
        $('html, body').animate({
          scrollTop: offset.top - 35
        });
      }
    });
  }

  document.addEventListener('DOMContentLoaded', function() {
    if (is_downlevel_browser())
      init_downlevel();
    else
      window.addEventListener('WebComponentsReady', init_distill);
  });

  </script>

  <!--/radix_placeholder_distill-->
  <script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
  <script src="site_libs/popper-2.6.0/popper.min.js"></script>
  <link href="site_libs/tippy-6.2.7/tippy.css" rel="stylesheet" />
  <link href="site_libs/tippy-6.2.7/tippy-light-border.css" rel="stylesheet" />
  <script src="site_libs/tippy-6.2.7/tippy.umd.min.js"></script>
  <script src="site_libs/anchor-4.2.2/anchor.min.js"></script>
  <script src="site_libs/bowser-1.9.3/bowser.min.js"></script>
  <script src="site_libs/webcomponents-2.0.0/webcomponents.js"></script>
  <script src="site_libs/distill-2.2.21/template.v2.js"></script>
  <!--radix_placeholder_site_in_header-->
  <!--/radix_placeholder_site_in_header-->


</head>

<body>

<!--radix_placeholder_front_matter-->

<script id="distill-front-matter" type="text/json">
{"title":"Analysis of the Socs3 knockout data with kallisto alignment.","authors":[{"author":"Amelia Dunstone","authorURL":"https://github.com/dunstone-a","affiliation":"St. Vincent's Institute of Medical Research","affiliationURL":"https://gitlab.svi.edu.au/biocellgen-public","orcidID":""}],"publishedDate":"2025-07-01T00:00:00.000+10:00","citationText":"Dunstone, 2025"}
</script>

<!--/radix_placeholder_front_matter-->
<!--radix_placeholder_navigation_before_body-->
<header class="header header--fixed" role="banner">
<nav class="distill-site-nav distill-site-header">
<div class="nav-left">
<a href="index.html" class="title">Socs3 Knockout Bulk RNA-seq</a>
<a href="index.html">Home</a>
<div class="nav-dropdown">
<button class="nav-dropbtn">
Kallisto
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<a href="Kallisto_analysis.html">Kallisto analysis</a>
</div>
</div>
<div class="nav-dropdown">
<button class="nav-dropbtn">
Hisat2 (old analysis)
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<a href="Filtered_output_3_analysis.html">Analysis of filtered output 3</a>
<a href="Lane_based_analysis.html">Lane based analysis</a>
</div>
</div>
<div class="nav-dropdown">
<button class="nav-dropbtn">
Supplementary files
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<a href="../output/Socs3_MultiQC_multiqc_report.html">MultiQC plot of lanes</a>
<a href="../output/Socs3_MultiQC_merged_multiqc_report.html">MultiQC plot of samples</a>
<a href="../output/DEGs/">Differential expression testing results</a>
</div>
</div>
<a href="FurtherAnalysis.html">Further analyses</a>
</div>
<div class="nav-right">
<a href="https://gitlab.svi.edu.au/biocellgen-private/socs3_knockout_rna" aria-label="Source code">
<i class="fab fa-github" aria-hidden="true"></i>
</a>
<a href="javascript:void(0);" class="nav-toggle">&#9776;</a>
</div>
</nav>
</header>
<!--/radix_placeholder_navigation_before_body-->
<!--radix_placeholder_site_before_body-->
<!--/radix_placeholder_site_before_body-->

<div class="d-title">
<h1>Analysis of the Socs3 knockout data with kallisto alignment.</h1>

<!--radix_placeholder_categories-->
<!--/radix_placeholder_categories-->

</div>

<div class="d-byline">
  Amelia Dunstone <a href="https://github.com/dunstone-a" class="uri">https://github.com/dunstone-a</a> (St.Vincents Institute of Medical Research)<a href="https://gitlab.svi.edu.au/biocellgen-public" class="uri">https://gitlab.svi.edu.au/biocellgen-public</a>
  
<br/>2025-07-01
</div>

<div class="d-article">
<div class="d-contents d-contents-float">
<nav class="l-text toc figcaption" id="TOC">
<h3>Contents</h3>
<ul>
<li><a href="#introduction" id="toc-introduction">Introduction</a></li>
<li><a href="#setting-up-the-data" id="toc-setting-up-the-data">Setting up the data</a></li>
<li><a href="#pre-processing" id="toc-pre-processing">Pre-processing</a></li>
<li><a href="#visualising-the-data" id="toc-visualising-the-data">Visualising the data</a></li>
<li><a href="#differential-expression-analyses" id="toc-differential-expression-analyses">Differential Expression Analyses</a></li>
<li><a href="#concluding-remarks" id="toc-concluding-remarks">Concluding remarks</a></li>
</ul>
</nav>
</div>
<h1 id="introduction">Introduction</h1>
<p>This report includes an analysis of the Socs3 knockout bulk RNA-sequencing data.
The data includes 6 samples, 3 from the control group (wild-type) and 3 from the Socs3 knockout group.
The samples are from bone tissue of mice (and some samples contain multiple mice).
We are interested in seeing if there are differences in gene expression between these groups, as this will help to understand the role of Socs3.</p>
<p>The gene filtering, normalisation, MDS plot and differential expression testing is based on the guide RNA-seq analysis is easy as 1-2-3 with limma, Glimma and edgeR (Law et al.2018), see <a href="https://pmc.ncbi.nlm.nih.gov/articles/PMC4937821/">https://pmc.ncbi.nlm.nih.gov/articles/PMC4937821/</a>.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># Load required libraries for the document </span></span>
<span><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://here.r-lib.org/'>here</a></span><span class='op'>)</span></span>
<span><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://github.com/thelovelab/tximport'>tximport</a></span><span class='op'>)</span></span>
<span><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://github.com/grimbough/biomaRt'>biomaRt</a></span><span class='op'>)</span></span>
<span><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://bioinf.wehi.edu.au/edgeR/'>edgeR</a></span><span class='op'>)</span></span>
<span><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://patchwork.data-imaginist.com'>patchwork</a></span><span class='op'>)</span></span>
<span><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://bioinf.wehi.edu.au/limma/'>limma</a></span><span class='op'>)</span> </span>
<span><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'>RColorBrewer</span><span class='op'>)</span> </span>
<span><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://ggplot2.tidyverse.org'>ggplot2</a></span><span class='op'>)</span> </span>
<span><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'>gridExtra</span><span class='op'>)</span> </span>
<span><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://dplyr.tidyverse.org'>dplyr</a></span><span class='op'>)</span></span>
<span><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'>pheatmap</span><span class='op'>)</span></span></code></pre>
</div>
</details>
</div>
<h1 id="setting-up-the-data">Setting up the data</h1>
<p>There are a number of pre-processing steps required to obtain a count matrix of this data.
First, the .fastq files from the same sample and read combination were merged together (see <code>code/03_merge_lanes.sh</code>.
There are 6 samples so this results in 12 .fastq files (read 1 and read 2 for each sample).</p>
<p>Next, the paired-end reads were aligned to the reference mouse transcriptome using <code>kallisto</code>.</p>
<h2 id="reading-in-count-data">Reading in count data</h2>
<p>Since <code>kallisto</code> uses transcript IDs instead of gene IDs, we will use the <code>biomaRt</code> package to map transcript IDs to gene IDs.
When we import the data using <code>tximport</code>, the transcripts are aggregated at the gene level using this mapping.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>samples_output</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/list.files.html'>list.files</a></span><span class='op'>(</span><span class='fu'><a href='https://here.r-lib.org/reference/here.html'>here</a></span><span class='op'>(</span><span class='st'>"data/kallisto"</span><span class='op'>)</span><span class='op'>)</span></span>
<span></span>
<span><span class='co'># Get this to work and look at the files</span></span>
<span><span class='va'>file_paths</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://here.r-lib.org/reference/here.html'>here</a></span><span class='op'>(</span><span class='st'>"data/kallisto/"</span>, <span class='va'>samples_output</span>, <span class='st'>"abundance.h5"</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>file_paths</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/grep.html'>gsub</a></span><span class='op'>(</span><span class='st'>"_output"</span>, <span class='st'>""</span>, <span class='va'>samples_output</span><span class='op'>)</span></span>
<span></span>
<span><span class='co'># Use biomaRt to map transcripts to gene_ids</span></span>
<span><span class='va'>mart</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/biomaRt/man/useMart.html'>useMart</a></span><span class='op'>(</span><span class='st'>"ensembl"</span>, dataset <span class='op'>=</span> <span class='st'>"mmusculus_gene_ensembl"</span>, verbose <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<pre><code>BioMartServer running BioMart version: 0.7
Mart virtual schema: default
Mart host: https://www.ensembl.org:443/biomart/martservice</code></pre>
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>transcript_map</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/biomaRt/man/getBM.html'>getBM</a></span><span class='op'>(</span></span>
<span>  attributes <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"ensembl_transcript_id"</span>,</span>
<span>                 <span class='st'>"ensembl_gene_id"</span>,</span>
<span>                 <span class='st'>"external_gene_name"</span>,</span>
<span>                 <span class='st'>"chromosome_name"</span><span class='op'>)</span>,</span>
<span>  mart <span class='op'>=</span> <span class='va'>mart</span></span>
<span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>txi.kallisto</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/tximport/man/tximport.html'>tximport</a></span><span class='op'>(</span></span>
<span>    <span class='va'>file_paths</span>, </span>
<span>    type <span class='op'>=</span> <span class='st'>"kallisto"</span>, </span>
<span>    txOut <span class='op'>=</span> <span class='cn'>FALSE</span>,</span>
<span>    tx2gene <span class='op'>=</span> <span class='va'>transcript_map</span><span class='op'>[</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"ensembl_transcript_id"</span>, <span class='st'>"ensembl_gene_id"</span><span class='op'>)</span><span class='op'>]</span>,</span>
<span>    ignoreTxVersion <span class='op'>=</span> <span class='cn'>TRUE</span></span>
<span>    <span class='op'>)</span></span></code></pre>
</div>
</details>
</div>
<p>Next, we create a DGEList object from the count data and the sample sheet.
The count data contains a column for each sample of the experiment (6 in total).
The sample sheet (located in <code>data/sample_sheet.csv</code>) includes information about the mouse genotype, the batches, and other experimental information.</p>
<p>The DGEList object includes counts for the numeric data, and the <code>samples</code> and <code>genes</code> data frames, which contains information about the samples and features, respectively.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># Read in count data</span></span>
<span><span class='va'>counts</span> <span class='op'>&lt;-</span> <span class='va'>txi.kallisto</span><span class='op'>$</span><span class='va'>counts</span></span>
<span></span>
<span><span class='co'># Subset the transcript_map to match our data</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/dim.html'>dim</a></span><span class='op'>(</span><span class='va'>transcript_map</span><span class='op'>)</span></span>
<span><span class='va'>gene_info</span> <span class='op'>&lt;-</span> <span class='va'>transcript_map</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/match.html'>match</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/row.names.html'>row.names</a></span><span class='op'>(</span><span class='va'>counts</span><span class='op'>)</span>, <span class='va'>transcript_map</span><span class='op'>$</span><span class='va'>ensembl_gene_id</span><span class='op'>)</span>, <span class='op'>]</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/dim.html'>dim</a></span><span class='op'>(</span><span class='va'>gene_info</span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>gene_info</span><span class='op'>$</span><span class='va'>CHR</span> <span class='op'>&lt;-</span> <span class='va'>gene_info</span><span class='op'>$</span><span class='va'>chromosome_name</span></span>
<span><span class='va'>gene_info</span><span class='op'>$</span><span class='va'>SYMBOL</span> <span class='op'>&lt;-</span> <span class='va'>gene_info</span><span class='op'>$</span><span class='va'>external_gene_name</span></span>
<span><span class='va'>gene_info</span><span class='op'>$</span><span class='va'>ENSEMBL</span> <span class='op'>&lt;-</span> <span class='va'>gene_info</span><span class='op'>$</span><span class='va'>ensembl_gene_id</span></span></code></pre>
</div>
</details>
</div>
<p>There are 36091 genes currently in the object.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># Read in sample sheet</span></span>
<span><span class='va'>samples</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/utils/read.table.html'>read.csv</a></span><span class='op'>(</span><span class='fu'><a href='https://here.r-lib.org/reference/here.html'>here</a></span><span class='op'>(</span><span class='st'>"data/sample_sheet.csv"</span><span class='op'>)</span>, row.names <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span></span>
<span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>counts</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/grep.html'>gsub</a></span><span class='op'>(</span><span class='st'>"_22YMFCLT3"</span>, <span class='st'>""</span>, <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>counts</span><span class='op'>)</span><span class='op'>)</span></span>
<span></span>
<span><span class='co'># The sample sheet is already in the correct order</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/stopifnot.html'>stopifnot</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/identical.html'>identical</a></span><span class='op'>(</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>Sample_name</span>, <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>counts</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span></span>
<span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>counts</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span></span></code></pre>
</div>
</details>
</div>
<h2 id="kallisto-mapping-information">Kallisto mapping information</h2>
<p>The table below shows the percentage of reads which are mapped.
This mapping rate looks okay.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://jeroen.r-universe.dev/jsonlite'>jsonlite</a></span><span class='op'>)</span></span>
<span></span>
<span><span class='co'># Path to each kallisto folder</span></span>
<span><span class='va'>sample_dirs</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/basename.html'>dirname</a></span><span class='op'>(</span><span class='va'>file_paths</span><span class='op'>)</span></span>
<span></span>
<span><span class='co'># Extract mapping rate from run_info.json for each sample</span></span>
<span><span class='va'>mapping_rates</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/lapply.html'>sapply</a></span><span class='op'>(</span><span class='va'>sample_dirs</span>, <span class='kw'>function</span><span class='op'>(</span><span class='va'>dir</span><span class='op'>)</span> <span class='op'>{</span></span>
<span>  <span class='va'>info</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://jeroen.r-universe.dev/jsonlite/reference/fromJSON.html'>fromJSON</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/file.path.html'>file.path</a></span><span class='op'>(</span><span class='va'>dir</span>, <span class='st'>"run_info.json"</span><span class='op'>)</span><span class='op'>)</span></span>
<span>  <span class='va'>info</span><span class='op'>$</span><span class='va'>p_pseudoaligned</span></span>
<span><span class='op'>}</span><span class='op'>)</span> <span class='op'>/</span> <span class='fl'>100</span></span>
<span></span>
<span><span class='va'>df</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span></span>
<span>    File <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/grep.html'>gsub</a></span><span class='op'>(</span><span class='st'>"_22YMFCLT3_output"</span>, <span class='st'>""</span>, <span class='fu'><a href='https://rdrr.io/r/base/basename.html'>basename</a></span><span class='op'>(</span><span class='va'>sample_dirs</span><span class='op'>)</span><span class='op'>)</span>,</span>
<span>    Mapping_rate <span class='op'>=</span> <span class='va'>mapping_rates</span><span class='op'>)</span></span>
<span><span class='va'>df</span><span class='op'>$</span><span class='va'>Group</span> <span class='op'>&lt;-</span> <span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/match.html'>match</a></span><span class='op'>(</span><span class='va'>df</span><span class='op'>$</span><span class='va'>File</span>, <span class='va'>samples</span><span class='op'>$</span><span class='va'>Sample_name</span><span class='op'>)</span><span class='op'>]</span></span>
<span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>df</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='cn'>NULL</span></span>
<span><span class='va'>df</span><span class='op'>[</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Group"</span>, <span class='st'>"Mapping_rate"</span><span class='op'>)</span><span class='op'>]</span><span class='op'>|&gt;</span> </span>
<span>    <span class='fu'>janitor</span><span class='fu'>::</span><span class='fu'><a href='https://sfirke.github.io/janitor/reference/adorn_pct_formatting.html'>adorn_pct_formatting</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>|&gt;</span></span>
<span>    <span class='fu'>knitr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/knitr/man/kable.html'>kable</a></span><span class='op'>(</span><span class='op'>)</span> </span></code></pre>
</div>
</details>
<table>
<thead>
<tr class="header">
<th style="text-align: left;">Group</th>
<th style="text-align: left;">Mapping_rate</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">WT_1_B2</td>
<td style="text-align: left;">53.8%</td>
</tr>
<tr class="even">
<td style="text-align: left;">SOCS3_1_B2</td>
<td style="text-align: left;">49.6%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">SOCS3_2_B1</td>
<td style="text-align: left;">73.0%</td>
</tr>
<tr class="even">
<td style="text-align: left;">SOCS3_3_B1</td>
<td style="text-align: left;">63.7%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">WT_2_B1</td>
<td style="text-align: left;">64.8%</td>
</tr>
<tr class="even">
<td style="text-align: left;">WT_3_B1</td>
<td style="text-align: left;">73.0%</td>
</tr>
</tbody>
</table>
</div>
<p>Next we create a DGEList from the count data and the sample sheet.
The count data contains a column for each sample of the experiment (6 in total).
The sample sheet includes information about the mouse genotype and the batches.</p>
<p>The DGEList object includes counts for the numeric data, and the <code>samples</code> and <code>genes</code> data frames, which contains information about the samples and features, respectively.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># Create the DGEList. </span></span>
<span><span class='va'>dge</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/DGEList.html'>DGEList</a></span><span class='op'>(</span></span>
<span>    <span class='va'>counts</span>, </span>
<span>    samples <span class='op'>=</span> <span class='va'>samples</span>, </span>
<span>    group <span class='op'>=</span> <span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span>, </span>
<span>    genes <span class='op'>=</span> <span class='va'>gene_info</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/dim.html'>dim</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<pre><code>[1] 36091     6</code></pre>
</div>
<h2 id="organising-sample-information">Organising sample information</h2>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>group</span> <span class='op'>&lt;-</span> <span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span></span>
<span><span class='va'>batch</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/grep.html'>gsub</a></span><span class='op'>(</span><span class='st'>".+_"</span>, <span class='st'>""</span>, <span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span><span class='op'>)</span></span>
<span><span class='va'>treatment</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/grep.html'>gsub</a></span><span class='op'>(</span><span class='st'>"_.+"</span>, <span class='st'>""</span>, <span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span><span class='op'>)</span></span>
<span></span>
<span><span class='co'># Make sample info factors</span></span>
<span><span class='va'>group</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span><span class='va'>group</span>, levels<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"WT_1_B2"</span>, <span class='st'>"WT_2_B1"</span>, <span class='st'>"WT_3_B1"</span>, </span>
<span>    <span class='st'>"SOCS3_1_B2"</span>, <span class='st'>"SOCS3_2_B1"</span>, <span class='st'>"SOCS3_3_B1"</span><span class='op'>)</span><span class='op'>)</span></span>
<span><span class='va'>batch</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span><span class='va'>batch</span><span class='op'>)</span></span>
<span><span class='va'>treatment</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span><span class='va'>treatment</span>, levels<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"WT"</span>, <span class='st'>"SOCS3"</span><span class='op'>)</span><span class='op'>)</span></span>
<span></span>
<span><span class='co'># Add sample info into samples slot. </span></span>
<span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>batch</span> <span class='op'>&lt;-</span> <span class='va'>batch</span></span>
<span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>treatment</span> <span class='op'>&lt;-</span> <span class='va'>treatment</span></span></code></pre>
</div>
</details>
</div>
<p>There are 6 samples, 3 for each treatment.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='fu'><a href='https://rdrr.io/r/base/table.html'>table</a></span><span class='op'>(</span><span class='va'>group</span>, <span class='va'>treatment</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<pre><code>            treatment
group        WT SOCS3
  WT_1_B2     1     0
  WT_2_B1     1     0
  WT_3_B1     1     0
  SOCS3_1_B2  0     1
  SOCS3_2_B1  0     1
  SOCS3_3_B1  0     1</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># Remove group, treatment and batch. These should only be accessed from dge!!!</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/rm.html'>rm</a></span><span class='op'>(</span><span class='va'>group</span>, <span class='va'>batch</span>, <span class='va'>treatment</span><span class='op'>)</span></span></code></pre>
</div>
</details>
</div>
<h2 id="organising-gene-annotations">Organising gene annotations</h2>
<p>After mapping the sequencing reads to the transcriptome, we have Ensembl-style transcript identifiers.
These are used by mapping tools as they are unambiguous and highly stable.
We have used the <code>biomaRt</code> package to match these transcript IDs to gene IDs.
From the gene IDs, we also obtain gene symbols and chromosome information, for easier interpretation of the results.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># There are hundreds of duplicate symbols.</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/table.html'>table</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/duplicated.html'>duplicated</a></span><span class='op'>(</span><span class='va'>gene_info</span><span class='op'>$</span><span class='va'>SYMBOL</span><span class='op'>)</span><span class='op'>)</span></span>
<span><span class='co'># No duplicate ENSEMBL id's now. </span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/table.html'>table</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/duplicated.html'>duplicated</a></span><span class='op'>(</span><span class='va'>gene_info</span><span class='op'>$</span><span class='va'>ENSEMBL</span><span class='op'>)</span><span class='op'>)</span></span></code></pre>
</div>
</details>
</div>
<p>There are some Ensembl IDs which map to the same gene symbols.
For these features we will refer to the genes using both the Ensembl ID and the symbol, for example Pakap_ENSMUSG00000038729.25.
This ensures that rownames are unique, giving us a robust way of subsetting the object and referring to genes in the analysis.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'>scuttle</span><span class='op'>)</span></span>
<span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scuttle/man/uniquifyFeatureNames.html'>uniquifyFeatureNames</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>genes</span><span class='op'>$</span><span class='va'>ENSEMBL</span>, <span class='va'>dge</span><span class='op'>$</span><span class='va'>genes</span><span class='op'>$</span><span class='va'>SYMBOL</span><span class='op'>)</span></span>
<span><span class='co'># Duplicate features with symbols have their ID concatenated.  </span></span>
<span><span class='co'># rownames(dge)[duplicated(dge$genes$SYMBOL) &amp; !is.na(dge$genes$SYMBOL)]</span></span>
<span><span class='co'># Now there are unique rownames. </span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/table.html'>table</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/duplicated.html'>duplicated</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span></span></code></pre>
</div>
</details>
</div>
<p>We save the DGEList object to file, located at <code>data/kallisto.dge.rds</code>.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='fu'><a href='https://rdrr.io/r/base/readRDS.html'>saveRDS</a></span><span class='op'>(</span><span class='va'>dge</span>, <span class='fu'><a href='https://here.r-lib.org/reference/here.html'>here</a></span><span class='op'>(</span><span class='st'>"data/kallisto.dge.rds"</span><span class='op'>)</span><span class='op'>)</span></span>
<span><span class='va'>dge</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/readRDS.html'>readRDS</a></span><span class='op'>(</span><span class='fu'><a href='https://here.r-lib.org/reference/here.html'>here</a></span><span class='op'>(</span><span class='st'>"data/kallisto.dge.rds"</span><span class='op'>)</span><span class='op'>)</span></span></code></pre>
</div>
</details>
</div>
<p>We remove features which arent expressed in any samples of our data.</p>
<p>In doing this we also check that this have not removed any of our genes of interest, which include <em>Socs1</em>, <em>Socs3</em> and <em>Bcl3</em>.
Reassuringly, these features are not removed during this step and thus have some expression in the data.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>genes_of_interest</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Socs1"</span>, <span class='st'>"Socs3"</span>, <span class='st'>"Bcl3"</span><span class='op'>)</span></span>
<span><span class='co'># genes of interest are in our rownames originally</span></span>
<span><span class='va'>genes_of_interest</span> <span class='op'><a href='https://rdrr.io/r/base/match.html'>%in%</a></span> <span class='va'>dge</span><span class='op'>$</span><span class='va'>genes</span><span class='op'>$</span><span class='va'>SYMBOL</span></span></code></pre>
</div>
</details>
<pre><code>[1] TRUE TRUE TRUE</code></pre>
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># Remove genes expressed in no samples. </span></span>
<span><span class='va'>dge_zeroes</span> <span class='op'>&lt;-</span> <span class='va'>dge</span><span class='op'>[</span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colSums.html'>rowSums</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>counts</span><span class='op'>)</span> <span class='op'>==</span> <span class='fl'>0</span><span class='op'>)</span>, <span class='op'>]</span></span>
<span></span>
<span><span class='co'># See if any of our genes of interest are not being expressed</span></span>
<span><span class='va'>genes_of_interest</span> <span class='op'><a href='https://rdrr.io/r/base/match.html'>%in%</a></span> <span class='va'>dge_zeroes</span><span class='op'>$</span><span class='va'>genes</span><span class='op'>$</span><span class='va'>SYMBOL</span></span></code></pre>
</div>
</details>
<pre><code>[1] FALSE FALSE FALSE</code></pre>
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># Remove features which aren't expressed in any samples. </span></span>
<span><span class='va'>dge</span> <span class='op'>&lt;-</span> <span class='va'>dge</span><span class='op'>[</span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colSums.html'>rowSums</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>counts</span><span class='op'>)</span> <span class='op'>&gt;</span> <span class='fl'>0</span><span class='op'>)</span>, <span class='op'>]</span></span></code></pre>
</div>
</details>
</div>
<p>There are 8905 genes which arent expressed in any of the samples.
These will be removed by any filtering metric that we choose to use.
There are 27186 genes remaining for analysis.</p>
<h1 id="pre-processing">Pre-processing</h1>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># NOTE: Annoyingly, my cpm function is being masked by SingleCellExperiment. </span></span>
<span><span class='co'># Having to specify the edgeR package to get this line to work. </span></span>
<span><span class='co'># Transform from raw scale</span></span>
<span><span class='va'>cpm</span> <span class='op'>&lt;-</span> <span class='fu'>edgeR</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/cpm.html'>cpm</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>)</span></span>
<span><span class='va'>lcpm</span> <span class='op'>&lt;-</span> <span class='fu'>edgeR</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/cpm.html'>cpm</a></span><span class='op'>(</span><span class='va'>dge</span>, log<span class='op'>=</span><span class='cn'>TRUE</span><span class='op'>)</span></span></code></pre>
</div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>keep.exprs</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/filterByExpr.html'>filterByExpr</a></span><span class='op'>(</span><span class='va'>dge</span>, group<span class='op'>=</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span><span class='op'>)</span></span>
<span><span class='va'>dge</span> <span class='op'>&lt;-</span> <span class='va'>dge</span><span class='op'>[</span><span class='va'>keep.exprs</span>,, keep.lib.sizes<span class='op'>=</span><span class='cn'>FALSE</span><span class='op'>]</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/dim.html'>dim</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>)</span></span>
<span><span class='co'># This function keeps genes with about 10 read counts or more in a minimum</span></span>
<span><span class='co'># number of samples.</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/sum.html'>sum</a></span><span class='op'>(</span><span class='va'>keep.exprs</span> <span class='op'>==</span> <span class='cn'>FALSE</span><span class='op'>)</span></span></code></pre>
</div>
</details>
</div>
<p>Filtering using <code>FilterByExpr</code> from the <code>edgeR</code> package removes genes with very low expression.
There are 5414 genes which are removed due to having low expression.
There are 21772 genes remaining for analysis.</p>
<p>These low quality genes result in a spike near 0 in the left-hand side of the plot below.
In the filtered data, the lowly expressed genes are mostly removed.
The samples have a similar distribution of genes in this plot.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># Figure 1: Density of the raw post filtered data</span></span>
<span></span>
<span><span class='co'># M is the mean, L is the median</span></span>
<span><span class='va'>L</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/mean.html'>mean</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>lib.size</span><span class='op'>)</span> <span class='op'>*</span> <span class='fl'>1e-6</span></span>
<span><span class='va'>M</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/median.html'>median</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>lib.size</span><span class='op'>)</span> <span class='op'>*</span> <span class='fl'>1e-6</span></span>
<span></span>
<span><span class='va'>lcpm.cutoff</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/Log.html'>log2</a></span><span class='op'>(</span><span class='fl'>10</span><span class='op'>/</span><span class='va'>M</span> <span class='op'>+</span> <span class='fl'>2</span><span class='op'>/</span><span class='va'>L</span><span class='op'>)</span></span>
<span><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'>RColorBrewer</span><span class='op'>)</span></span>
<span><span class='va'>nsamples</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/unique.html'>unique</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span><span class='op'>)</span><span class='op'>)</span></span>
<span><span class='va'>col</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html'>brewer.pal</a></span><span class='op'>(</span><span class='va'>nsamples</span>, <span class='st'>"Paired"</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/par.html'>par</a></span><span class='op'>(</span>mfrow<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>,<span class='fl'>2</span><span class='op'>)</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/plot.default.html'>plot</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/stats/density.html'>density</a></span><span class='op'>(</span><span class='va'>lcpm</span><span class='op'>[</span>,<span class='fl'>1</span><span class='op'>]</span><span class='op'>)</span>, col<span class='op'>=</span><span class='va'>col</span><span class='op'>[</span><span class='fl'>1</span><span class='op'>]</span>, lwd<span class='op'>=</span><span class='fl'>2</span>, ylim<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>0</span>,<span class='fl'>0.26</span><span class='op'>)</span>, las<span class='op'>=</span><span class='fl'>2</span>, main<span class='op'>=</span><span class='st'>""</span>, xlab<span class='op'>=</span><span class='st'>""</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/title.html'>title</a></span><span class='op'>(</span>main<span class='op'>=</span><span class='st'>"A. Raw data"</span>, xlab<span class='op'>=</span><span class='st'>"Log-cpm"</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/abline.html'>abline</a></span><span class='op'>(</span>v<span class='op'>=</span><span class='va'>lcpm.cutoff</span>, lty<span class='op'>=</span><span class='fl'>3</span><span class='op'>)</span></span>
<span><span class='kw'>for</span> <span class='op'>(</span><span class='va'>i</span> <span class='kw'>in</span> <span class='fl'>2</span><span class='op'>:</span><span class='va'>nsamples</span><span class='op'>)</span><span class='op'>{</span></span>
<span>  <span class='va'>den</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/density.html'>density</a></span><span class='op'>(</span><span class='va'>lcpm</span><span class='op'>[</span>,<span class='va'>i</span><span class='op'>]</span><span class='op'>)</span></span>
<span>  <span class='fu'><a href='https://rdrr.io/r/graphics/lines.html'>lines</a></span><span class='op'>(</span><span class='va'>den</span><span class='op'>$</span><span class='va'>x</span>, <span class='va'>den</span><span class='op'>$</span><span class='va'>y</span>, col<span class='op'>=</span><span class='va'>col</span><span class='op'>[</span><span class='va'>i</span><span class='op'>]</span>, lwd<span class='op'>=</span><span class='fl'>2</span><span class='op'>)</span></span>
<span><span class='op'>}</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/legend.html'>legend</a></span><span class='op'>(</span><span class='st'>"topright"</span>, <span class='fu'><a href='https://rdrr.io/r/base/character.html'>as.character</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/unique.html'>unique</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span><span class='op'>)</span><span class='op'>)</span>, text.col<span class='op'>=</span><span class='va'>col</span>, bty<span class='op'>=</span><span class='st'>"n"</span><span class='op'>)</span></span>
<span><span class='va'>lcpm</span> <span class='op'>&lt;-</span> <span class='fu'>edgeR</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/cpm.html'>cpm</a></span><span class='op'>(</span><span class='va'>dge</span>, log<span class='op'>=</span><span class='cn'>TRUE</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/plot.default.html'>plot</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/stats/density.html'>density</a></span><span class='op'>(</span><span class='va'>lcpm</span><span class='op'>[</span>,<span class='fl'>1</span><span class='op'>]</span><span class='op'>)</span>, col<span class='op'>=</span><span class='va'>col</span><span class='op'>[</span><span class='fl'>1</span><span class='op'>]</span>, lwd<span class='op'>=</span><span class='fl'>2</span>, ylim<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>0</span>,<span class='fl'>0.26</span><span class='op'>)</span>, las<span class='op'>=</span><span class='fl'>2</span>, main<span class='op'>=</span><span class='st'>""</span>, xlab<span class='op'>=</span><span class='st'>""</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/title.html'>title</a></span><span class='op'>(</span>main<span class='op'>=</span><span class='st'>"B. Filtered data"</span>, xlab<span class='op'>=</span><span class='st'>"Log-cpm"</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/abline.html'>abline</a></span><span class='op'>(</span>v<span class='op'>=</span><span class='va'>lcpm.cutoff</span>, lty<span class='op'>=</span><span class='fl'>3</span><span class='op'>)</span></span>
<span><span class='kw'>for</span> <span class='op'>(</span><span class='va'>i</span> <span class='kw'>in</span> <span class='fl'>2</span><span class='op'>:</span><span class='va'>nsamples</span><span class='op'>)</span><span class='op'>{</span></span>
<span>  <span class='va'>den</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/density.html'>density</a></span><span class='op'>(</span><span class='va'>lcpm</span><span class='op'>[</span>,<span class='va'>i</span><span class='op'>]</span><span class='op'>)</span></span>
<span>  <span class='fu'><a href='https://rdrr.io/r/graphics/lines.html'>lines</a></span><span class='op'>(</span><span class='va'>den</span><span class='op'>$</span><span class='va'>x</span>, <span class='va'>den</span><span class='op'>$</span><span class='va'>y</span>, col<span class='op'>=</span><span class='va'>col</span><span class='op'>[</span><span class='va'>i</span><span class='op'>]</span>, lwd<span class='op'>=</span><span class='fl'>2</span><span class='op'>)</span></span>
<span><span class='op'>}</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/legend.html'>legend</a></span><span class='op'>(</span><span class='st'>"topright"</span>, <span class='fu'><a href='https://rdrr.io/r/base/character.html'>as.character</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/unique.html'>unique</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span><span class='op'>)</span><span class='op'>)</span>, text.col<span class='op'>=</span><span class='va'>col</span>, bty<span class='op'>=</span><span class='st'>"n"</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<p><img src="Kallisto_analysis_files/figure-html5/unnamed-chunk-17-1.png" width="624" /></p>
</div>
<h2 id="quality-control">Quality control</h2>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># Set colours for this data set</span></span>
<span><span class='va'>group_colours</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/setNames.html'>setNames</a></span><span class='op'>(</span></span>
<span>    <span class='fu'>ggthemes</span><span class='fu'>::</span><span class='fu'><a href='http://jrnold.github.io/ggthemes/reference/tableau_color_pal.html'>tableau_color_pal</a></span><span class='op'>(</span><span class='st'>"Tableau 10"</span><span class='op'>)</span><span class='op'>(</span><span class='fl'>6</span><span class='op'>)</span>, </span>
<span>    <span class='fu'><a href='https://rdrr.io/r/base/levels.html'>levels</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span><span class='op'>)</span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>batch_colours</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/setNames.html'>setNames</a></span><span class='op'>(</span></span>
<span>    <span class='fu'><a href='https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html'>brewer.pal</a></span><span class='op'>(</span><span class='fl'>8</span>, <span class='st'>"Dark2"</span><span class='op'>)</span><span class='op'>[</span><span class='fl'>3</span><span class='op'>:</span><span class='fl'>4</span><span class='op'>]</span>, </span>
<span>    <span class='fu'><a href='https://rdrr.io/r/base/levels.html'>levels</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>batch</span><span class='op'>)</span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>treatment_colours</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/setNames.html'>setNames</a></span><span class='op'>(</span></span>
<span>    <span class='fu'><a href='https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html'>brewer.pal</a></span><span class='op'>(</span><span class='fl'>8</span>, <span class='st'>"Dark2"</span><span class='op'>)</span><span class='op'>[</span><span class='fl'>1</span><span class='op'>:</span><span class='fl'>2</span><span class='op'>]</span>, </span>
<span>    <span class='fu'><a href='https://rdrr.io/r/base/levels.html'>levels</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>treatment</span><span class='op'>)</span><span class='op'>)</span></span></code></pre>
</div>
</details>
</div>
<h3 id="library-size">Library size</h3>
<p>The bar charts below show the library sizes for each sample, coloured by sample, batch and treatment.
The samples from batch 2 have smaller library sizes on average than the batch 1 samples.
The sample <code>WT_2_B1</code> from batch 1 also has a small library size.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># this factor ordering for group is helpful for QC, as it puts everything from </span></span>
<span><span class='co'># the same batch next to each other. </span></span>
<span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span>,</span>
<span>    levels <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"WT_2_B1"</span>, <span class='st'>"WT_3_B1"</span>, <span class='st'>"SOCS3_2_B1"</span>, <span class='st'>"SOCS3_3_B1"</span>, <span class='st'>"WT_1_B2"</span>, <span class='st'>"SOCS3_1_B2"</span><span class='op'>)</span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>p1</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span>data <span class='op'>=</span> <span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span>, mapping <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>y <span class='op'>=</span> <span class='va'>group</span>, x <span class='op'>=</span> <span class='va'>lib.size</span>, fill <span class='op'>=</span> <span class='va'>group</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> </span>
<span>    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_bar.html'>geom_col</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_fill_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>group_colours</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_discrete.html'>scale_y_discrete</a></span><span class='op'>(</span>limits <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/rev.html'>rev</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/levels.html'>levels</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggtheme.html'>theme_minimal</a></span><span class='op'>(</span><span class='op'>)</span></span>
<span><span class='va'>p2</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span>data <span class='op'>=</span> <span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span>, mapping <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>y <span class='op'>=</span> <span class='va'>group</span>, x <span class='op'>=</span> <span class='va'>lib.size</span>, fill <span class='op'>=</span> <span class='va'>batch</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> </span>
<span>    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_bar.html'>geom_col</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_fill_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>batch_colours</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_discrete.html'>scale_y_discrete</a></span><span class='op'>(</span>limits <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/rev.html'>rev</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/levels.html'>levels</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggtheme.html'>theme_minimal</a></span><span class='op'>(</span><span class='op'>)</span></span>
<span><span class='va'>p3</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span>data <span class='op'>=</span> <span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span>, mapping <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>y <span class='op'>=</span> <span class='va'>group</span>, x <span class='op'>=</span> <span class='va'>lib.size</span>, fill <span class='op'>=</span> <span class='va'>treatment</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> </span>
<span>    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_bar.html'>geom_col</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_fill_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>treatment_colours</span><span class='op'>)</span> <span class='op'>+</span> </span>
<span>    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_discrete.html'>scale_y_discrete</a></span><span class='op'>(</span>limits <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/rev.html'>rev</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/levels.html'>levels</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggtheme.html'>theme_minimal</a></span><span class='op'>(</span><span class='op'>)</span></span>
<span><span class='op'>(</span><span class='va'>p1</span> <span class='op'>/</span> <span class='va'>p2</span> <span class='op'>/</span> <span class='va'>p3</span><span class='op'>)</span> <span class='op'>+</span> <span class='fu'><a href='https://patchwork.data-imaginist.com/reference/plot_layout.html'>plot_layout</a></span><span class='op'>(</span>guides <span class='op'>=</span> <span class='st'>"collect"</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<p><img src="Kallisto_analysis_files/figure-html5/unnamed-chunk-19-1.png" width="624" /></p>
</div>
<p>We can also look at some other experimental variables that could explain the different library sizes.
The bar charts below are coloured by the number of mice used, and the concentration of RNA.
Since the batch 2 samples only had one mice each, this could explain the lower library size for these samples.
The concentration also has a moderate correlation with the library size.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='fu'><a href='https://rdrr.io/r/base/stopifnot.html'>stopifnot</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/identical.html'>identical</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>)</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"WT_1_B2"</span>, <span class='st'>"SOCS3_1_B2"</span>, <span class='st'>"SOCS3_2_B1"</span>, <span class='st'>"SOCS3_3_B1"</span>, <span class='st'>"WT_2_B1"</span>, <span class='st'>"WT_3_B1"</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span></span>
<span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>nMice</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='fl'>1</span>, <span class='fl'>3</span>, <span class='fl'>2</span>, <span class='fl'>3</span>, <span class='fl'>2</span><span class='op'>)</span><span class='op'>)</span></span></code></pre>
</div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>p1</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span>data <span class='op'>=</span> <span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span>, mapping <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>y <span class='op'>=</span> <span class='va'>group</span>, x <span class='op'>=</span> <span class='va'>lib.size</span>, fill <span class='op'>=</span> <span class='va'>group</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> </span>
<span>    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_bar.html'>geom_col</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_fill_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>group_colours</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_discrete.html'>scale_y_discrete</a></span><span class='op'>(</span>limits <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/rev.html'>rev</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/levels.html'>levels</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggtheme.html'>theme_minimal</a></span><span class='op'>(</span><span class='op'>)</span></span>
<span><span class='va'>p2</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span>data <span class='op'>=</span> <span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span>, mapping <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>y <span class='op'>=</span> <span class='va'>group</span>, x <span class='op'>=</span> <span class='va'>lib.size</span>, fill <span class='op'>=</span> <span class='va'>nMice</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> </span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_bar.html'>geom_col</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_discrete.html'>scale_y_discrete</a></span><span class='op'>(</span>limits <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/rev.html'>rev</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/levels.html'>levels</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_fill_manual</a></span><span class='op'>(</span></span>
<span>    values <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"1"</span> <span class='op'>=</span> <span class='st'>"#6a51a3"</span>, <span class='st'>"2"</span> <span class='op'>=</span> <span class='st'>"#c09be6"</span>, <span class='st'>"3"</span> <span class='op'>=</span> <span class='st'>"#f2e5ff"</span><span class='op'>)</span>,</span>
<span>    name <span class='op'>=</span> <span class='st'>"Number of Mice"</span></span>
<span>  <span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggtheme.html'>theme_minimal</a></span><span class='op'>(</span><span class='op'>)</span></span>
<span><span class='va'>p3</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span>data <span class='op'>=</span> <span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span>, mapping <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>y <span class='op'>=</span> <span class='va'>group</span>, x <span class='op'>=</span> <span class='va'>lib.size</span>, fill <span class='op'>=</span> <span class='va'>Concentration</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> </span>
<span>    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_bar.html'>geom_col</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_discrete.html'>scale_y_discrete</a></span><span class='op'>(</span>limits <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/rev.html'>rev</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/levels.html'>levels</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggtheme.html'>theme_minimal</a></span><span class='op'>(</span><span class='op'>)</span></span>
<span><span class='op'>(</span><span class='va'>p1</span> <span class='op'>/</span> <span class='va'>p2</span> <span class='op'>/</span> <span class='va'>p3</span><span class='op'>)</span> <span class='op'>+</span> <span class='fu'><a href='https://patchwork.data-imaginist.com/reference/plot_layout.html'>plot_layout</a></span><span class='op'>(</span>guides <span class='op'>=</span> <span class='st'>"collect"</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<p><img src="Kallisto_analysis_files/figure-html5/unnamed-chunk-22-1.png" width="624" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://rpkgs.datanovia.com/ggpubr/'>ggpubr</a></span><span class='op'>)</span></span>
<span></span>
<span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span>, <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>Concentration</span>, y <span class='op'>=</span> <span class='va'>lib.size</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://rpkgs.datanovia.com/ggpubr/reference/stat_cor.html'>stat_cor</a></span><span class='op'>(</span>method <span class='op'>=</span> <span class='st'>"pearson"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggtheme.html'>theme_light</a></span><span class='op'>(</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<p><img src="Kallisto_analysis_files/figure-html5/unnamed-chunk-24-1.png" width="624" /></p>
</div>
<h3 id="number-of-expressed-features">Number of expressed features</h3>
<p>Expressed features are defined as those with at least one count in a sample.
There is a similar number of expressed features in each sample.
Each sample has around 20,000 genes with at least one count.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>nFeatures</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/apply.html'>apply</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>counts</span>, MARGIN <span class='op'>=</span> <span class='fl'>2</span>, FUN <span class='op'>=</span> <span class='kw'>function</span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span> <span class='op'>{</span><span class='fu'><a href='https://rdrr.io/r/base/sum.html'>sum</a></span><span class='op'>(</span><span class='va'>x</span> <span class='op'>&gt;</span> <span class='fl'>0</span><span class='op'>)</span><span class='op'>}</span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span> <span class='op'>|&gt;</span></span>
<span>    <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/select.html'>select</a></span><span class='op'>(</span><span class='va'>group</span>, <span class='va'>nFeatures</span><span class='op'>)</span> <span class='op'>|&gt;</span></span>
<span>    <span class='fu'>knitr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/knitr/man/kable.html'>kable</a></span><span class='op'>(</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<table>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: left;">group</th>
<th style="text-align: right;">nFeatures</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">WT_1_B2</td>
<td style="text-align: left;">WT_1_B2</td>
<td style="text-align: right;">21414</td>
</tr>
<tr class="even">
<td style="text-align: left;">SOCS3_1_B2</td>
<td style="text-align: left;">SOCS3_1_B2</td>
<td style="text-align: right;">21375</td>
</tr>
<tr class="odd">
<td style="text-align: left;">SOCS3_2_B1</td>
<td style="text-align: left;">SOCS3_2_B1</td>
<td style="text-align: right;">20939</td>
</tr>
<tr class="even">
<td style="text-align: left;">SOCS3_3_B1</td>
<td style="text-align: left;">SOCS3_3_B1</td>
<td style="text-align: right;">21250</td>
</tr>
<tr class="odd">
<td style="text-align: left;">WT_2_B1</td>
<td style="text-align: left;">WT_2_B1</td>
<td style="text-align: right;">21272</td>
</tr>
<tr class="even">
<td style="text-align: left;">WT_3_B1</td>
<td style="text-align: left;">WT_3_B1</td>
<td style="text-align: right;">20876</td>
</tr>
</tbody>
</table>
</div>
<h3 id="proportion-of-mitochondrial-genes">Proportion of mitochondrial genes</h3>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='fu'><a href='https://rdrr.io/r/base/table.html'>table</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>genes</span><span class='op'>$</span><span class='va'>CHR</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<pre><code>
         1         10         11         12         13         14 
      1302       1027       1621        886        876        873 
        15         16         17         18         19          2 
       804        678       1088        539        694       1715 
         3          4          5          6          7          8 
      1012       1281       1263       1260       1772       1022 
         9 GL456210.1 GL456211.1 GL456212.1 GL456221.1 JH584304.1 
      1202          3          3          2          5          1 
        MT          X          Y 
        13        793         37 </code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>mito_set</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>genes</span><span class='op'>)</span><span class='op'>[</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>genes</span><span class='op'>$</span><span class='va'>CHR</span> <span class='op'>==</span> <span class='st'>"MT"</span><span class='op'>]</span></span></code></pre>
</div>
</details>
</div>
<p>We can identify mitochondrial genes as genes with the chromosome number MT.
There are 13 mitochondrial genes.</p>
<p>The chunk below works out the proportion of counts in each sample which are mitochondrial.
High proportions of mitochondrial RNA can indicate that the cells in the sample are under stress.
In this case, the proportion is very low (0.5-1.5%), which is good.
There is however noticeably higher percentages in the batch2 data.
There is no trend between treatment and mitochondrial proportion, which is good.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># Apply to each sample</span></span>
<span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>mito_pct</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/apply.html'>apply</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>counts</span>, MARGIN <span class='op'>=</span> <span class='fl'>2</span>, FUN <span class='op'>=</span> <span class='kw'>function</span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span> <span class='op'>{</span><span class='fu'><a href='https://rdrr.io/r/base/sum.html'>sum</a></span><span class='op'>(</span><span class='va'>x</span><span class='op'>[</span><span class='va'>mito_set</span><span class='op'>]</span><span class='op'>)</span> <span class='op'>/</span> <span class='fu'><a href='https://rdrr.io/r/base/sum.html'>sum</a></span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span><span class='op'>}</span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span> <span class='op'>|&gt;</span></span>
<span>    <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/select.html'>select</a></span><span class='op'>(</span><span class='va'>group</span>, <span class='va'>mito_pct</span><span class='op'>)</span> <span class='op'>|&gt;</span></span>
<span>    <span class='fu'>janitor</span><span class='fu'>::</span><span class='fu'><a href='https://sfirke.github.io/janitor/reference/adorn_pct_formatting.html'>adorn_pct_formatting</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>|&gt;</span></span>
<span>    <span class='fu'>knitr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/knitr/man/kable.html'>kable</a></span><span class='op'>(</span><span class='op'>)</span> </span></code></pre>
</div>
</details>
<table>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: left;">group</th>
<th style="text-align: left;">mito_pct</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">WT_1_B2</td>
<td style="text-align: left;">WT_1_B2</td>
<td style="text-align: left;">1.1%</td>
</tr>
<tr class="even">
<td style="text-align: left;">SOCS3_1_B2</td>
<td style="text-align: left;">SOCS3_1_B2</td>
<td style="text-align: left;">1.2%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">SOCS3_2_B1</td>
<td style="text-align: left;">SOCS3_2_B1</td>
<td style="text-align: left;">0.7%</td>
</tr>
<tr class="even">
<td style="text-align: left;">SOCS3_3_B1</td>
<td style="text-align: left;">SOCS3_3_B1</td>
<td style="text-align: left;">0.6%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">WT_2_B1</td>
<td style="text-align: left;">WT_2_B1</td>
<td style="text-align: left;">0.6%</td>
</tr>
<tr class="even">
<td style="text-align: left;">WT_3_B1</td>
<td style="text-align: left;">WT_3_B1</td>
<td style="text-align: left;">0.5%</td>
</tr>
</tbody>
</table>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://scales.r-lib.org'>scales</a></span><span class='op'>)</span></span>
<span></span>
<span><span class='co'># No trends here</span></span>
<span><span class='va'>p1</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span>data <span class='op'>=</span> <span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span>, mapping <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>y <span class='op'>=</span> <span class='va'>group</span>, x <span class='op'>=</span> <span class='va'>mito_pct</span>, fill <span class='op'>=</span> <span class='va'>group</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> </span>
<span>    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_bar.html'>geom_col</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_fill_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>group_colours</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_continuous.html'>scale_x_continuous</a></span><span class='op'>(</span>labels <span class='op'>=</span> <span class='fu'><a href='https://scales.r-lib.org/reference/label_percent.html'>label_percent</a></span><span class='op'>(</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_discrete.html'>scale_y_discrete</a></span><span class='op'>(</span>limits <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/rev.html'>rev</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/levels.html'>levels</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggtheme.html'>theme_minimal</a></span><span class='op'>(</span><span class='op'>)</span></span>
<span><span class='va'>p2</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span>data <span class='op'>=</span> <span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span>, mapping <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>y <span class='op'>=</span> <span class='va'>group</span>, x <span class='op'>=</span> <span class='va'>mito_pct</span>, fill <span class='op'>=</span> <span class='va'>batch</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> </span>
<span>    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_bar.html'>geom_col</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_fill_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>batch_colours</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_continuous.html'>scale_x_continuous</a></span><span class='op'>(</span>labels <span class='op'>=</span> <span class='fu'><a href='https://scales.r-lib.org/reference/label_percent.html'>label_percent</a></span><span class='op'>(</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_discrete.html'>scale_y_discrete</a></span><span class='op'>(</span>limits <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/rev.html'>rev</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/levels.html'>levels</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggtheme.html'>theme_minimal</a></span><span class='op'>(</span><span class='op'>)</span></span>
<span><span class='va'>p3</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span>data <span class='op'>=</span> <span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span>, mapping <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>y <span class='op'>=</span> <span class='va'>group</span>, x <span class='op'>=</span> <span class='va'>mito_pct</span>, fill <span class='op'>=</span> <span class='va'>treatment</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> </span>
<span>    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_bar.html'>geom_col</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_fill_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>treatment_colours</span><span class='op'>)</span> <span class='op'>+</span> </span>
<span>    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_continuous.html'>scale_x_continuous</a></span><span class='op'>(</span>labels <span class='op'>=</span> <span class='fu'><a href='https://scales.r-lib.org/reference/label_percent.html'>label_percent</a></span><span class='op'>(</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_discrete.html'>scale_y_discrete</a></span><span class='op'>(</span>limits <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/rev.html'>rev</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/levels.html'>levels</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggtheme.html'>theme_minimal</a></span><span class='op'>(</span><span class='op'>)</span></span>
<span><span class='op'>(</span><span class='va'>p1</span> <span class='op'>/</span> <span class='va'>p2</span> <span class='op'>/</span> <span class='va'>p3</span><span class='op'>)</span> <span class='op'>+</span> <span class='fu'><a href='https://patchwork.data-imaginist.com/reference/plot_layout.html'>plot_layout</a></span><span class='op'>(</span>guides <span class='op'>=</span> <span class='st'>"collect"</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<p><img src="Kallisto_analysis_files/figure-html5/unnamed-chunk-31-1.png" width="624" /></p>
</div>
<h2 id="normalisation">Normalisation</h2>
<p>The impact of normalisation is subtle.
Most notably, the third quartiles have been brought into alignment.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>dge_raw</span> <span class='op'>&lt;-</span> <span class='va'>dge</span></span>
<span><span class='va'>dge</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/calcNormFactors.html'>calcNormFactors</a></span><span class='op'>(</span><span class='va'>dge</span>, method <span class='op'>=</span> <span class='st'>"TMM"</span><span class='op'>)</span></span></code></pre>
</div>
</details>
</div>
<p>The library sizes for the samples are listed in the table below.
This is the sum of counts across every gene for each sample.
The counts in the Hisat2 analysis were roughly 10 million per sample, and the kallisto analysis has about 60 million counts per sample.
Possibly this is due to the way each method handles PCR duplicates as well as some reads not mapping in the hisat2 version.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># Library sizes</span></span>
<span><span class='va'>dge_raw</span><span class='op'>$</span><span class='va'>sample</span> <span class='op'>|&gt;</span> </span>
<span>    <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/select.html'>select</a></span><span class='op'>(</span><span class='va'>lib.size</span><span class='op'>)</span> <span class='op'>|&gt;</span></span>
<span>    <span class='fu'>knitr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/knitr/man/kable.html'>kable</a></span><span class='op'>(</span>col.names <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Sample"</span>, <span class='st'>"Library size"</span><span class='op'>)</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<table>
<thead>
<tr class="header">
<th style="text-align: left;">Sample</th>
<th style="text-align: right;">Library size</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">WT_1_B2</td>
<td style="text-align: right;">59608523</td>
</tr>
<tr class="even">
<td style="text-align: left;">SOCS3_1_B2</td>
<td style="text-align: right;">58669677</td>
</tr>
<tr class="odd">
<td style="text-align: left;">SOCS3_2_B1</td>
<td style="text-align: right;">85103114</td>
</tr>
<tr class="even">
<td style="text-align: left;">SOCS3_3_B1</td>
<td style="text-align: right;">67361444</td>
</tr>
<tr class="odd">
<td style="text-align: left;">WT_2_B1</td>
<td style="text-align: right;">61145081</td>
</tr>
<tr class="even">
<td style="text-align: left;">WT_3_B1</td>
<td style="text-align: right;">74887905</td>
</tr>
</tbody>
</table>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># log-cpm in appropriate order for plot</span></span>
<span><span class='co'># NOTE: NEVER USE WITH dge$samples!!! For plotting only!!</span></span>
<span><span class='va'>lcpm_raw_plot</span> <span class='op'>&lt;-</span> <span class='fu'>edgeR</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/cpm.html'>cpm</a></span><span class='op'>(</span><span class='va'>dge_raw</span>, log<span class='op'>=</span><span class='cn'>TRUE</span><span class='op'>)</span><span class='op'>[</span>, <span class='fu'><a href='https://rdrr.io/r/base/levels.html'>levels</a></span><span class='op'>(</span><span class='va'>dge_raw</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span><span class='op'>)</span><span class='op'>]</span></span>
<span><span class='va'>lcpm_plot</span> <span class='op'>&lt;-</span> <span class='fu'>edgeR</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/cpm.html'>cpm</a></span><span class='op'>(</span><span class='va'>dge</span>, log <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span><span class='op'>[</span>, <span class='fu'><a href='https://rdrr.io/r/base/levels.html'>levels</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span><span class='op'>)</span><span class='op'>]</span></span>
<span></span>
<span><span class='co'># y-limit for plot </span></span>
<span><span class='va'>ylims</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/range.html'>range</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='va'>lcpm_raw_plot</span>, <span class='va'>lcpm_plot</span><span class='op'>)</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/par.html'>par</a></span><span class='op'>(</span>mfrow<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>,<span class='fl'>2</span><span class='op'>)</span><span class='op'>)</span></span>
<span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/boxplot.html'>boxplot</a></span><span class='op'>(</span><span class='va'>lcpm_raw_plot</span>, las<span class='op'>=</span><span class='fl'>2</span>, col<span class='op'>=</span><span class='va'>col</span>, main<span class='op'>=</span><span class='st'>""</span>, <span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/title.html'>title</a></span><span class='op'>(</span>main<span class='op'>=</span><span class='st'>"A. Example: Unnormalised data"</span>, ylab<span class='op'>=</span><span class='st'>"Log-cpm"</span><span class='op'>)</span></span>
<span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/boxplot.html'>boxplot</a></span><span class='op'>(</span><span class='va'>lcpm_plot</span>, las<span class='op'>=</span><span class='fl'>2</span>, col<span class='op'>=</span><span class='va'>col</span>, main<span class='op'>=</span><span class='st'>""</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/title.html'>title</a></span><span class='op'>(</span>main<span class='op'>=</span><span class='st'>"B. Example: Normalised data"</span>, ylab<span class='op'>=</span><span class='st'>"Log-cpm"</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<p><img src="Kallisto_analysis_files/figure-html5/unnamed-chunk-35-1.png" width="624" /></p>
</div>
<p>The plots below show the MDS plot before and after normalisation.
This has a more obvious difference than the boxplot above.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='fu'><a href='https://rdrr.io/r/graphics/par.html'>par</a></span><span class='op'>(</span>mfrow <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>,<span class='fl'>2</span><span class='op'>)</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/pkg/limma/man/plotMDS.html'>plotMDS</a></span><span class='op'>(</span><span class='va'>dge_raw</span>, main<span class='op'>=</span><span class='st'>"Before normalization"</span>, col<span class='op'>=</span><span class='va'>batch_colours</span><span class='op'>[</span><span class='va'>dge_raw</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>batch</span><span class='op'>]</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/pkg/limma/man/plotMDS.html'>plotMDS</a></span><span class='op'>(</span><span class='va'>dge</span>, main<span class='op'>=</span><span class='st'>"After normalization"</span>, col<span class='op'>=</span><span class='va'>batch_colours</span><span class='op'>[</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>batch</span><span class='op'>]</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<p><img src="Kallisto_analysis_files/figure-html5/unnamed-chunk-36-1.png" width="624" /></p>
</div>
<p>After normalising the data, we need to evaluate the log counts again.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>cpm</span> <span class='op'>&lt;-</span> <span class='fu'>edgeR</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/cpm.html'>cpm</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>)</span></span>
<span><span class='va'>lcpm</span> <span class='op'>&lt;-</span> <span class='fu'>edgeR</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/cpm.html'>cpm</a></span><span class='op'>(</span><span class='va'>dge</span>, log<span class='op'>=</span><span class='cn'>TRUE</span><span class='op'>)</span></span></code></pre>
</div>
</details>
</div>
<h1 id="visualising-the-data">Visualising the data</h1>
<p>MDS plot with the library size overlayed on the samples shows that the samples from batch 2 have lower library sizes than the samples from batch 1.
The sample <code>WT_2_B1</code> clusters in the centre of the data which may indicate that it has different expression profiles to both the WT and SOCS3 samples. It also has a low library size.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='fu'><a href='https://rdrr.io/r/graphics/par.html'>par</a></span><span class='op'>(</span>mfrow <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>2</span>,<span class='fl'>2</span><span class='op'>)</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/pkg/limma/man/plotMDS.html'>plotMDS</a></span><span class='op'>(</span><span class='va'>lcpm</span>, labels<span class='op'>=</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span>, col<span class='op'>=</span><span class='va'>group_colours</span><span class='op'>[</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span><span class='op'>]</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/title.html'>title</a></span><span class='op'>(</span>main<span class='op'>=</span><span class='st'>"A. Sample groups"</span><span class='op'>)</span></span>
<span></span>
<span><span class='fu'><a href='https://rdrr.io/pkg/limma/man/plotMDS.html'>plotMDS</a></span><span class='op'>(</span><span class='va'>lcpm</span>, labels<span class='op'>=</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>treatment</span>, col<span class='op'>=</span><span class='va'>treatment_colours</span><span class='op'>[</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>treatment</span><span class='op'>]</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/title.html'>title</a></span><span class='op'>(</span>main<span class='op'>=</span><span class='st'>"B. Treatment"</span><span class='op'>)</span></span>
<span></span>
<span><span class='fu'><a href='https://rdrr.io/pkg/limma/man/plotMDS.html'>plotMDS</a></span><span class='op'>(</span><span class='va'>lcpm</span>, labels<span class='op'>=</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>batch</span>, col<span class='op'>=</span><span class='va'>batch_colours</span><span class='op'>[</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>batch</span><span class='op'>]</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/title.html'>title</a></span><span class='op'>(</span>main<span class='op'>=</span><span class='st'>"C. Batch"</span><span class='op'>)</span></span>
<span></span>
<span><span class='co'># MDS plot with library size</span></span>
<span><span class='va'>libsizes</span> <span class='op'>&lt;-</span> <span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>lib.size</span></span>
<span></span>
<span><span class='co'># Create a viridis color function</span></span>
<span><span class='va'>color_func</span> <span class='op'>&lt;-</span> <span class='fu'>scales</span><span class='fu'>::</span><span class='fu'><a href='https://scales.r-lib.org/reference/col_numeric.html'>col_numeric</a></span><span class='op'>(</span><span class='fu'>viridis</span><span class='fu'>::</span><span class='fu'><a href='https://sjmgarnier.github.io/viridisLite/reference/viridis.html'>viridis</a></span><span class='op'>(</span><span class='fl'>100</span><span class='op'>)</span>, domain <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/range.html'>range</a></span><span class='op'>(</span><span class='va'>libsizes</span><span class='op'>)</span><span class='op'>)</span></span>
<span></span>
<span><span class='co'># Assign colors according to library sizes</span></span>
<span><span class='va'>point_colors</span> <span class='op'>&lt;-</span> <span class='fu'>color_func</span><span class='op'>(</span><span class='va'>libsizes</span><span class='op'>)</span></span>
<span></span>
<span><span class='fu'><a href='https://rdrr.io/pkg/limma/man/plotMDS.html'>plotMDS</a></span><span class='op'>(</span><span class='va'>lcpm</span>, labels<span class='op'>=</span><span class='cn'>NULL</span>, pch <span class='op'>=</span> <span class='fl'>16</span>, cex <span class='op'>=</span> <span class='fl'>4</span>, col<span class='op'>=</span><span class='va'>point_colors</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/title.html'>title</a></span><span class='op'>(</span>main<span class='op'>=</span><span class='st'>"D. Library Size"</span><span class='op'>)</span> </span></code></pre>
</div>
</details>
<p><img src="Kallisto_analysis_files/figure-html5/unnamed-chunk-38-1.png" width="624" /></p>
</div>
<h2 id="comparing-batches-for-genes-of-interest">Comparing batches for genes of interest</h2>
<p>The plots below show the expression of three of our genes of interest.
Based on a priori knowledge, the SOCS3 knockout samples should express Socs1 and Socs3 less than the WT.
Bcl3 should be higher in the SOCS3 group.
The difference in these 6 samples is small and will not come out as significant in DE testing.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>genes_of_interest</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Socs3"</span>, <span class='st'>"Socs1"</span>, <span class='st'>"Bcl3"</span><span class='op'>)</span></span>
<span><span class='va'>selected_genes</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/grep.html'>grep</a></span><span class='op'>(</span><span class='fu'>glue</span><span class='fu'>::</span><span class='fu'><a href='https://glue.tidyverse.org/reference/glue_collapse.html'>glue_collapse</a></span><span class='op'>(</span><span class='va'>genes_of_interest</span>, <span class='st'>"|"</span><span class='op'>)</span>, <span class='va'>dge</span><span class='op'>$</span><span class='va'>genes</span><span class='op'>$</span><span class='va'>SYMBOL</span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>cpm</span> <span class='op'>&lt;-</span> <span class='fu'>edgeR</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/cpm.html'>cpm</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>)</span></span>
<span><span class='va'>lcpm</span> <span class='op'>&lt;-</span> <span class='fu'>edgeR</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/cpm.html'>cpm</a></span><span class='op'>(</span><span class='va'>dge</span>, log<span class='op'>=</span><span class='cn'>TRUE</span><span class='op'>)</span></span>
<span><span class='va'>df</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span></span>
<span>    y <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/t.html'>t</a></span><span class='op'>(</span><span class='va'>lcpm</span><span class='op'>[</span><span class='va'>selected_genes</span>, <span class='op'>]</span><span class='op'>)</span>, </span>
<span>    treatment <span class='op'>=</span> <span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>treatment</span>, </span>
<span>    batch <span class='op'>=</span> <span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>batch</span><span class='op'>)</span></span>
<span></span>
<span><span class='co'># NOTE: The order of the columns corresponds to the order of these genes in the </span></span>
<span><span class='co'>#   rownames, NOT the order that was listed in genes_of_interest.</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>df</span><span class='op'>)</span><span class='op'>[</span><span class='fl'>1</span><span class='op'>:</span><span class='fl'>3</span><span class='op'>]</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/grep.html'>gsub</a></span><span class='op'>(</span><span class='st'>"y."</span>, <span class='st'>""</span>, <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>df</span><span class='op'>)</span><span class='op'>[</span><span class='fl'>1</span><span class='op'>:</span><span class='fl'>3</span><span class='op'>]</span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>df</span><span class='op'>$</span><span class='va'>labels</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='va'>df</span><span class='op'>$</span><span class='va'>batch</span>, <span class='st'>"."</span>, <span class='va'>df</span><span class='op'>$</span><span class='va'>treatment</span><span class='op'>)</span>,</span>
<span>    levels<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"B1.WT"</span>, <span class='st'>"B1.SOCS3"</span>, <span class='st'>"B2.WT"</span>, <span class='st'>"B2.SOCS3"</span><span class='op'>)</span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>plot_list</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span><span class='op'>)</span></span>
<span><span class='kw'>for</span> <span class='op'>(</span><span class='va'>i</span> <span class='kw'>in</span> <span class='fl'>1</span><span class='op'>:</span><span class='fl'>3</span><span class='op'>)</span> <span class='op'>{</span></span>
<span>    <span class='va'>plot_list</span><span class='op'>[[</span><span class='va'>i</span><span class='op'>]</span><span class='op'>]</span> <span class='op'>&lt;-</span></span>
<span>        <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span>data <span class='op'>=</span> <span class='va'>df</span>, mapping <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>labels</span>, y <span class='op'>=</span> <span class='va'>.data</span><span class='op'>[[</span><span class='va'>genes_of_interest</span><span class='op'>[</span><span class='va'>i</span><span class='op'>]</span><span class='op'>]</span><span class='op'>]</span>, group <span class='op'>=</span> <span class='va'>labels</span>, colour <span class='op'>=</span> <span class='va'>treatment</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> </span>
<span>            <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span>size <span class='op'>=</span> <span class='fl'>4</span><span class='op'>)</span> <span class='op'>+</span> </span>
<span>            <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggtheme.html'>theme_minimal</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span> </span>
<span>            <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/theme.html'>theme</a></span><span class='op'>(</span>axis.text.x <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/element.html'>element_text</a></span><span class='op'>(</span>angle <span class='op'>=</span> <span class='fl'>45</span>, hjust <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> </span>
<span>            <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Expression (log CPM)"</span><span class='op'>)</span> <span class='op'>+</span> </span>
<span>            <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>treatment_colours</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>            <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ggtitle</a></span><span class='op'>(</span>label <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='va'>genes_of_interest</span><span class='op'>[</span><span class='va'>i</span><span class='op'>]</span><span class='op'>)</span></span>
<span>    <span class='op'>)</span></span>
<span><span class='op'>}</span></span>
<span><span class='fu'>patchwork</span><span class='fu'>::</span><span class='fu'><a href='https://patchwork.data-imaginist.com/reference/wrap_plots.html'>wrap_plots</a></span><span class='op'>(</span><span class='va'>plot_list</span>, ncol <span class='op'>=</span> <span class='fl'>3</span>, guides <span class='op'>=</span> <span class='st'>"collect"</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<p><img src="Kallisto_analysis_files/figure-html5/unnamed-chunk-39-1.png" width="624" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># Table format</span></span>
<span><span class='va'>lcpm</span><span class='op'>[</span><span class='va'>selected_genes</span>, <span class='op'>]</span> <span class='op'>|&gt;</span></span>
<span>    <span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>as.data.frame</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>|&gt;</span></span>
<span>    <span class='fu'>tibble</span><span class='fu'>::</span><span class='fu'><a href='https://tibble.tidyverse.org/reference/rownames.html'>rownames_to_column</a></span><span class='op'>(</span><span class='st'>"ENSEMBL"</span><span class='op'>)</span> <span class='op'>|&gt;</span></span>
<span>    <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/mutate.html'>mutate</a></span><span class='op'>(</span>SYMBOL <span class='op'>=</span> <span class='va'>dge</span><span class='op'>$</span><span class='va'>genes</span><span class='op'>$</span><span class='va'>SYMBOL</span><span class='op'>[</span><span class='va'>selected_genes</span><span class='op'>]</span>, .after <span class='op'>=</span> <span class='st'>"ENSEMBL"</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<pre><code>  ENSEMBL SYMBOL  WT_1_B2 SOCS3_1_B2 SOCS3_2_B1 SOCS3_3_B1  WT_2_B1
1   Socs1  Socs1 4.193607   3.863719   4.273091   4.255331 4.560619
2   Socs3  Socs3 7.550153   7.641535   8.646510   8.415269 8.926129
3    Bcl3   Bcl3 5.337065   5.360624   5.889254   5.586850 5.751546
   WT_3_B1
1 4.181088
2 8.492555
3 5.422480</code></pre>
</div>
<h1 id="differential-expression-analyses">Differential Expression Analyses</h1>
<p>We will be including 21772 genes in our differential expression analyses.
Due to the high number of tests being performed, it is very important to use some form of multiple testing correction such as the False Discovery Rate (FDR).
When performing tens of thousands of tests, there will be many significant genes (p-value &lt; 0.05) even if there are no real differences between the treatments.
Multiple testing correction makes the threshold for p-values stricter to account for this.</p>
<h2 id="treatment-with-limmavoom-method">Treatment with limma::voom method</h2>
<p>Since there is a strong batch effect, we will include batch as a covariate in the differential expression analysis.
The analysis below uses the <code>limma::voom()</code> method for differential expression testing.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>design</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/model.matrix.html'>model.matrix</a></span><span class='op'>(</span><span class='op'>~</span><span class='fl'>0</span><span class='op'>+</span><span class='va'>treatment</span><span class='op'>+</span><span class='va'>batch</span>, data <span class='op'>=</span> <span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>)</span></span>
<span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>design</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/grep.html'>gsub</a></span><span class='op'>(</span><span class='st'>"treatment|batch"</span>, <span class='st'>""</span>, <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>design</span><span class='op'>)</span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>contr.matrix</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/limma/man/makeContrasts.html'>makeContrasts</a></span><span class='op'>(</span></span>
<span>   SOCS3vsWT <span class='op'>=</span> <span class='va'>SOCS3</span> <span class='op'>-</span> <span class='va'>WT</span>,</span>
<span>   levels <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>design</span><span class='op'>)</span><span class='op'>)</span></span>
<span><span class='va'>contr.matrix</span></span></code></pre>
</div>
</details>
<pre><code>       Contrasts
Levels  SOCS3vsWT
  WT           -1
  SOCS3         1
  B2            0</code></pre>
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>v</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/limma/man/voom.html'>voom</a></span><span class='op'>(</span><span class='va'>dge</span>, <span class='va'>design</span>, plot<span class='op'>=</span><span class='cn'>TRUE</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<img src="Kallisto_analysis_files/figure-html5/unnamed-chunk-41-1.png" width="624" />
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>vfit</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/limma/man/lmFit.html'>lmFit</a></span><span class='op'>(</span><span class='va'>v</span>, <span class='va'>design</span><span class='op'>)</span></span>
<span><span class='va'>vfit</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/limma/man/contrasts.fit.html'>contrasts.fit</a></span><span class='op'>(</span><span class='va'>vfit</span>, contrasts<span class='op'>=</span><span class='va'>contr.matrix</span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>efit</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/limma/man/ebayes.html'>eBayes</a></span><span class='op'>(</span><span class='va'>vfit</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/pkg/limma/man/plotSA.html'>plotSA</a></span><span class='op'>(</span><span class='va'>efit</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<img src="Kallisto_analysis_files/figure-html5/unnamed-chunk-41-2.png" width="624" />
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>de_results</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/limma/man/decideTests.html'>decideTests</a></span><span class='op'>(</span><span class='va'>efit</span><span class='op'>)</span></span></code></pre>
</div>
</details>
</div>
<p>There is only one DE gene, and it is the immunoglobulin gene Igkv10-96.
The top 20 genes (ranked by multiple-testing adjusted p-value) also includes some other Ig genes.</p>
<p>These genes are related to B-cells, and often indicate contamination from immune cells.
Assess if it makes sense biologically to see less immunoglobulin expression in the Socs3 samples.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='fu'><a href='https://rdrr.io/r/base/summary.html'>summary</a></span><span class='op'>(</span><span class='va'>de_results</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<pre><code>       SOCS3vsWT
Down           1
NotSig     21771
Up             0</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># Sort the top table by adjusted P value</span></span>
<span><span class='va'>de_table</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/limma/man/toptable.html'>topTable</a></span><span class='op'>(</span><span class='va'>efit</span>, number<span class='op'>=</span><span class='cn'>Inf</span>, sort.by<span class='op'>=</span><span class='st'>"P"</span><span class='op'>)</span></span>
<span><span class='va'>sorted_de</span> <span class='op'>&lt;-</span> <span class='va'>de_table</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/order.html'>order</a></span><span class='op'>(</span><span class='va'>de_table</span><span class='op'>$</span><span class='va'>adj.P.Val</span><span class='op'>)</span>, <span class='op'>]</span></span>
<span></span>
<span><span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span><span class='op'>(</span><span class='va'>sorted_de</span>, n<span class='op'>=</span><span class='fl'>20</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<pre><code>          ensembl_transcript_id    ensembl_gene_id external_gene_name
Ighv1-53     ENSMUST00000103523 ENSMUSG00000093894           Ighv1-53
Igkv10-96    ENSMUST00000103328 ENSMUSG00000094420          Igkv10-96
Col18a1      ENSMUST00000081654 ENSMUSG00000001435            Col18a1
Pla2g2d      ENSMUST00000030528 ENSMUSG00000041202            Pla2g2d
Ighv1-82     ENSMUST00000196991 ENSMUSG00000095127           Ighv1-82
Ighv3-6      ENSMUST00000195124 ENSMUSG00000076672            Ighv3-6
Igkv4-79     ENSMUST00000103342 ENSMUSG00000076541           Igkv4-79
Ighv5-4      ENSMUST00000103444 ENSMUSG00000095612            Ighv5-4
Kera         ENSMUST00000105286 ENSMUSG00000019932               Kera
Ptx4         ENSMUST00000054930 ENSMUSG00000044172               Ptx4
Serpina1d    ENSMUST00000078869 ENSMUSG00000071177          Serpina1d
Atp6v0d2     ENSMUST00000029900 ENSMUSG00000028238           Atp6v0d2
Scx          ENSMUST00000229271 ENSMUSG00000034161                Scx
Ighv2-2      ENSMUST00000103443 ENSMUSG00000096464            Ighv2-2
Adamts3      ENSMUST00000061427 ENSMUSG00000043635            Adamts3
Itk          ENSMUST00000020664 ENSMUSG00000020395                Itk
St18         ENSMUST00000140079 ENSMUSG00000033740               St18
Spon2        ENSMUST00000200884 ENSMUSG00000037379              Spon2
Serpina1a    ENSMUST00000072876 ENSMUSG00000066366          Serpina1a
Gm16867      ENSMUST00000183917 ENSMUSG00000093954            Gm16867
          chromosome_name CHR    SYMBOL            ENSEMBL      logFC
Ighv1-53               12  12  Ighv1-53 ENSMUSG00000093894 -4.0124141
Igkv10-96               6   6 Igkv10-96 ENSMUSG00000094420 -1.9378250
Col18a1                10  10   Col18a1 ENSMUSG00000001435  0.8990599
Pla2g2d                 4   4   Pla2g2d ENSMUSG00000041202  1.3628040
Ighv1-82               12  12  Ighv1-82 ENSMUSG00000095127 -2.4104646
Ighv3-6                12  12   Ighv3-6 ENSMUSG00000076672 -2.1316224
Igkv4-79                6   6  Igkv4-79 ENSMUSG00000076541 -8.5587333
Ighv5-4                12  12   Ighv5-4 ENSMUSG00000095612 -4.6985150
Kera                   10  10      Kera ENSMUSG00000019932  1.3713546
Ptx4                   17  17      Ptx4 ENSMUSG00000044172  1.7728906
Serpina1d              12  12 Serpina1d ENSMUSG00000071177 -8.9632740
Atp6v0d2                4   4  Atp6v0d2 ENSMUSG00000028238  0.8859395
Scx                    15  15       Scx ENSMUSG00000034161  0.9467223
Ighv2-2                12  12   Ighv2-2 ENSMUSG00000096464 -1.3168779
Adamts3                 5   5   Adamts3 ENSMUSG00000043635  0.8820399
Itk                    11  11       Itk ENSMUSG00000020395 -1.0334949
St18                    1   1      St18 ENSMUSG00000033740  1.0191991
Spon2                   5   5     Spon2 ENSMUSG00000037379  0.9688598
Serpina1a              12  12 Serpina1a ENSMUSG00000066366 -7.1386943
Gm16867                14  14   Gm16867 ENSMUSG00000093954  0.7835625
            AveExpr          t      P.Value  adj.P.Val           B
Ighv1-53   4.959076 -15.353987 9.758266e-07 0.02124570  3.97176427
Igkv10-96  5.608252 -11.850155 5.842989e-06 0.06360678  3.98972269
Col18a1    4.994588   9.083638 3.517068e-05 0.24273285  2.66188572
Pla2g2d    4.331964   8.659995 4.825114e-05 0.24273285  2.16152507
Ighv1-82   4.091629  -8.069133 7.670576e-05 0.24273285  1.73742654
Ighv3-6    5.559966  -7.996783 8.134194e-05 0.24273285  1.97373795
Igkv4-79  -2.865525  -7.924391 8.629903e-05 0.24273285 -4.03276623
Ighv5-4    2.046985  -7.884291 8.919083e-05 0.24273285 -0.56749900
Kera       4.484776   7.406839 1.334934e-04 0.32293529  1.54837431
Ptx4       2.414982   6.526239 2.973713e-04 0.64743689  0.39576299
Serpina1d -1.916314  -6.311802 3.658982e-04 0.66979123 -3.89348036
Atp6v0d2   6.733194   6.301115 3.697491e-04 0.66979123  0.69001177
Scx        2.755483   6.169555 4.210637e-04 0.66979123  0.07625135
Ighv2-2    3.877833  -6.080932 4.601085e-04 0.66979123  0.33740415
Adamts3    3.354136   6.007886 4.953433e-04 0.66979123  0.16487991
Itk        4.535894  -5.997187 5.007532e-04 0.66979123  0.34735917
St18       3.434900   5.954529 5.229860e-04 0.66979123  0.09974794
Spon2      3.673332   5.868768 5.710967e-04 0.69077324  0.12662941
Serpina1a -1.348386  -5.796686 6.153696e-04 0.70514881 -3.62160400
Gm16867    4.730063   5.633636 7.303374e-04 0.79032187  0.02506500</code></pre>
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># Write to a file</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/files2.html'>dir.create</a></span><span class='op'>(</span><span class='fu'><a href='https://here.r-lib.org/reference/here.html'>here</a></span><span class='op'>(</span><span class='st'>"output/DEGs/kallisto_analysis"</span><span class='op'>)</span>, recursive <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/utils/write.table.html'>write.table</a></span><span class='op'>(</span><span class='va'>sorted_de</span>, file<span class='op'>=</span><span class='fu'><a href='https://here.r-lib.org/reference/here.html'>here</a></span><span class='op'>(</span><span class='st'>"output/DEGs/kallisto_analysis/kallisto_treatment_voom.tsv"</span><span class='op'>)</span>,</span>
<span>            sep<span class='op'>=</span><span class='st'>"\t"</span>, quote<span class='op'>=</span><span class='cn'>FALSE</span>, row.names<span class='op'>=</span><span class='cn'>FALSE</span><span class='op'>)</span></span></code></pre>
</div>
</details>
</div>
<p>A table of the differential expression results are saved in <code>output/DEGs/kallisto_analysis/kallisto_treatment_voom.tsv</code>.
The rows are sorted so the genes with the lowest adjusted p-value (FDR) are at the top.</p>
<p>The results of testing for our genes of interest are shown below.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># Results for the genes of interest are not significant</span></span>
<span><span class='va'>sorted_de</span><span class='op'>[</span><span class='va'>sorted_de</span><span class='op'>$</span><span class='va'>SYMBOL</span> <span class='op'><a href='https://rdrr.io/r/base/match.html'>%in%</a></span> <span class='va'>genes_of_interest</span>, <span class='op'>]</span></span></code></pre>
</div>
</details>
<pre><code>      ensembl_transcript_id    ensembl_gene_id external_gene_name
Socs1    ENSMUST00000229866 ENSMUSG00000038037              Socs1
Bcl3     ENSMUST00000135609 ENSMUSG00000053175               Bcl3
Socs3    ENSMUST00000054002 ENSMUSG00000053113              Socs3
      chromosome_name CHR SYMBOL            ENSEMBL       logFC
Socs1              16  16  Socs1 ENSMUSG00000038037 -0.16771833
Bcl3                7   7   Bcl3 ENSMUSG00000053175  0.11362070
Socs3              11  11  Socs3 ENSMUSG00000053113 -0.08669159
       AveExpr          t   P.Value adj.P.Val         B
Socs1 4.219505 -1.2222948 0.2603406 0.9998741 -5.393089
Bcl3  5.557284  0.8798906 0.4075070 0.9998741 -5.931607
Socs3 8.278582 -0.6141822 0.5581169 0.9998741 -6.163115</code></pre>
</div>
<h3 id="plots-of-the-most-differentially-expressed-genes">Plots of the most differentially expressed genes</h3>
<p>Plots of the most differentially expressed genes to see if the test is doing what we want it to do.
There are several Ig genes here, and a collagen gene.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>top_genes</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>sorted_de</span><span class='op'>)</span>, n <span class='op'>=</span> <span class='fl'>6</span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>selected_genes</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/match.html'>match</a></span><span class='op'>(</span><span class='va'>top_genes</span>, <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>)</span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>df</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span></span>
<span>    y <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/t.html'>t</a></span><span class='op'>(</span><span class='va'>lcpm</span><span class='op'>[</span><span class='va'>selected_genes</span>, <span class='op'>]</span><span class='op'>)</span>, </span>
<span>    treatment <span class='op'>=</span> <span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>treatment</span>, </span>
<span>    batch <span class='op'>=</span> <span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>batch</span><span class='op'>)</span></span>
<span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>df</span><span class='op'>)</span><span class='op'>[</span><span class='fl'>1</span><span class='op'>:</span><span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span><span class='op'>(</span><span class='va'>top_genes</span><span class='op'>)</span><span class='op'>]</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>)</span><span class='op'>[</span><span class='va'>selected_genes</span><span class='op'>]</span></span>
<span></span>
<span><span class='va'>df</span><span class='op'>$</span><span class='va'>labels</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='va'>df</span><span class='op'>$</span><span class='va'>batch</span>, <span class='st'>"."</span>, <span class='va'>df</span><span class='op'>$</span><span class='va'>treatment</span><span class='op'>)</span>,</span>
<span>    levels<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"B1.WT"</span>, <span class='st'>"B1.SOCS3"</span>, <span class='st'>"B2.WT"</span>, <span class='st'>"B2.SOCS3"</span><span class='op'>)</span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>plot_list</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span><span class='op'>)</span></span>
<span><span class='kw'>for</span> <span class='op'>(</span><span class='va'>i</span> <span class='kw'>in</span> <span class='fl'>1</span><span class='op'>:</span><span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span><span class='op'>(</span><span class='va'>top_genes</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>{</span></span>
<span>    <span class='va'>plot_list</span><span class='op'>[[</span><span class='va'>i</span><span class='op'>]</span><span class='op'>]</span> <span class='op'>&lt;-</span></span>
<span>        <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span>data <span class='op'>=</span> <span class='va'>df</span>, mapping <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>labels</span>, y <span class='op'>=</span> <span class='va'>.data</span><span class='op'>[[</span><span class='va'>top_genes</span><span class='op'>[</span><span class='va'>i</span><span class='op'>]</span><span class='op'>]</span><span class='op'>]</span>, group <span class='op'>=</span> <span class='va'>labels</span>, colour <span class='op'>=</span> <span class='va'>treatment</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> </span>
<span>            <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span>size <span class='op'>=</span> <span class='fl'>4</span><span class='op'>)</span> <span class='op'>+</span> </span>
<span>            <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggtheme.html'>theme_minimal</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span> </span>
<span>            <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/theme.html'>theme</a></span><span class='op'>(</span>axis.text.x <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/element.html'>element_text</a></span><span class='op'>(</span>angle <span class='op'>=</span> <span class='fl'>45</span>, hjust <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> </span>
<span>            <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Expression (log CPM)"</span><span class='op'>)</span> <span class='op'>+</span> </span>
<span>            <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>treatment_colours</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>            <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ggtitle</a></span><span class='op'>(</span>label <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='va'>top_genes</span><span class='op'>[</span><span class='va'>i</span><span class='op'>]</span><span class='op'>)</span></span>
<span>    <span class='op'>)</span></span>
<span><span class='op'>}</span></span>
<span><span class='fu'>patchwork</span><span class='fu'>::</span><span class='fu'><a href='https://patchwork.data-imaginist.com/reference/wrap_plots.html'>wrap_plots</a></span><span class='op'>(</span><span class='va'>plot_list</span>, ncol <span class='op'>=</span> <span class='fl'>3</span>, guides <span class='op'>=</span> <span class='st'>"collect"</span><span class='op'>)</span> </span></code></pre>
</div>
</details>
<img src="Kallisto_analysis_files/figure-html5/unnamed-chunk-45-1.png" width="624" />
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># Corresponds to:</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span><span class='op'>(</span><span class='va'>de_table</span>, n <span class='op'>=</span> <span class='fl'>6</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<pre><code>          ensembl_transcript_id    ensembl_gene_id external_gene_name
Ighv1-53     ENSMUST00000103523 ENSMUSG00000093894           Ighv1-53
Igkv10-96    ENSMUST00000103328 ENSMUSG00000094420          Igkv10-96
Col18a1      ENSMUST00000081654 ENSMUSG00000001435            Col18a1
Pla2g2d      ENSMUST00000030528 ENSMUSG00000041202            Pla2g2d
Ighv1-82     ENSMUST00000196991 ENSMUSG00000095127           Ighv1-82
Ighv3-6      ENSMUST00000195124 ENSMUSG00000076672            Ighv3-6
          chromosome_name CHR    SYMBOL            ENSEMBL      logFC
Ighv1-53               12  12  Ighv1-53 ENSMUSG00000093894 -4.0124141
Igkv10-96               6   6 Igkv10-96 ENSMUSG00000094420 -1.9378250
Col18a1                10  10   Col18a1 ENSMUSG00000001435  0.8990599
Pla2g2d                 4   4   Pla2g2d ENSMUSG00000041202  1.3628040
Ighv1-82               12  12  Ighv1-82 ENSMUSG00000095127 -2.4104646
Ighv3-6                12  12   Ighv3-6 ENSMUSG00000076672 -2.1316224
           AveExpr          t      P.Value  adj.P.Val        B
Ighv1-53  4.959076 -15.353987 9.758266e-07 0.02124570 3.971764
Igkv10-96 5.608252 -11.850155 5.842989e-06 0.06360678 3.989723
Col18a1   4.994588   9.083638 3.517068e-05 0.24273285 2.661886
Pla2g2d   4.331964   8.659995 4.825114e-05 0.24273285 2.161525
Ighv1-82  4.091629  -8.069133 7.670576e-05 0.24273285 1.737427
Ighv3-6   5.559966  -7.996783 8.134194e-05 0.24273285 1.973738</code></pre>
</div>
<p>The top 20 differentially expressed genes using voom are shown below.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># Top 20 genes</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>sorted_de</span><span class='op'>)</span>, n<span class='op'>=</span><span class='fl'>20</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<pre><code> [1] &quot;Ighv1-53&quot;  &quot;Igkv10-96&quot; &quot;Col18a1&quot;   &quot;Pla2g2d&quot;   &quot;Ighv1-82&quot; 
 [6] &quot;Ighv3-6&quot;   &quot;Igkv4-79&quot;  &quot;Ighv5-4&quot;   &quot;Kera&quot;      &quot;Ptx4&quot;     
[11] &quot;Serpina1d&quot; &quot;Atp6v0d2&quot;  &quot;Scx&quot;       &quot;Ighv2-2&quot;   &quot;Adamts3&quot;  
[16] &quot;Itk&quot;       &quot;St18&quot;      &quot;Spon2&quot;     &quot;Serpina1a&quot; &quot;Gm16867&quot;  </code></pre>
</div>
<h2 id="treatment-with-edgerlrt-method">Treatment with edgeR::LRT method</h2>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>design</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/model.matrix.html'>model.matrix</a></span><span class='op'>(</span><span class='op'>~</span> <span class='va'>treatment</span> <span class='op'>+</span> <span class='va'>batch</span>, data<span class='op'>=</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>)</span> </span>
<span><span class='va'>dge</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/estimateDisp.html'>estimateDisp</a></span><span class='op'>(</span><span class='va'>dge</span>, <span class='va'>design</span><span class='op'>)</span> </span>
<span><span class='va'>fit</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/glmfit.html'>glmFit</a></span><span class='op'>(</span><span class='va'>dge</span>, <span class='va'>design</span><span class='op'>)</span> </span>
<span><span class='va'>lrt</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/glmLRT.html'>glmLRT</a></span><span class='op'>(</span><span class='va'>fit</span>, coef <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> </span>
<span></span>
<span><span class='co'># Find significant genes</span></span>
<span><span class='va'>de_results</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/limma/man/decideTests.html'>decideTests</a></span><span class='op'>(</span><span class='va'>lrt</span>, adjust.method <span class='op'>=</span> <span class='st'>"BH"</span>, p.value <span class='op'>=</span> <span class='fl'>0.05</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/summary.html'>summary</a></span><span class='op'>(</span><span class='va'>de_results</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<pre><code>       treatmentSOCS3
Down               59
NotSig          21691
Up                 22</code></pre>
</div>
<p>Using <code>glmLRT</code> there are 59 genes which are downregulated in SOCS3, and 22 which are upregulated.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># MD plot</span></span>
<span><span class='fu'><a href='https://rdrr.io/pkg/limma/man/plotMD.html'>plotMD</a></span><span class='op'>(</span><span class='va'>lrt</span>, status <span class='op'>=</span> <span class='va'>de_results</span>, values <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='op'>-</span><span class='fl'>1</span>, <span class='fl'>1</span><span class='op'>)</span>, col <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"blue"</span>, <span class='st'>"red"</span><span class='op'>)</span>, legend <span class='op'>=</span> <span class='st'>"topright"</span>, main <span class='op'>=</span> <span class='st'>"SOCS3 vs. WT"</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<p><img src="Kallisto_analysis_files/figure-html5/unnamed-chunk-48-1.png" width="624" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>de_table</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/topTags.html'>topTags</a></span><span class='op'>(</span><span class='va'>lrt</span>, n <span class='op'>=</span> <span class='cn'>Inf</span><span class='op'>)</span><span class='op'>$</span><span class='va'>table</span> </span>
<span><span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span><span class='op'>(</span><span class='va'>de_table</span><span class='op'>)</span> </span></code></pre>
</div>
</details>
<pre><code>          ensembl_transcript_id    ensembl_gene_id external_gene_name
Ighv1-53     ENSMUST00000103523 ENSMUSG00000093894           Ighv1-53
Serpina1d    ENSMUST00000078869 ENSMUSG00000071177          Serpina1d
Igkv4-79     ENSMUST00000103342 ENSMUSG00000076541           Igkv4-79
Igkv10-96    ENSMUST00000103328 ENSMUSG00000094420          Igkv10-96
Igkv9-124    ENSMUST00000196768 ENSMUSG00000096632          Igkv9-124
Ighv3-5      ENSMUST00000195619 ENSMUSG00000076670            Ighv3-5
          chromosome_name CHR    SYMBOL            ENSEMBL      logFC
Ighv1-53               12  12  Ighv1-53 ENSMUSG00000093894  -4.007728
Serpina1d              12  12 Serpina1d ENSMUSG00000071177  -9.776004
Igkv4-79                6   6  Igkv4-79 ENSMUSG00000076541 -10.546102
Igkv10-96               6   6 Igkv10-96 ENSMUSG00000094420  -1.934752
Igkv9-124               6   6 Igkv9-124 ENSMUSG00000096632  -6.219066
Ighv3-5                12  12   Ighv3-5 ENSMUSG00000076670  -7.575426
             logCPM        LR       PValue          FDR
Ighv1-53  6.0916737 143.76671 3.995733e-33 8.699510e-29
Serpina1d 2.5596776 128.73270 7.759510e-30 8.447003e-26
Igkv4-79  0.5236326 120.86874 4.082714e-28 2.962962e-24
Igkv10-96 5.9219024  85.87907 1.912824e-20 1.041150e-16
Igkv9-124 2.7873572  83.08807 7.847782e-20 3.417238e-16
Ighv3-5   0.6886644  52.57793 4.135092e-13 1.500487e-09</code></pre>
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>de_table</span><span class='op'>)</span>, <span class='fl'>20</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<pre><code> [1] &quot;Ighv1-53&quot;  &quot;Serpina1d&quot; &quot;Igkv4-79&quot;  &quot;Igkv10-96&quot; &quot;Igkv9-124&quot;
 [6] &quot;Ighv3-5&quot;   &quot;Ighv1-82&quot;  &quot;Ighv5-4&quot;   &quot;Ighv2-5&quot;   &quot;Ighv3-6&quot;  
[11] &quot;Pla2g2d&quot;   &quot;Igkv4-73&quot;  &quot;Igkv1-133&quot; &quot;Col18a1&quot;   &quot;Ighv1-25&quot; 
[16] &quot;Serpina1c&quot; &quot;Igkv3-5&quot;   &quot;Ighv1-63&quot;  &quot;Ighv1-55&quot;  &quot;Atp6v0d2&quot; </code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># See the values of the DE for genes of interest</span></span>
<span><span class='va'>genes_of_interest</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Socs3"</span>, <span class='st'>"Socs1"</span>, <span class='st'>"Bcl3"</span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>de_table</span><span class='op'>[</span><span class='va'>de_table</span><span class='op'>$</span><span class='va'>SYMBOL</span> <span class='op'><a href='https://rdrr.io/r/base/match.html'>%in%</a></span> <span class='va'>genes_of_interest</span>, <span class='op'>]</span></span></code></pre>
</div>
</details>
<pre><code>      ensembl_transcript_id    ensembl_gene_id external_gene_name
Socs1    ENSMUST00000229866 ENSMUSG00000038037              Socs1
Bcl3     ENSMUST00000135609 ENSMUSG00000053175               Bcl3
Socs3    ENSMUST00000054002 ENSMUSG00000053113              Socs3
      chromosome_name CHR SYMBOL            ENSEMBL      logFC
Socs1              16  16  Socs1 ENSMUSG00000038037 -0.1890986
Bcl3                7   7   Bcl3 ENSMUSG00000053175  0.1079021
Socs3              11  11  Socs3 ENSMUSG00000053113 -0.0964003
        logCPM        LR    PValue       FDR
Socs1 4.235511 0.8994515 0.3429288 0.9999886
Bcl3  5.572940 0.3537078 0.5520217 0.9999886
Socs3 8.363292 0.2785081 0.5976808 0.9999886</code></pre>
</div>
<p>The LRT test also returns a number of Ig genes.</p>
<h3 id="plots-of-the-most-differentially-expressed-genes-1">Plots of the most differentially expressed genes</h3>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>top_genes</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>de_table</span><span class='op'>)</span>, n <span class='op'>=</span> <span class='fl'>6</span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>selected_genes</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/match.html'>match</a></span><span class='op'>(</span><span class='va'>top_genes</span>, <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>)</span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>df</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span></span>
<span>    y <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/t.html'>t</a></span><span class='op'>(</span><span class='va'>lcpm</span><span class='op'>[</span><span class='va'>selected_genes</span>, <span class='op'>]</span><span class='op'>)</span>, </span>
<span>    treatment <span class='op'>=</span> <span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>treatment</span>, </span>
<span>    batch <span class='op'>=</span> <span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>batch</span><span class='op'>)</span></span>
<span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>df</span><span class='op'>)</span><span class='op'>[</span><span class='fl'>1</span><span class='op'>:</span><span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span><span class='op'>(</span><span class='va'>top_genes</span><span class='op'>)</span><span class='op'>]</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>dge</span><span class='op'>)</span><span class='op'>[</span><span class='va'>selected_genes</span><span class='op'>]</span></span>
<span></span>
<span><span class='va'>df</span><span class='op'>$</span><span class='va'>labels</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='va'>df</span><span class='op'>$</span><span class='va'>batch</span>, <span class='st'>"."</span>, <span class='va'>df</span><span class='op'>$</span><span class='va'>treatment</span><span class='op'>)</span>,</span>
<span>    levels<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"B1.WT"</span>, <span class='st'>"B1.SOCS3"</span>, <span class='st'>"B2.WT"</span>, <span class='st'>"B2.SOCS3"</span><span class='op'>)</span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>plot_list</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span><span class='op'>)</span></span>
<span><span class='kw'>for</span> <span class='op'>(</span><span class='va'>i</span> <span class='kw'>in</span> <span class='fl'>1</span><span class='op'>:</span><span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span><span class='op'>(</span><span class='va'>top_genes</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>{</span></span>
<span>    <span class='va'>plot_list</span><span class='op'>[[</span><span class='va'>i</span><span class='op'>]</span><span class='op'>]</span> <span class='op'>&lt;-</span></span>
<span>        <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span>data <span class='op'>=</span> <span class='va'>df</span>, mapping <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>labels</span>, y <span class='op'>=</span> <span class='va'>.data</span><span class='op'>[[</span><span class='va'>top_genes</span><span class='op'>[</span><span class='va'>i</span><span class='op'>]</span><span class='op'>]</span><span class='op'>]</span>, group <span class='op'>=</span> <span class='va'>labels</span>, colour <span class='op'>=</span> <span class='va'>treatment</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> </span>
<span>            <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span>size <span class='op'>=</span> <span class='fl'>4</span><span class='op'>)</span> <span class='op'>+</span> </span>
<span>            <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggtheme.html'>theme_minimal</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span> </span>
<span>            <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/theme.html'>theme</a></span><span class='op'>(</span>axis.text.x <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/element.html'>element_text</a></span><span class='op'>(</span>angle <span class='op'>=</span> <span class='fl'>45</span>, hjust <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> </span>
<span>            <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Expression (log CPM)"</span><span class='op'>)</span> <span class='op'>+</span> </span>
<span>            <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>treatment_colours</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>            <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ggtitle</a></span><span class='op'>(</span>label <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='va'>top_genes</span><span class='op'>[</span><span class='va'>i</span><span class='op'>]</span><span class='op'>)</span></span>
<span>    <span class='op'>)</span></span>
<span><span class='op'>}</span></span>
<span><span class='fu'>patchwork</span><span class='fu'>::</span><span class='fu'><a href='https://patchwork.data-imaginist.com/reference/wrap_plots.html'>wrap_plots</a></span><span class='op'>(</span><span class='va'>plot_list</span>, ncol <span class='op'>=</span> <span class='fl'>3</span>, guides <span class='op'>=</span> <span class='st'>"collect"</span><span class='op'>)</span> </span></code></pre>
</div>
</details>
<img src="Kallisto_analysis_files/figure-html5/unnamed-chunk-51-1.png" width="624" />
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># Corresponds to:</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span><span class='op'>(</span><span class='va'>de_table</span>, n <span class='op'>=</span> <span class='fl'>6</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<pre><code>          ensembl_transcript_id    ensembl_gene_id external_gene_name
Ighv1-53     ENSMUST00000103523 ENSMUSG00000093894           Ighv1-53
Serpina1d    ENSMUST00000078869 ENSMUSG00000071177          Serpina1d
Igkv4-79     ENSMUST00000103342 ENSMUSG00000076541           Igkv4-79
Igkv10-96    ENSMUST00000103328 ENSMUSG00000094420          Igkv10-96
Igkv9-124    ENSMUST00000196768 ENSMUSG00000096632          Igkv9-124
Ighv3-5      ENSMUST00000195619 ENSMUSG00000076670            Ighv3-5
          chromosome_name CHR    SYMBOL            ENSEMBL      logFC
Ighv1-53               12  12  Ighv1-53 ENSMUSG00000093894  -4.007728
Serpina1d              12  12 Serpina1d ENSMUSG00000071177  -9.776004
Igkv4-79                6   6  Igkv4-79 ENSMUSG00000076541 -10.546102
Igkv10-96               6   6 Igkv10-96 ENSMUSG00000094420  -1.934752
Igkv9-124               6   6 Igkv9-124 ENSMUSG00000096632  -6.219066
Ighv3-5                12  12   Ighv3-5 ENSMUSG00000076670  -7.575426
             logCPM        LR       PValue          FDR
Ighv1-53  6.0916737 143.76671 3.995733e-33 8.699510e-29
Serpina1d 2.5596776 128.73270 7.759510e-30 8.447003e-26
Igkv4-79  0.5236326 120.86874 4.082714e-28 2.962962e-24
Igkv10-96 5.9219024  85.87907 1.912824e-20 1.041150e-16
Igkv9-124 2.7873572  83.08807 7.847782e-20 3.417238e-16
Ighv3-5   0.6886644  52.57793 4.135092e-13 1.500487e-09</code></pre>
</div>
<p>The heatmap below shows the top 50 genes with the highest differential expression.
This is not the cleanest heatmap, with many genes which are only expressed highly in one of the three samples for a given treatment.
In other words, many of the effects are driven by strong signals from one sample, rather than consistent differences across all three samples for a condition.
The wild-type samples have higher expression of many Ig genes, while the Socs3 samples have higher expression of genes such as the mitochondrial gene <code>mt-Co2</code> and the collagen gene <code>Col18a1</code>.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># Filter top 50 DE genes with FDR &lt; 0.05</span></span>
<span><span class='va'>de_table_sig</span> <span class='op'>&lt;-</span> <span class='va'>de_table</span><span class='op'>[</span><span class='va'>de_table</span><span class='op'>$</span><span class='va'>FDR</span> <span class='op'>&lt;</span> <span class='fl'>0.05</span>, <span class='op'>]</span></span>
<span><span class='va'>top50_genes</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span><span class='op'>(</span><span class='va'>de_table_sig</span>, <span class='fl'>50</span><span class='op'>)</span><span class='op'>)</span></span>
<span></span>
<span><span class='co'># Log2 CPM-normalized expression matrix</span></span>
<span><span class='va'>lcpm_top50</span> <span class='op'>&lt;-</span> <span class='va'>lcpm</span><span class='op'>[</span><span class='va'>top50_genes</span>, <span class='op'>]</span></span>
<span></span>
<span><span class='co'># Sample annotations: Treatment and Batch</span></span>
<span><span class='va'>annotation_col</span> <span class='op'>&lt;-</span> <span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>[</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"treatment"</span>, <span class='st'>"batch"</span><span class='op'>)</span><span class='op'>]</span></span>
<span></span>
<span><span class='co'># Z-score normalize rows (genes)</span></span>
<span><span class='va'>lcpm_scaled</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/t.html'>t</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/scale.html'>scale</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/t.html'>t</a></span><span class='op'>(</span><span class='va'>lcpm_top50</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span></span>
<span></span>
<span><span class='co'># Plot heatmap</span></span>
<span><span class='fu'><a href='https://rdrr.io/pkg/pheatmap/man/pheatmap.html'>pheatmap</a></span><span class='op'>(</span><span class='va'>lcpm_scaled</span>,</span>
<span>         annotation_col <span class='op'>=</span> <span class='va'>annotation_col</span>,</span>
<span>         annotation_colors <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span></span>
<span>             batch <span class='op'>=</span> <span class='va'>batch_colours</span>,</span>
<span>             treatment <span class='op'>=</span> <span class='va'>treatment_colours</span><span class='op'>)</span>,</span>
<span>         show_rownames <span class='op'>=</span> <span class='cn'>TRUE</span>,</span>
<span>         show_colnames <span class='op'>=</span> <span class='cn'>FALSE</span>,</span>
<span>         cluster_cols <span class='op'>=</span> <span class='cn'>TRUE</span>,</span>
<span>         clustering_distance_rows <span class='op'>=</span> <span class='st'>"euclidean"</span>,</span>
<span>         clustering_distance_cols <span class='op'>=</span> <span class='st'>"euclidean"</span>,</span>
<span>         clustering_method <span class='op'>=</span> <span class='st'>"complete"</span>,</span>
<span>         fontsize_row <span class='op'>=</span> <span class='fl'>6</span>,</span>
<span>         main <span class='op'>=</span> <span class='st'>"Top DE genes for treatment using edgeR LRT \n (Uncorrected)"</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<p><img src="Kallisto_analysis_files/figure-html5/unnamed-chunk-52-1.png" width="624" /></p>
</div>
<p>The heatmap below uses <code>limma::removeBatchEffect</code> to improve the visualisation. Note that the statistical test has not changed, this is just adjusting the values for plotting.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># Filter top 50 DE genes with FDR &lt; 0.05</span></span>
<span><span class='va'>de_table_sig</span> <span class='op'>&lt;-</span> <span class='va'>de_table</span><span class='op'>[</span><span class='va'>de_table</span><span class='op'>$</span><span class='va'>FDR</span> <span class='op'>&lt;</span> <span class='fl'>0.05</span>, <span class='op'>]</span></span>
<span><span class='va'>top50_genes</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span><span class='op'>(</span><span class='va'>de_table_sig</span>, <span class='fl'>50</span><span class='op'>)</span><span class='op'>)</span></span>
<span></span>
<span></span>
<span><span class='co'># Remove batch effect from logCPM for plotting</span></span>
<span><span class='va'>lcpm_corrected</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/limma/man/removeBatchEffect.html'>removeBatchEffect</a></span><span class='op'>(</span><span class='va'>lcpm</span>, batch<span class='op'>=</span><span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>batch</span><span class='op'>)</span></span>
<span></span>
<span><span class='co'># Log2 CPM-normalized expression matrix</span></span>
<span><span class='va'>lcpm_top50</span> <span class='op'>&lt;-</span> <span class='va'>lcpm_corrected</span><span class='op'>[</span><span class='va'>top50_genes</span>, <span class='op'>]</span></span>
<span></span>
<span><span class='co'># Sample annotations: Treatment and Batch</span></span>
<span><span class='va'>annotation_col</span> <span class='op'>&lt;-</span> <span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>[</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"treatment"</span>, <span class='st'>"batch"</span><span class='op'>)</span><span class='op'>]</span></span>
<span></span>
<span><span class='co'># Z-score normalize rows (genes)</span></span>
<span><span class='va'>lcpm_scaled</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/t.html'>t</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/scale.html'>scale</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/t.html'>t</a></span><span class='op'>(</span><span class='va'>lcpm_top50</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span></span>
<span></span>
<span><span class='co'># Plot heatmap</span></span>
<span><span class='fu'><a href='https://rdrr.io/pkg/pheatmap/man/pheatmap.html'>pheatmap</a></span><span class='op'>(</span><span class='va'>lcpm_scaled</span>,</span>
<span>         annotation_col <span class='op'>=</span> <span class='va'>annotation_col</span>,</span>
<span>         annotation_colors <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span></span>
<span>             batch <span class='op'>=</span> <span class='va'>batch_colours</span>,</span>
<span>             treatment <span class='op'>=</span> <span class='va'>treatment_colours</span><span class='op'>)</span>,</span>
<span>         show_rownames <span class='op'>=</span> <span class='cn'>TRUE</span>,</span>
<span>         show_colnames <span class='op'>=</span> <span class='cn'>FALSE</span>,</span>
<span>         cluster_cols <span class='op'>=</span> <span class='cn'>TRUE</span>,</span>
<span>         clustering_distance_rows <span class='op'>=</span> <span class='st'>"euclidean"</span>,</span>
<span>         clustering_distance_cols <span class='op'>=</span> <span class='st'>"euclidean"</span>,</span>
<span>         clustering_method <span class='op'>=</span> <span class='st'>"complete"</span>,</span>
<span>         fontsize_row <span class='op'>=</span> <span class='fl'>6</span>,</span>
<span>         main <span class='op'>=</span> <span class='st'>"Top DE genes for treatment using edgeR LRT \n (Corrected)"</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<p><img src="Kallisto_analysis_files/figure-html5/unnamed-chunk-53-1.png" width="624" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># The LRT de_results object is already sorted by FDR.</span></span>
<span></span>
<span><span class='co'># Write to a file</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/files2.html'>dir.create</a></span><span class='op'>(</span><span class='fu'><a href='https://here.r-lib.org/reference/here.html'>here</a></span><span class='op'>(</span><span class='st'>"output/DEGs/kallisto_analysis"</span><span class='op'>)</span>, recursive <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/utils/write.table.html'>write.table</a></span><span class='op'>(</span><span class='va'>de_table</span>, file<span class='op'>=</span><span class='fu'><a href='https://here.r-lib.org/reference/here.html'>here</a></span><span class='op'>(</span><span class='st'>"output/DEGs/kallisto_analysis/kallisto_treatment_LRT.tsv"</span><span class='op'>)</span>,</span>
<span>            sep<span class='op'>=</span><span class='st'>"\t"</span>, quote<span class='op'>=</span><span class='cn'>FALSE</span>, row.names<span class='op'>=</span><span class='cn'>FALSE</span><span class='op'>)</span></span></code></pre>
</div>
</details>
</div>
<p>A table of the differential expression results are saved in <code>output/DEGs/kallisto_analysis/kallisto_treatment_LRT.tsv</code>.
The rows are sorted so the genes with the lowest FDR are at the top.</p>
<h2 id="batch-with-limmavoom-method">Batch with limma::voom method</h2>
<p>There is a strong batch effect in this data set.
In this section, I perform a DE test of <em>just batch</em>.
This shows the genes that are upregulated in one batch but not the other, and are purely technical effects.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>design</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/model.matrix.html'>model.matrix</a></span><span class='op'>(</span><span class='op'>~</span><span class='fl'>0</span><span class='op'>+</span><span class='va'>batch</span><span class='op'>+</span><span class='va'>treatment</span>, data <span class='op'>=</span> <span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>)</span></span>
<span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>design</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/grep.html'>gsub</a></span><span class='op'>(</span><span class='st'>"batch"</span>, <span class='st'>""</span>, <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>design</span><span class='op'>)</span><span class='op'>)</span></span>
<span><span class='va'>design</span></span></code></pre>
</div>
</details>
<pre><code>           B1 B2 treatmentSOCS3
WT_1_B2     0  1              0
SOCS3_1_B2  0  1              1
SOCS3_2_B1  1  0              1
SOCS3_3_B1  1  0              1
WT_2_B1     1  0              0
WT_3_B1     1  0              0
attr(,&quot;assign&quot;)
[1] 1 1 2
attr(,&quot;contrasts&quot;)
attr(,&quot;contrasts&quot;)$batch
[1] &quot;contr.treatment&quot;

attr(,&quot;contrasts&quot;)$treatment
[1] &quot;contr.treatment&quot;</code></pre>
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>contr.matrix</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/limma/man/makeContrasts.html'>makeContrasts</a></span><span class='op'>(</span></span>
<span>   B2vsB1 <span class='op'>=</span> <span class='va'>B2</span> <span class='op'>-</span> <span class='va'>B1</span>,</span>
<span>   levels <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>design</span><span class='op'>)</span><span class='op'>)</span></span>
<span><span class='va'>contr.matrix</span></span></code></pre>
</div>
</details>
<pre><code>                Contrasts
Levels           B2vsB1
  B1                 -1
  B2                  1
  treatmentSOCS3      0</code></pre>
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>v</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/limma/man/voom.html'>voom</a></span><span class='op'>(</span><span class='va'>dge</span>, <span class='va'>design</span>, plot<span class='op'>=</span><span class='cn'>TRUE</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<img src="Kallisto_analysis_files/figure-html5/unnamed-chunk-55-1.png" width="624" />
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>vfit</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/limma/man/lmFit.html'>lmFit</a></span><span class='op'>(</span><span class='va'>v</span>, <span class='va'>design</span><span class='op'>)</span></span>
<span><span class='va'>vfit</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/limma/man/contrasts.fit.html'>contrasts.fit</a></span><span class='op'>(</span><span class='va'>vfit</span>, contrasts<span class='op'>=</span><span class='va'>contr.matrix</span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>efit</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/limma/man/ebayes.html'>eBayes</a></span><span class='op'>(</span><span class='va'>vfit</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/pkg/limma/man/plotSA.html'>plotSA</a></span><span class='op'>(</span><span class='va'>efit</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<p><img src="Kallisto_analysis_files/figure-html5/unnamed-chunk-55-2.png" width="624" /></p>
</div>
<p>There are hundreds of genes in each direction which are significant when testing differences between batch 1 and batch 2.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>tfit</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/limma/man/ebayes.html'>treat</a></span><span class='op'>(</span><span class='va'>vfit</span>, lfc<span class='op'>=</span><span class='fl'>0.1</span><span class='op'>)</span></span>
<span><span class='va'>de_results</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/limma/man/decideTests.html'>decideTests</a></span><span class='op'>(</span><span class='va'>tfit</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/summary.html'>summary</a></span><span class='op'>(</span><span class='va'>de_results</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<pre><code>       B2vsB1
Down      997
NotSig  20307
Up        468</code></pre>
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># Sort the top table by adjusted P value</span></span>
<span><span class='va'>de_table</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/limma/man/toptable.html'>topTable</a></span><span class='op'>(</span><span class='va'>tfit</span>, number<span class='op'>=</span><span class='cn'>Inf</span>, sort.by<span class='op'>=</span><span class='st'>"P"</span><span class='op'>)</span></span>
<span></span>
<span><span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>de_table</span><span class='op'>)</span>, n<span class='op'>=</span><span class='fl'>20</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<pre><code> [1] &quot;Il1a&quot;     &quot;Rgs5&quot;     &quot;Tnf&quot;      &quot;Il1b&quot;     &quot;Rasgef1b&quot; &quot;Mfap4&quot;   
 [7] &quot;Cd14&quot;     &quot;Thbs4&quot;    &quot;Il10&quot;     &quot;Maff&quot;     &quot;Ccl4&quot;     &quot;Trim30b&quot; 
[13] &quot;Cd83&quot;     &quot;Sqstm1&quot;   &quot;Jcad&quot;     &quot;Sparcl1&quot;  &quot;Timp3&quot;    &quot;Col18a1&quot; 
[19] &quot;Clic4&quot;    &quot;Marcksl1&quot;</code></pre>
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>sorted_de</span> <span class='op'>&lt;-</span> <span class='va'>de_table</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/order.html'>order</a></span><span class='op'>(</span><span class='va'>de_table</span><span class='op'>$</span><span class='va'>adj.P.Val</span><span class='op'>)</span>, <span class='op'>]</span></span>
<span></span>
<span><span class='co'># Write to a file</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/files2.html'>dir.create</a></span><span class='op'>(</span><span class='fu'><a href='https://here.r-lib.org/reference/here.html'>here</a></span><span class='op'>(</span><span class='st'>"output/DEGs/batch_effect"</span><span class='op'>)</span>, recursive <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/utils/write.table.html'>write.table</a></span><span class='op'>(</span><span class='va'>sorted_de</span>, file<span class='op'>=</span><span class='fu'><a href='https://here.r-lib.org/reference/here.html'>here</a></span><span class='op'>(</span><span class='st'>"output/DEGs/batch_effect/kallisto_batch_voom.tsv"</span><span class='op'>)</span>,</span>
<span>            sep<span class='op'>=</span><span class='st'>"\t"</span>, quote<span class='op'>=</span><span class='cn'>FALSE</span>, row.names<span class='op'>=</span><span class='cn'>FALSE</span><span class='op'>)</span></span></code></pre>
</div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># Filter DE genes with FDR &lt; 0.05</span></span>
<span><span class='va'>de_table_sig</span> <span class='op'>&lt;-</span> <span class='va'>de_table</span><span class='op'>[</span><span class='va'>de_table</span><span class='op'>$</span><span class='va'>adj.P.Val</span> <span class='op'>&lt;</span> <span class='fl'>0.05</span>, <span class='op'>]</span></span>
<span></span>
<span><span class='co'># Select top 50 DE genes by smallest FDR</span></span>
<span><span class='va'>top50_genes</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>de_table_sig</span><span class='op'>)</span>, <span class='fl'>50</span><span class='op'>)</span></span>
<span></span>
<span><span class='co'># Use voom-transformed expression matrix</span></span>
<span><span class='va'>expr_mat</span> <span class='op'>&lt;-</span> <span class='va'>v</span><span class='op'>$</span><span class='va'>E</span><span class='op'>[</span><span class='va'>top50_genes</span>, <span class='op'>]</span></span>
<span></span>
<span><span class='co'># Sample annotations</span></span>
<span><span class='va'>annotation_col</span> <span class='op'>&lt;-</span> <span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>[</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"batch"</span>, <span class='st'>"treatment"</span><span class='op'>)</span><span class='op'>]</span></span>
<span></span>
<span><span class='co'># Z-score normalize rows (genes)</span></span>
<span><span class='va'>lcpm_scaled</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/t.html'>t</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/scale.html'>scale</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/t.html'>t</a></span><span class='op'>(</span><span class='va'>expr_mat</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span></span>
<span></span>
<span><span class='co'># Plot heatmap</span></span>
<span><span class='fu'><a href='https://rdrr.io/pkg/pheatmap/man/pheatmap.html'>pheatmap</a></span><span class='op'>(</span><span class='va'>lcpm_scaled</span>,</span>
<span>         annotation_col <span class='op'>=</span> <span class='va'>annotation_col</span>,</span>
<span>         annotation_colors <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span></span>
<span>             batch <span class='op'>=</span> <span class='va'>batch_colours</span>,</span>
<span>             treatment <span class='op'>=</span> <span class='va'>treatment_colours</span><span class='op'>)</span>,</span>
<span>         show_rownames <span class='op'>=</span> <span class='cn'>TRUE</span>,</span>
<span>         show_colnames <span class='op'>=</span> <span class='cn'>FALSE</span>,</span>
<span>         cluster_cols <span class='op'>=</span> <span class='cn'>FALSE</span>,</span>
<span>         clustering_distance_rows <span class='op'>=</span> <span class='st'>"euclidean"</span>,</span>
<span>         clustering_distance_cols <span class='op'>=</span> <span class='st'>"euclidean"</span>,</span>
<span>         clustering_method <span class='op'>=</span> <span class='st'>"complete"</span>,</span>
<span>         fontsize_row <span class='op'>=</span> <span class='fl'>6</span>,</span>
<span>         main <span class='op'>=</span> <span class='st'>"Top DE genes for batch using voom"</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<p><img src="Kallisto_analysis_files/figure-html5/unnamed-chunk-57-1.png" width="624" /></p>
</div>
<h2 id="de-if-we-remove-wt_2_b1">DE if we remove WT_2_B1</h2>
<p>This analysis uses the <code>limma::voom</code> method.
We remove the problematic sample in the batch 1 group and reprocess the data.
MDS plots of the data without this sample are shown below.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>dge2</span> <span class='op'>&lt;-</span> <span class='va'>dge</span><span class='op'>[</span>, <span class='va'>dge</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span> <span class='op'>!=</span> <span class='st'>"WT_2_B1"</span><span class='op'>]</span></span>
<span></span>
<span><span class='va'>lcpm2</span> <span class='op'>&lt;-</span> <span class='fu'>edgeR</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/cpm.html'>cpm</a></span><span class='op'>(</span><span class='va'>dge2</span>, log <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span></span>
<span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/par.html'>par</a></span><span class='op'>(</span>mfrow <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>2</span>,<span class='fl'>2</span><span class='op'>)</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/pkg/limma/man/plotMDS.html'>plotMDS</a></span><span class='op'>(</span><span class='va'>lcpm2</span>, labels<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>lcpm2</span><span class='op'>)</span>, col<span class='op'>=</span><span class='va'>group_colours</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>lcpm2</span><span class='op'>)</span><span class='op'>]</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/title.html'>title</a></span><span class='op'>(</span>main<span class='op'>=</span><span class='st'>"A. Sample groups"</span><span class='op'>)</span></span>
<span></span>
<span><span class='fu'><a href='https://rdrr.io/pkg/limma/man/plotMDS.html'>plotMDS</a></span><span class='op'>(</span><span class='va'>lcpm2</span>, labels<span class='op'>=</span><span class='va'>dge2</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>treatment</span>, col<span class='op'>=</span><span class='va'>treatment_colours</span><span class='op'>[</span><span class='va'>dge2</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>treatment</span><span class='op'>]</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/title.html'>title</a></span><span class='op'>(</span>main<span class='op'>=</span><span class='st'>"B. Treatment"</span><span class='op'>)</span></span>
<span></span>
<span><span class='fu'><a href='https://rdrr.io/pkg/limma/man/plotMDS.html'>plotMDS</a></span><span class='op'>(</span><span class='va'>lcpm2</span>, labels<span class='op'>=</span><span class='va'>dge2</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>batch</span>, col<span class='op'>=</span><span class='va'>batch_colours</span><span class='op'>[</span><span class='va'>dge2</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>batch</span><span class='op'>]</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/title.html'>title</a></span><span class='op'>(</span>main<span class='op'>=</span><span class='st'>"C. Batch"</span><span class='op'>)</span></span>
<span></span>
<span><span class='co'># MDS plot with library size</span></span>
<span><span class='va'>libsizes2</span> <span class='op'>&lt;-</span> <span class='va'>dge2</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>lib.size</span></span>
<span></span>
<span><span class='co'># Create a viridis color function</span></span>
<span><span class='va'>color_func</span> <span class='op'>&lt;-</span> <span class='fu'>scales</span><span class='fu'>::</span><span class='fu'><a href='https://scales.r-lib.org/reference/col_numeric.html'>col_numeric</a></span><span class='op'>(</span><span class='fu'>viridis</span><span class='fu'>::</span><span class='fu'><a href='https://sjmgarnier.github.io/viridisLite/reference/viridis.html'>viridis</a></span><span class='op'>(</span><span class='fl'>100</span><span class='op'>)</span>, domain <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/range.html'>range</a></span><span class='op'>(</span><span class='va'>libsizes2</span><span class='op'>)</span><span class='op'>)</span></span>
<span></span>
<span><span class='co'># Assign colors according to library sizes</span></span>
<span><span class='va'>point_colors2</span> <span class='op'>&lt;-</span> <span class='fu'>color_func</span><span class='op'>(</span><span class='va'>libsizes2</span><span class='op'>)</span></span>
<span></span>
<span><span class='fu'><a href='https://rdrr.io/pkg/limma/man/plotMDS.html'>plotMDS</a></span><span class='op'>(</span><span class='va'>lcpm2</span>, labels<span class='op'>=</span><span class='cn'>NULL</span>, pch <span class='op'>=</span> <span class='fl'>16</span>, cex <span class='op'>=</span> <span class='fl'>4</span>, col<span class='op'>=</span><span class='va'>point_colors2</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/title.html'>title</a></span><span class='op'>(</span>main<span class='op'>=</span><span class='st'>"D. Library Size"</span><span class='op'>)</span> </span></code></pre>
</div>
</details>
<p><img src="Kallisto_analysis_files/figure-html5/unnamed-chunk-58-1.png" width="624" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>design</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/model.matrix.html'>model.matrix</a></span><span class='op'>(</span><span class='op'>~</span><span class='fl'>0</span><span class='op'>+</span><span class='va'>treatment</span><span class='op'>+</span><span class='va'>batch</span>, data <span class='op'>=</span> <span class='va'>dge2</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>)</span></span>
<span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>design</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/grep.html'>gsub</a></span><span class='op'>(</span><span class='st'>"treatment|batch"</span>, <span class='st'>""</span>, <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>design</span><span class='op'>)</span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>contr.matrix</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/limma/man/makeContrasts.html'>makeContrasts</a></span><span class='op'>(</span></span>
<span>   SOCS3vsWT <span class='op'>=</span> <span class='va'>SOCS3</span> <span class='op'>-</span> <span class='va'>WT</span>,</span>
<span>   levels <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>design</span><span class='op'>)</span><span class='op'>)</span></span>
<span><span class='va'>contr.matrix</span></span></code></pre>
</div>
</details>
<pre><code>       Contrasts
Levels  SOCS3vsWT
  WT           -1
  SOCS3         1
  B2            0</code></pre>
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>v</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/limma/man/voom.html'>voom</a></span><span class='op'>(</span><span class='va'>dge2</span>, <span class='va'>design</span>, plot<span class='op'>=</span><span class='cn'>TRUE</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<img src="Kallisto_analysis_files/figure-html5/unnamed-chunk-59-1.png" width="624" />
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>vfit</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/limma/man/lmFit.html'>lmFit</a></span><span class='op'>(</span><span class='va'>v</span>, <span class='va'>design</span><span class='op'>)</span></span>
<span><span class='va'>vfit</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/limma/man/contrasts.fit.html'>contrasts.fit</a></span><span class='op'>(</span><span class='va'>vfit</span>, contrasts<span class='op'>=</span><span class='va'>contr.matrix</span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>efit</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/limma/man/ebayes.html'>eBayes</a></span><span class='op'>(</span><span class='va'>vfit</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/pkg/limma/man/plotSA.html'>plotSA</a></span><span class='op'>(</span><span class='va'>efit</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<img src="Kallisto_analysis_files/figure-html5/unnamed-chunk-59-2.png" width="624" />
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>tfit</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/limma/man/ebayes.html'>treat</a></span><span class='op'>(</span><span class='va'>vfit</span>, lfc<span class='op'>=</span><span class='fl'>0.1</span><span class='op'>)</span></span>
<span><span class='va'>de_results</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/limma/man/decideTests.html'>decideTests</a></span><span class='op'>(</span><span class='va'>tfit</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/summary.html'>summary</a></span><span class='op'>(</span><span class='va'>de_results</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<pre><code>       SOCS3vsWT
Down          11
NotSig     21761
Up             0</code></pre>
</div>
<p>There are 11 genes which are more highly expressed in the controls.
The Ig genes and Jchain are related to B-cells and plasma cells.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># Sort the top table by adjusted P value</span></span>
<span><span class='va'>sorted_de</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/limma/man/toptable.html'>topTable</a></span><span class='op'>(</span><span class='va'>tfit</span>, number<span class='op'>=</span><span class='cn'>Inf</span>, sort.by<span class='op'>=</span><span class='st'>"P"</span><span class='op'>)</span></span>
<span><span class='va'>sorted_de</span> <span class='op'>&lt;-</span> <span class='va'>sorted_de</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/order.html'>order</a></span><span class='op'>(</span><span class='va'>sorted_de</span><span class='op'>$</span><span class='va'>adj.P.Val</span><span class='op'>)</span>, <span class='op'>]</span></span>
<span></span>
<span><span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>sorted_de</span><span class='op'>)</span>, n<span class='op'>=</span><span class='fl'>20</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<pre><code> [1] &quot;Igkv6-23&quot;  &quot;Ighv1-53&quot;  &quot;Igkv6-15&quot;  &quot;Igkv10-96&quot; &quot;Ighv1-55&quot; 
 [6] &quot;Jchain&quot;    &quot;Tox&quot;       &quot;Ighv1-82&quot;  &quot;Igkv9-124&quot; &quot;Igha&quot;     
[11] &quot;Gpr55&quot;     &quot;Ighv1-72&quot;  &quot;Cnmd&quot;      &quot;Ighv1-58&quot;  &quot;Pla2g2d&quot;  
[16] &quot;Ighv5-4&quot;   &quot;Serpina1d&quot; &quot;Bcl11b&quot;    &quot;Tnfsf8&quot;    &quot;Ptpn5&quot;    </code></pre>
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># Write to a file</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/utils/write.table.html'>write.table</a></span><span class='op'>(</span><span class='va'>sorted_de</span>, file<span class='op'>=</span><span class='fu'><a href='https://here.r-lib.org/reference/here.html'>here</a></span><span class='op'>(</span><span class='st'>"output/DEGs/kallisto_analysis/kallisto_5samples_treatment_voom.tsv"</span><span class='op'>)</span>,</span>
<span>            sep<span class='op'>=</span><span class='st'>"\t"</span>, quote<span class='op'>=</span><span class='cn'>FALSE</span>, row.names<span class='op'>=</span><span class='cn'>FALSE</span><span class='op'>)</span></span></code></pre>
</div>
</details>
</div>
<h2 id="de-with-random-treatment-labels">DE with random treatment labels</h2>
<p>If we randomly permute the treatment labels, i.e., intentionally mess up our data, what does the LRT test look like?
This section uses <code>limma::voom()</code>.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='fu'><a href='https://rdrr.io/r/base/Random.html'>set.seed</a></span><span class='op'>(</span><span class='fl'>123</span><span class='op'>)</span></span></code></pre>
</div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># Copy the object</span></span>
<span><span class='va'>dge_</span> <span class='op'>&lt;-</span> <span class='va'>dge</span></span>
<span></span>
<span><span class='co'># Shuffle treatment labels</span></span>
<span><span class='va'>dge_</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>treatment</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/sample.html'>sample</a></span><span class='op'>(</span><span class='va'>dge_</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>treatment</span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>design_</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/model.matrix.html'>model.matrix</a></span><span class='op'>(</span><span class='op'>~</span> <span class='va'>treatment</span> <span class='op'>+</span> <span class='va'>batch</span>, data <span class='op'>=</span> <span class='va'>dge_</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>)</span> </span>
<span><span class='va'>dge_</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/estimateDisp.html'>estimateDisp</a></span><span class='op'>(</span><span class='va'>dge_</span>, <span class='va'>design_</span><span class='op'>)</span> </span>
<span><span class='va'>fit_</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/glmfit.html'>glmFit</a></span><span class='op'>(</span><span class='va'>dge_</span>, <span class='va'>design_</span><span class='op'>)</span> </span>
<span><span class='va'>lrt_</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/glmLRT.html'>glmLRT</a></span><span class='op'>(</span><span class='va'>fit_</span>, coef <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> </span>
<span></span>
<span><span class='co'># Find significant genes</span></span>
<span><span class='va'>de_results_</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/limma/man/decideTests.html'>decideTests</a></span><span class='op'>(</span><span class='va'>lrt_</span>, adjust.method <span class='op'>=</span> <span class='st'>"BH"</span>, p.value <span class='op'>=</span> <span class='fl'>0.05</span><span class='op'>)</span></span></code></pre>
</div>
</details>
</div>
<p>There are far fewer DE genes in this case, which shows that our signal above is likely to be real.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='fu'><a href='https://rdrr.io/r/base/summary.html'>summary</a></span><span class='op'>(</span><span class='va'>de_results_</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<pre><code>       treatmentSOCS3
Down                4
NotSig          21767
Up                  1</code></pre>
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>de_table_</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/topTags.html'>topTags</a></span><span class='op'>(</span><span class='va'>lrt_</span>, n <span class='op'>=</span> <span class='cn'>Inf</span><span class='op'>)</span><span class='op'>$</span><span class='va'>table</span> </span>
<span><span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span><span class='op'>(</span><span class='va'>de_table_</span>, n<span class='op'>=</span><span class='fl'>20</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<pre><code>          ensembl_transcript_id    ensembl_gene_id external_gene_name
Gm2007       ENSMUST00000177981 ENSMUSG00000094932             Gm2007
Gm15229      ENSMUST00000146636 ENSMUSG00000085880            Gm15229
Foxc1        ENSMUST00000062292 ENSMUSG00000050295              Foxc1
Ighv1-75     ENSMUST00000103544 ENSMUSG00000096020           Ighv1-75
Ighv8-5      ENSMUST00000194257 ENSMUSG00000102364            Ighv8-5
Potefam3a    ENSMUST00000084046 ENSMUSG00000074449          Potefam3a
Myh4         ENSMUST00000018632 ENSMUSG00000057003               Myh4
Ighv3-8      ENSMUST00000103483 ENSMUSG00000076674            Ighv3-8
Kcnk18       ENSMUST00000065204 ENSMUSG00000040901             Kcnk18
Lum          ENSMUST00000038160 ENSMUSG00000036446                Lum
Mroh4        ENSMUST00000137963 ENSMUSG00000022603              Mroh4
Bglap        ENSMUST00000076048 ENSMUSG00000074483              Bglap
Dsc3         ENSMUST00000115848 ENSMUSG00000059898               Dsc3
Alpl         ENSMUST00000030551 ENSMUSG00000028766               Alpl
Stmnd1       ENSMUST00000076622 ENSMUSG00000063529             Stmnd1
Enpp3        ENSMUST00000020169 ENSMUSG00000019989              Enpp3
Kpna2-ps     ENSMUST00000120025 ENSMUSG00000083672           Kpna2-ps
Gm12226      ENSMUST00000144172 ENSMUSG00000078151            Gm12226
Igkv5-39     ENSMUST00000103370 ENSMUSG00000076569           Igkv5-39
Ighv3-4      ENSMUST00000193408 ENSMUSG00000103939            Ighv3-4
          chromosome_name CHR    SYMBOL            ENSEMBL      logFC
Gm2007                  2   2    Gm2007 ENSMUSG00000094932 -9.7052087
Gm15229                 X   X   Gm15229 ENSMUSG00000085880  7.2062234
Foxc1                  13  13     Foxc1 ENSMUSG00000050295 -0.8019353
Ighv1-75               12  12  Ighv1-75 ENSMUSG00000096020 -3.7895471
Ighv8-5                12  12   Ighv8-5 ENSMUSG00000102364 -6.9525198
Potefam3a               8   8 Potefam3a ENSMUSG00000074449  6.8569197
Myh4                   11  11      Myh4 ENSMUSG00000057003  5.1951482
Ighv3-8                12  12   Ighv3-8 ENSMUSG00000076674 -4.6346699
Kcnk18                 19  19    Kcnk18 ENSMUSG00000040901 -6.3120558
Lum                    10  10       Lum ENSMUSG00000036446 -1.0588937
Mroh4                  15  15     Mroh4 ENSMUSG00000022603  6.3786357
Bglap                   3   3     Bglap ENSMUSG00000074483 -0.8781689
Dsc3                   18  18      Dsc3 ENSMUSG00000059898  6.4164619
Alpl                    4   4      Alpl ENSMUSG00000028766 -0.6351976
Stmnd1                 13  13    Stmnd1 ENSMUSG00000063529  6.2499584
Enpp3                  10  10     Enpp3 ENSMUSG00000019989 -1.0425094
Kpna2-ps               17  17  Kpna2-ps ENSMUSG00000083672  7.2980602
Gm12226                11  11   Gm12226 ENSMUSG00000078151 -5.0751752
Igkv5-39                6   6  Igkv5-39 ENSMUSG00000076569 -2.4204263
Ighv3-4                12  12   Ighv3-4 ENSMUSG00000103939  7.9808127
              logCPM       LR       PValue          FDR
Gm2007    -0.2703938 66.79157 3.017859e-16 6.570482e-12
Gm15229   -2.5835771 23.12997 1.514125e-06 1.132512e-02
Foxc1      6.0478221 23.07195 1.560508e-06 1.132512e-02
Ighv1-75   2.6232300 21.42692 3.675736e-06 1.602908e-02
Ighv8-5   -2.8281855 21.42411 3.681123e-06 1.602908e-02
Potefam3a -2.7869402 18.82521 1.432611e-05 5.198469e-02
Myh4      -1.3660936 18.17569 2.014340e-05 5.894397e-02
Ighv3-8    3.1170949 18.03759 2.165863e-05 5.894397e-02
Kcnk18    -3.3130169 17.20649 3.352893e-05 8.053663e-02
Lum        8.6332902 16.89493 3.950702e-05 8.053663e-02
Mroh4     -3.2180534 16.83308 4.081556e-05 8.053663e-02
Bglap      6.0856786 16.66580 4.457736e-05 8.053663e-02
Dsc3      -3.1228978 16.52203 4.808820e-05 8.053663e-02
Alpl       7.3573092 16.04277 6.192748e-05 8.109438e-02
Stmnd1    -3.2621636 16.01031 6.299860e-05 8.109438e-02
Enpp3      3.6304868 15.99791 6.341246e-05 8.109438e-02
Kpna2-ps  -2.3413436 15.94451 6.522661e-05 8.109438e-02
Gm12226   -2.2529186 15.89247 6.704478e-05 8.109438e-02
Igkv5-39   5.0866049 15.30925 9.126871e-05 9.706066e-02
Ighv3-4   -1.6262437 15.28273 9.255876e-05 9.706066e-02</code></pre>
</div>
<h1 id="concluding-remarks">Concluding remarks</h1>
<p>The <a href="../output/DEG%20s/"><code>output/DEGs/</code></a> directory contains CSV files summarising the differential expression results. The following are available on request:</p>
<ul>
<li>Full CSV tables of any data presented.</li>
<li>PDF/PNG files of any static plots.</li>
</ul>
<div class="sourceCode" id="cb28"><pre class="sourceCode r distill-force-highlighting-css"><code class="sourceCode r"></code></pre></div>
<!--radix_placeholder_article_footer-->
<!--/radix_placeholder_article_footer-->
</div>

<div class="d-appendix">
</div>


<!--radix_placeholder_site_after_body-->
<!--/radix_placeholder_site_after_body-->
<!--radix_placeholder_appendices-->
<div class="appendix-bottom"></div>
<!--/radix_placeholder_appendices-->
<!--radix_placeholder_navigation_after_body-->
<!--/radix_placeholder_navigation_after_body-->

</body>

</html>
